<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 17 Survival Analysis | As a Statistician</title>
  <meta name="description" content="This book involves different statistical principles and methods, including the application of SAS and R, and aims to accumulate personal statistical knowledge." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 17 Survival Analysis | As a Statistician" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book involves different statistical principles and methods, including the application of SAS and R, and aims to accumulate personal statistical knowledge." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 17 Survival Analysis | As a Statistician" />
  
  <meta name="twitter:description" content="This book involves different statistical principles and methods, including the application of SAS and R, and aims to accumulate personal statistical knowledge." />
  

<meta name="author" content="Zehui Bai" />


<meta name="date" content="2022-05-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="proportion-response-regression.html"/>
<link rel="next" href="advanced-survival-analysis.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/d3-3.5.17/d3.min.js"></script>
<link href="libs/markmap-0.3.3/view.mindmap.css" rel="stylesheet" />
<script src="libs/markmap-0.3.3/view.mindmap.js"></script>
<script src="libs/markmap-0.3.3/plugins/parsemd.min.js"></script>
<script src="libs/markmap-binding-1.3.2/markmap.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">As a Statistician</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html"><i class="fa fa-check"></i><b>2</b> Data Management in SAS</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#input-data-into-sas"><i class="fa fa-check"></i><b>2.1</b> Input data into SAS</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#native-datasets-input"><i class="fa fa-check"></i><b>2.1.1</b> Native datasets input</a></li>
<li class="chapter" data-level="2.1.2" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#reading-free-formatted-data-instream"><i class="fa fa-check"></i><b>2.1.2</b> Reading free formatted data instream</a></li>
<li class="chapter" data-level="2.1.3" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#reading-fixed-formatted-data-instream"><i class="fa fa-check"></i><b>2.1.3</b> Reading fixed formatted data instream</a></li>
<li class="chapter" data-level="2.1.4" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#infile-reading-fixed-formatted-data-from-an-external-file"><i class="fa fa-check"></i><b>2.1.4</b> INFILE: Reading fixed formatted data from an external file</a></li>
<li class="chapter" data-level="2.1.5" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#write-the-sas-file"><i class="fa fa-check"></i><b>2.1.5</b> Write the SAS File</a></li>
<li class="chapter" data-level="2.1.6" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#import-csv-data"><i class="fa fa-check"></i><b>2.1.6</b> Import csv data</a></li>
<li class="chapter" data-level="2.1.7" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#import"><i class="fa fa-check"></i><b>2.1.7</b> %Import</a></li>
<li class="chapter" data-level="2.1.8" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#url"><i class="fa fa-check"></i><b>2.1.8</b> URL</a></li>
<li class="chapter" data-level="2.1.9" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#infile-read-multiple-raw-data-files"><i class="fa fa-check"></i><b>2.1.9</b> Infile Read multiple raw data files</a></li>
<li class="chapter" data-level="2.1.10" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#generate-automatic-file-name"><i class="fa fa-check"></i><b>2.1.10</b> Generate automatic file name</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#format-the-variables"><i class="fa fa-check"></i><b>2.2</b> Format the variables</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#proc-format"><i class="fa fa-check"></i><b>2.2.1</b> Proc format</a></li>
<li class="chapter" data-level="2.2.2" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#copy-and-combine-sas-format-libraries"><i class="fa fa-check"></i><b>2.2.2</b> Copy and combine SAS format libraries</a></li>
<li class="chapter" data-level="2.2.3" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#build-a-format-from-a-dataset"><i class="fa fa-check"></i><b>2.2.3</b> Build a format from a dataset</a></li>
<li class="chapter" data-level="2.2.4" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#output-format-as-datasets"><i class="fa fa-check"></i><b>2.2.4</b> Output format as datasets</a></li>
<li class="chapter" data-level="2.2.5" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#delete-the-format"><i class="fa fa-check"></i><b>2.2.5</b> Delete the format</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#array"><i class="fa fa-check"></i><b>2.3</b> Array</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#defining"><i class="fa fa-check"></i><b>2.3.1</b> Defining</a></li>
<li class="chapter" data-level="2.3.2" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#format-multiple-variables"><i class="fa fa-check"></i><b>2.3.2</b> Format multiple variables</a></li>
<li class="chapter" data-level="2.3.3" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#arrays"><i class="fa fa-check"></i><b>2.3.3</b> 2*2 arrays</a></li>
<li class="chapter" data-level="2.3.4" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#dynamic-element-list-using-macro-variables"><i class="fa fa-check"></i><b>2.3.4</b> Dynamic Element List using Macro Variables</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#retain"><i class="fa fa-check"></i><b>2.4</b> Retain</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#generate-serial-number"><i class="fa fa-check"></i><b>2.4.1</b> Generate Serial Number</a></li>
<li class="chapter" data-level="2.4.2" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#change-from-basilne"><i class="fa fa-check"></i><b>2.4.2</b> Change from Basilne</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#data-utilities"><i class="fa fa-check"></i><b>2.5</b> Data utilities</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#scan"><i class="fa fa-check"></i><b>2.5.1</b> %Scan</a></li>
<li class="chapter" data-level="2.5.2" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#eval-and-syseval"><i class="fa fa-check"></i><b>2.5.2</b> %eval and %syseval</a></li>
<li class="chapter" data-level="2.5.3" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#macro-variable-status"><i class="fa fa-check"></i><b>2.5.3</b> Macro variable status</a></li>
<li class="chapter" data-level="2.5.4" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#sysfunc"><i class="fa fa-check"></i><b>2.5.4</b> %SYSFUNC</a></li>
<li class="chapter" data-level="2.5.5" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#quoting-function"><i class="fa fa-check"></i><b>2.5.5</b> Quoting Function</a></li>
<li class="chapter" data-level="2.5.6" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#call-symput"><i class="fa fa-check"></i><b>2.5.6</b> Call Symput</a></li>
<li class="chapter" data-level="2.5.7" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#call-execute"><i class="fa fa-check"></i><b>2.5.7</b> Call Execute</a></li>
<li class="chapter" data-level="2.5.8" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#sysfunc-get-the-observations"><i class="fa fa-check"></i><b>2.5.8</b> %sysfunc get the observations</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#clean-up"><i class="fa fa-check"></i><b>2.6</b> Clean Up</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#basic-setting"><i class="fa fa-check"></i><b>2.6.1</b> Basic setting</a></li>
<li class="chapter" data-level="2.6.2" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#delete-datasets"><i class="fa fa-check"></i><b>2.6.2</b> Delete datasets</a></li>
<li class="chapter" data-level="2.6.3" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#deleting-formats"><i class="fa fa-check"></i><b>2.6.3</b> Deleting Formats</a></li>
<li class="chapter" data-level="2.6.4" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#remove-assigned-formats"><i class="fa fa-check"></i><b>2.6.4</b> Remove assigned formats</a></li>
<li class="chapter" data-level="2.6.5" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#delete-macro-variables"><i class="fa fa-check"></i><b>2.6.5</b> Delete macro variables</a></li>
<li class="chapter" data-level="2.6.6" data-path="data-management-in-sas.html"><a href="data-management-in-sas.html#delete-macro"><i class="fa fa-check"></i><b>2.6.6</b> Delete Macro</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-management-in-r.html"><a href="data-management-in-r.html"><i class="fa fa-check"></i><b>3</b> Data Management in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-management-in-r.html"><a href="data-management-in-r.html#basic-data-management"><i class="fa fa-check"></i><b>3.1</b> Basic data management</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="data-management-in-r.html"><a href="data-management-in-r.html#apply-family"><i class="fa fa-check"></i><b>3.1.1</b> apply family</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="data-management-in-r.html"><a href="data-management-in-r.html#importimput"><i class="fa fa-check"></i><b>3.2</b> import/imput</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="data-management-in-r.html"><a href="data-management-in-r.html#package-readr"><i class="fa fa-check"></i><b>3.2.1</b> Package readr</a></li>
<li class="chapter" data-level="3.2.2" data-path="data-management-in-r.html"><a href="data-management-in-r.html#import-csv-data-1"><i class="fa fa-check"></i><b>3.2.2</b> Import csv data</a></li>
<li class="chapter" data-level="3.2.3" data-path="data-management-in-r.html"><a href="data-management-in-r.html#import-txt-data"><i class="fa fa-check"></i><b>3.2.3</b> Import txt data</a></li>
<li class="chapter" data-level="3.2.4" data-path="data-management-in-r.html"><a href="data-management-in-r.html#import-excel-data"><i class="fa fa-check"></i><b>3.2.4</b> Import excel data</a></li>
<li class="chapter" data-level="3.2.5" data-path="data-management-in-r.html"><a href="data-management-in-r.html#import-stata-data"><i class="fa fa-check"></i><b>3.2.5</b> Import stata data</a></li>
<li class="chapter" data-level="3.2.6" data-path="data-management-in-r.html"><a href="data-management-in-r.html#import-sas-data"><i class="fa fa-check"></i><b>3.2.6</b> Import SAS data</a></li>
<li class="chapter" data-level="3.2.7" data-path="data-management-in-r.html"><a href="data-management-in-r.html#copy-from-clipboard"><i class="fa fa-check"></i><b>3.2.7</b> Copy from clipboard</a></li>
<li class="chapter" data-level="3.2.8" data-path="data-management-in-r.html"><a href="data-management-in-r.html#save-and-write-objective-in-r"><i class="fa fa-check"></i><b>3.2.8</b> Save and write objective in R</a></li>
<li class="chapter" data-level="3.2.9" data-path="data-management-in-r.html"><a href="data-management-in-r.html#save-the-plot"><i class="fa fa-check"></i><b>3.2.9</b> Save the plot</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="data-management-in-r.html"><a href="data-management-in-r.html#package-tidyr"><i class="fa fa-check"></i><b>3.3</b> Package tidyr</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="data-management-in-r.html"><a href="data-management-in-r.html#cheat-sheet"><i class="fa fa-check"></i><b>3.3.1</b> CHEAT SHEET</a></li>
<li class="chapter" data-level="3.3.2" data-path="data-management-in-r.html"><a href="data-management-in-r.html#pivoting"><i class="fa fa-check"></i><b>3.3.2</b> Pivoting</a></li>
<li class="chapter" data-level="3.3.3" data-path="data-management-in-r.html"><a href="data-management-in-r.html#gather-and-spread"><i class="fa fa-check"></i><b>3.3.3</b> gather and spread</a></li>
<li class="chapter" data-level="3.3.4" data-path="data-management-in-r.html"><a href="data-management-in-r.html#separate-and-unite"><i class="fa fa-check"></i><b>3.3.4</b> separate and unite</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="data-management-in-r.html"><a href="data-management-in-r.html#package-dplyr"><i class="fa fa-check"></i><b>3.4</b> Package dplyr</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="data-management-in-r.html"><a href="data-management-in-r.html#cheat-sheet-1"><i class="fa fa-check"></i><b>3.4.1</b> CHEAT SHEET</a></li>
<li class="chapter" data-level="3.4.2" data-path="data-management-in-r.html"><a href="data-management-in-r.html#across"><i class="fa fa-check"></i><b>3.4.2</b> across</a></li>
<li class="chapter" data-level="3.4.3" data-path="data-management-in-r.html"><a href="data-management-in-r.html#arrange"><i class="fa fa-check"></i><b>3.4.3</b> arrange</a></li>
<li class="chapter" data-level="3.4.4" data-path="data-management-in-r.html"><a href="data-management-in-r.html#coalesce"><i class="fa fa-check"></i><b>3.4.4</b> coalesce</a></li>
<li class="chapter" data-level="3.4.5" data-path="data-management-in-r.html"><a href="data-management-in-r.html#filter"><i class="fa fa-check"></i><b>3.4.5</b> filter</a></li>
<li class="chapter" data-level="3.4.6" data-path="data-management-in-r.html"><a href="data-management-in-r.html#if_else"><i class="fa fa-check"></i><b>3.4.6</b> if_else</a></li>
<li class="chapter" data-level="3.4.7" data-path="data-management-in-r.html"><a href="data-management-in-r.html#join"><i class="fa fa-check"></i><b>3.4.7</b> join</a></li>
<li class="chapter" data-level="3.4.8" data-path="data-management-in-r.html"><a href="data-management-in-r.html#mutate"><i class="fa fa-check"></i><b>3.4.8</b> mutate</a></li>
<li class="chapter" data-level="3.4.9" data-path="data-management-in-r.html"><a href="data-management-in-r.html#select"><i class="fa fa-check"></i><b>3.4.9</b> select</a></li>
<li class="chapter" data-level="3.4.10" data-path="data-management-in-r.html"><a href="data-management-in-r.html#summarise"><i class="fa fa-check"></i><b>3.4.10</b> summarise</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="data-management-in-r.html"><a href="data-management-in-r.html#package-stringr"><i class="fa fa-check"></i><b>3.5</b> Package stringr</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="data-management-in-r.html"><a href="data-management-in-r.html#cheat-sheet-2"><i class="fa fa-check"></i><b>3.5.1</b> CHEAT SHEET</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="data-management-in-r.html"><a href="data-management-in-r.html#package-forcats"><i class="fa fa-check"></i><b>3.6</b> Package forcats</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="data-management-in-r.html"><a href="data-management-in-r.html#cheat-sheet-3"><i class="fa fa-check"></i><b>3.6.1</b> CHEAT SHEET</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="data-management-in-r.html"><a href="data-management-in-r.html#package-lubridate"><i class="fa fa-check"></i><b>3.7</b> Package lubridate</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="data-management-in-r.html"><a href="data-management-in-r.html#cheat-sheet-4"><i class="fa fa-check"></i><b>3.7.1</b> CHEAT SHEET</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="ggplot2.html"><a href="ggplot2.html"><i class="fa fa-check"></i><b>4</b> ggplot2</a>
<ul>
<li class="chapter" data-level="4.1" data-path="ggplot2.html"><a href="ggplot2.html#parametric-rendering"><i class="fa fa-check"></i><b>4.1</b> Parametric Rendering</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="ggplot2.html"><a href="ggplot2.html#grammar"><i class="fa fa-check"></i><b>4.1.1</b> Grammar</a></li>
<li class="chapter" data-level="4.1.2" data-path="ggplot2.html"><a href="ggplot2.html#coordinate-system"><i class="fa fa-check"></i><b>4.1.2</b> Coordinate system</a></li>
<li class="chapter" data-level="4.1.3" data-path="ggplot2.html"><a href="ggplot2.html#legend"><i class="fa fa-check"></i><b>4.1.3</b> Legend</a></li>
<li class="chapter" data-level="4.1.4" data-path="ggplot2.html"><a href="ggplot2.html#geom"><i class="fa fa-check"></i><b>4.1.4</b> geom</a></li>
<li class="chapter" data-level="4.1.5" data-path="ggplot2.html"><a href="ggplot2.html#position"><i class="fa fa-check"></i><b>4.1.5</b> position</a></li>
<li class="chapter" data-level="4.1.6" data-path="ggplot2.html"><a href="ggplot2.html#scale"><i class="fa fa-check"></i><b>4.1.6</b> scale</a></li>
<li class="chapter" data-level="4.1.7" data-path="ggplot2.html"><a href="ggplot2.html#stat"><i class="fa fa-check"></i><b>4.1.7</b> stat</a></li>
<li class="chapter" data-level="4.1.8" data-path="ggplot2.html"><a href="ggplot2.html#color"><i class="fa fa-check"></i><b>4.1.8</b> Color</a></li>
<li class="chapter" data-level="4.1.9" data-path="ggplot2.html"><a href="ggplot2.html#thema"><i class="fa fa-check"></i><b>4.1.9</b> Thema</a></li>
<li class="chapter" data-level="4.1.10" data-path="ggplot2.html"><a href="ggplot2.html#saving-plots"><i class="fa fa-check"></i><b>4.1.10</b> Saving plots</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="ggplot2.html"><a href="ggplot2.html#scatter-plot"><i class="fa fa-check"></i><b>4.2</b> Scatter plot</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="ggplot2.html"><a href="ggplot2.html#grouping-aesthetic"><i class="fa fa-check"></i><b>4.2.1</b> Grouping Aesthetic</a></li>
<li class="chapter" data-level="4.2.2" data-path="ggplot2.html"><a href="ggplot2.html#facet"><i class="fa fa-check"></i><b>4.2.2</b> facet</a></li>
<li class="chapter" data-level="4.2.3" data-path="ggplot2.html"><a href="ggplot2.html#geom_smooth"><i class="fa fa-check"></i><b>4.2.3</b> geom_smooth</a></li>
<li class="chapter" data-level="4.2.4" data-path="ggplot2.html"><a href="ggplot2.html#dot-plot"><i class="fa fa-check"></i><b>4.2.4</b> Dot Plot</a></li>
<li class="chapter" data-level="4.2.5" data-path="ggplot2.html"><a href="ggplot2.html#label-and-title"><i class="fa fa-check"></i><b>4.2.5</b> Label and Title</a></li>
<li class="chapter" data-level="4.2.6" data-path="ggplot2.html"><a href="ggplot2.html#residuals"><i class="fa fa-check"></i><b>4.2.6</b> Residuals</a></li>
<li class="chapter" data-level="4.2.7" data-path="ggplot2.html"><a href="ggplot2.html#encircling"><i class="fa fa-check"></i><b>4.2.7</b> Encircling</a></li>
<li class="chapter" data-level="4.2.8" data-path="ggplot2.html"><a href="ggplot2.html#dumbbell-plot"><i class="fa fa-check"></i><b>4.2.8</b> Dumbbell Plot</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="ggplot2.html"><a href="ggplot2.html#histogram"><i class="fa fa-check"></i><b>4.3</b> Histogram</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="ggplot2.html"><a href="ggplot2.html#general-appearance"><i class="fa fa-check"></i><b>4.3.1</b> General appearance</a></li>
<li class="chapter" data-level="4.3.2" data-path="ggplot2.html"><a href="ggplot2.html#themes"><i class="fa fa-check"></i><b>4.3.2</b> Themes</a></li>
<li class="chapter" data-level="4.3.3" data-path="ggplot2.html"><a href="ggplot2.html#geom_freqpoly"><i class="fa fa-check"></i><b>4.3.3</b> geom_freqpoly()</a></li>
<li class="chapter" data-level="4.3.4" data-path="ggplot2.html"><a href="ggplot2.html#marginal-histogram-boxplot"><i class="fa fa-check"></i><b>4.3.4</b> Marginal Histogram / Boxplot</a></li>
<li class="chapter" data-level="4.3.5" data-path="ggplot2.html"><a href="ggplot2.html#scales"><i class="fa fa-check"></i><b>4.3.5</b> Scales</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="ggplot2.html"><a href="ggplot2.html#bar-plot"><i class="fa fa-check"></i><b>4.4</b> Bar plot</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="ggplot2.html"><a href="ggplot2.html#aesthetic"><i class="fa fa-check"></i><b>4.4.1</b> Aesthetic</a></li>
<li class="chapter" data-level="4.4.2" data-path="ggplot2.html"><a href="ggplot2.html#proportion"><i class="fa fa-check"></i><b>4.4.2</b> Proportion</a></li>
<li class="chapter" data-level="4.4.3" data-path="ggplot2.html"><a href="ggplot2.html#coord"><i class="fa fa-check"></i><b>4.4.3</b> coord</a></li>
<li class="chapter" data-level="4.4.4" data-path="ggplot2.html"><a href="ggplot2.html#statidentity"><i class="fa fa-check"></i><b>4.4.4</b> stat=“identity”</a></li>
<li class="chapter" data-level="4.4.5" data-path="ggplot2.html"><a href="ggplot2.html#ranking"><i class="fa fa-check"></i><b>4.4.5</b> Ranking</a></li>
<li class="chapter" data-level="4.4.6" data-path="ggplot2.html"><a href="ggplot2.html#means-and-error-bars"><i class="fa fa-check"></i><b>4.4.6</b> Means and error bars</a></li>
<li class="chapter" data-level="4.4.7" data-path="ggplot2.html"><a href="ggplot2.html#waffle-chart"><i class="fa fa-check"></i><b>4.4.7</b> Waffle Chart</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="ggplot2.html"><a href="ggplot2.html#box-plot"><i class="fa fa-check"></i><b>4.5</b> Box plot</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="ggplot2.html"><a href="ggplot2.html#varwidth"><i class="fa fa-check"></i><b>4.5.1</b> varwidth</a></li>
<li class="chapter" data-level="4.5.2" data-path="ggplot2.html"><a href="ggplot2.html#reorder"><i class="fa fa-check"></i><b>4.5.2</b> Reorder</a></li>
<li class="chapter" data-level="4.5.3" data-path="ggplot2.html"><a href="ggplot2.html#fill"><i class="fa fa-check"></i><b>4.5.3</b> Fill</a></li>
<li class="chapter" data-level="4.5.4" data-path="ggplot2.html"><a href="ggplot2.html#statidentity-1"><i class="fa fa-check"></i><b>4.5.4</b> stat=“identity”</a></li>
<li class="chapter" data-level="4.5.5" data-path="ggplot2.html"><a href="ggplot2.html#dot-box-plot"><i class="fa fa-check"></i><b>4.5.5</b> Dot + Box Plot</a></li>
<li class="chapter" data-level="4.5.6" data-path="ggplot2.html"><a href="ggplot2.html#violin-plot"><i class="fa fa-check"></i><b>4.5.6</b> Violin Plot</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="ggplot2.html"><a href="ggplot2.html#other-rendering"><i class="fa fa-check"></i><b>4.6</b> Other rendering</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="ggplot2.html"><a href="ggplot2.html#annotation"><i class="fa fa-check"></i><b>4.6.1</b> Annotation</a></li>
<li class="chapter" data-level="4.6.2" data-path="ggplot2.html"><a href="ggplot2.html#text-ggtext"><i class="fa fa-check"></i><b>4.6.2</b> Text: ggtext</a></li>
<li class="chapter" data-level="4.6.3" data-path="ggplot2.html"><a href="ggplot2.html#text-ggrepel"><i class="fa fa-check"></i><b>4.6.3</b> Text: ggrepel</a></li>
<li class="chapter" data-level="4.6.4" data-path="ggplot2.html"><a href="ggplot2.html#arrage-gridextra"><i class="fa fa-check"></i><b>4.6.4</b> Arrage: gridExtra</a></li>
<li class="chapter" data-level="4.6.5" data-path="ggplot2.html"><a href="ggplot2.html#interactive-charts-plotly"><i class="fa fa-check"></i><b>4.6.5</b> Interactive charts: plotly</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="ggplot2.html"><a href="ggplot2.html#diagrammer"><i class="fa fa-check"></i><b>4.7</b> DiagrammeR</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="ggplot2.html"><a href="ggplot2.html#basic-flowchart-using-grviz"><i class="fa fa-check"></i><b>4.7.1</b> Basic flowchart using <code>grViz</code></a></li>
<li class="chapter" data-level="4.7.2" data-path="ggplot2.html"><a href="ggplot2.html#graphviz-attributes"><i class="fa fa-check"></i><b>4.7.2</b> Graphviz Attributes</a></li>
<li class="chapter" data-level="4.7.3" data-path="ggplot2.html"><a href="ggplot2.html#gantt-chart"><i class="fa fa-check"></i><b>4.7.3</b> Gantt chart</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="ggplot2.html"><a href="ggplot2.html#ggstatsplot"><i class="fa fa-check"></i><b>4.8</b> ggstatsplot</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="ggplot2.html"><a href="ggplot2.html#ggbetweenstats"><i class="fa fa-check"></i><b>4.8.1</b> ggbetweenstats</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html"><i class="fa fa-check"></i><b>5</b> Descriptive Statistics in R</a>
<ul>
<li class="chapter" data-level="5.1" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html#package-pape"><i class="fa fa-check"></i><b>5.1</b> Package pape</a></li>
<li class="chapter" data-level="5.2" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html#package-summarytools"><i class="fa fa-check"></i><b>5.2</b> Package summarytools</a></li>
<li class="chapter" data-level="5.3" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html#package-comparegroups"><i class="fa fa-check"></i><b>5.3</b> Package compareGroups</a></li>
<li class="chapter" data-level="5.4" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html#package-sjplot-for-models-summary"><i class="fa fa-check"></i><b>5.4</b> Package sjPlot for Models Summary</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html#tab_model-and-options"><i class="fa fa-check"></i><b>5.4.1</b> tab_model and options</a></li>
<li class="chapter" data-level="5.4.2" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html#mixed-models"><i class="fa fa-check"></i><b>5.4.2</b> Mixed Models</a></li>
<li class="chapter" data-level="5.4.3" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html#plot_model"><i class="fa fa-check"></i><b>5.4.3</b> plot_model</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html#package-kableextra"><i class="fa fa-check"></i><b>5.5</b> Package kableExtra</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html#kable_styling"><i class="fa fa-check"></i><b>5.5.1</b> kable_styling</a></li>
<li class="chapter" data-level="5.5.2" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html#columnrow-specification"><i class="fa fa-check"></i><b>5.5.2</b> Column/Row Specification</a></li>
<li class="chapter" data-level="5.5.3" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html#grouping-columnsrows"><i class="fa fa-check"></i><b>5.5.3</b> Grouping columns/rows</a></li>
<li class="chapter" data-level="5.5.4" data-path="descriptive-statistics-in-r.html"><a href="descriptive-statistics-in-r.html#add-footnote"><i class="fa fa-check"></i><b>5.5.4</b> Add Footnote</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="parametric-test.html"><a href="parametric-test.html"><i class="fa fa-check"></i><b>6</b> Parametric Test</a>
<ul>
<li class="chapter" data-level="6.1" data-path="parametric-test.html"><a href="parametric-test.html#binomial-test"><i class="fa fa-check"></i><b>6.1</b> Binomial test</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="parametric-test.html"><a href="parametric-test.html#mathematical-formula"><i class="fa fa-check"></i><b>6.1.1</b> Mathematical Formula</a></li>
<li class="chapter" data-level="6.1.2" data-path="parametric-test.html"><a href="parametric-test.html#r-implementation"><i class="fa fa-check"></i><b>6.1.2</b> R implementation</a></li>
<li class="chapter" data-level="6.1.3" data-path="parametric-test.html"><a href="parametric-test.html#sas-implementation"><i class="fa fa-check"></i><b>6.1.3</b> SAS Implementation</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="parametric-test.html"><a href="parametric-test.html#fishers-exact-test"><i class="fa fa-check"></i><b>6.2</b> Fisher’s Exact Test</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="parametric-test.html"><a href="parametric-test.html#introduction-1"><i class="fa fa-check"></i><b>6.2.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2.2" data-path="parametric-test.html"><a href="parametric-test.html#sas-implementation-1"><i class="fa fa-check"></i><b>6.2.2</b> SAS implementation</a></li>
<li class="chapter" data-level="6.2.3" data-path="parametric-test.html"><a href="parametric-test.html#r-implementation-1"><i class="fa fa-check"></i><b>6.2.3</b> R implementation</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="parametric-test.html"><a href="parametric-test.html#sensitivity-and-specificity"><i class="fa fa-check"></i><b>6.3</b> Sensitivity and specificity</a></li>
<li class="chapter" data-level="6.4" data-path="parametric-test.html"><a href="parametric-test.html#mcnemars-test"><i class="fa fa-check"></i><b>6.4</b> McNemar’s test</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="parametric-test.html"><a href="parametric-test.html#introduction-2"><i class="fa fa-check"></i><b>6.4.1</b> introduction</a></li>
<li class="chapter" data-level="6.4.2" data-path="parametric-test.html"><a href="parametric-test.html#sas-implementation-2"><i class="fa fa-check"></i><b>6.4.2</b> SAS Implementation</a></li>
<li class="chapter" data-level="6.4.3" data-path="parametric-test.html"><a href="parametric-test.html#r-implementation-2"><i class="fa fa-check"></i><b>6.4.3</b> R Implementation</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="parametric-test.html"><a href="parametric-test.html#cochranmantelhaenszel-test"><i class="fa fa-check"></i><b>6.5</b> Cochran–Mantel–Haenszel Test</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="parametric-test.html"><a href="parametric-test.html#r-implementation-3"><i class="fa fa-check"></i><b>6.5.1</b> R implementation</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="parametric-test.html"><a href="parametric-test.html#correlation-test"><i class="fa fa-check"></i><b>6.6</b> Correlation Test</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="parametric-test.html"><a href="parametric-test.html#pearson-correlation"><i class="fa fa-check"></i><b>6.6.1</b> Pearson correlation</a></li>
<li class="chapter" data-level="6.6.2" data-path="parametric-test.html"><a href="parametric-test.html#spearman-correlation"><i class="fa fa-check"></i><b>6.6.2</b> Spearman correlation</a></li>
<li class="chapter" data-level="6.6.3" data-path="parametric-test.html"><a href="parametric-test.html#kendall-correlation"><i class="fa fa-check"></i><b>6.6.3</b> Kendall correlation</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="parametric-test.html"><a href="parametric-test.html#two-sample-t-test"><i class="fa fa-check"></i><b>6.7</b> Two Sample T-Test</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="parametric-test.html"><a href="parametric-test.html#inreoduction"><i class="fa fa-check"></i><b>6.7.1</b> Inreoduction</a></li>
<li class="chapter" data-level="6.7.2" data-path="parametric-test.html"><a href="parametric-test.html#sas-implementation-3"><i class="fa fa-check"></i><b>6.7.2</b> SAS implementation</a></li>
<li class="chapter" data-level="6.7.3" data-path="parametric-test.html"><a href="parametric-test.html#r-implementation-4"><i class="fa fa-check"></i><b>6.7.3</b> R implementation</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="parametric-test.html"><a href="parametric-test.html#normality-test"><i class="fa fa-check"></i><b>6.8</b> Normality test</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="parametric-test.html"><a href="parametric-test.html#sas-implementation-4"><i class="fa fa-check"></i><b>6.8.1</b> SAS implementation</a></li>
<li class="chapter" data-level="6.8.2" data-path="parametric-test.html"><a href="parametric-test.html#r-implementation-5"><i class="fa fa-check"></i><b>6.8.2</b> R implementation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="non-parametric-test.html"><a href="non-parametric-test.html"><i class="fa fa-check"></i><b>7</b> Non-Parametric Test</a>
<ul>
<li class="chapter" data-level="7.1" data-path="non-parametric-test.html"><a href="non-parametric-test.html#two-samples-hypotheses-testing"><i class="fa fa-check"></i><b>7.1</b> Two Samples Hypotheses Testing</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="non-parametric-test.html"><a href="non-parametric-test.html#sign-test-for-location-parameter-for-matched-paired-samples"><i class="fa fa-check"></i><b>7.1.1</b> Sign Test for Location Parameter for Matched Paired Samples</a></li>
<li class="chapter" data-level="7.1.2" data-path="non-parametric-test.html"><a href="non-parametric-test.html#wilcoxon-signed-rank-test-for-location-parameter-for-matched"><i class="fa fa-check"></i><b>7.1.2</b> Wilcoxon Signed-Rank Test for Location Parameter for Matched</a></li>
<li class="chapter" data-level="7.1.3" data-path="non-parametric-test.html"><a href="non-parametric-test.html#wilcoxon-rank-sum-test-for-location-parameter-for-two-independent-samples"><i class="fa fa-check"></i><b>7.1.3</b> Wilcoxon Rank-Sum Test for Location Parameter for Two Independent Samples</a></li>
<li class="chapter" data-level="7.1.4" data-path="non-parametric-test.html"><a href="non-parametric-test.html#ansari-bradley-test-for-scale-parameter-for-two-independent-samples"><i class="fa fa-check"></i><b>7.1.4</b> Ansari-Bradley Test for Scale Parameter for Two Independent Samples</a></li>
<li class="chapter" data-level="7.1.5" data-path="non-parametric-test.html"><a href="non-parametric-test.html#kolmogorov-smirnov-test-for-equality-of-distributions"><i class="fa fa-check"></i><b>7.1.5</b> Kolmogorov-Smirnov Test for Equality of Distributions</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="non-parametric-test.html"><a href="non-parametric-test.html#several-samples-hypotheses-testing"><i class="fa fa-check"></i><b>7.2</b> Several Samples Hypotheses Testing</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="non-parametric-test.html"><a href="non-parametric-test.html#friedman-rank-test-for-location-parameter-for-several-dependent-samples"><i class="fa fa-check"></i><b>7.2.1</b> Friedman Rank Test for Location Parameter for Several Dependent Samples</a></li>
<li class="chapter" data-level="7.2.2" data-path="non-parametric-test.html"><a href="non-parametric-test.html#kruskal-wallis-h-test-for-location-parameter"><i class="fa fa-check"></i><b>7.2.2</b> Kruskal-Wallis H-Test for Location Parameter</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="non-parametric-test.html"><a href="non-parametric-test.html#tests-for-categorical-data"><i class="fa fa-check"></i><b>7.3</b> Tests for Categorical Data</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="non-parametric-test.html"><a href="non-parametric-test.html#spearman-rank-correlation-coefficient-test"><i class="fa fa-check"></i><b>7.3.1</b> Spearman Rank Correlation Coefficient Test</a></li>
<li class="chapter" data-level="7.3.2" data-path="non-parametric-test.html"><a href="non-parametric-test.html#fisher-exact-test"><i class="fa fa-check"></i><b>7.3.2</b> Fisher Exact Test</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="non-parametric-test.html"><a href="non-parametric-test.html#permutation-test"><i class="fa fa-check"></i><b>7.4</b> Permutation test</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="non-parametric-test.html"><a href="non-parametric-test.html#introduction-3"><i class="fa fa-check"></i><b>7.4.1</b> Introduction</a></li>
<li class="chapter" data-level="7.4.2" data-path="non-parametric-test.html"><a href="non-parametric-test.html#package-coin"><i class="fa fa-check"></i><b>7.4.2</b> Package: coin</a></li>
<li class="chapter" data-level="7.4.3" data-path="non-parametric-test.html"><a href="non-parametric-test.html#one-way-permutation-test-of-independence-for-ordinal-data"><i class="fa fa-check"></i><b>7.4.3</b> One-way Permutation Test of Independence for Ordinal Data</a></li>
<li class="chapter" data-level="7.4.4" data-path="non-parametric-test.html"><a href="non-parametric-test.html#one-way-permutation-test-of-symmetry-for-ordinal-data"><i class="fa fa-check"></i><b>7.4.4</b> One-way Permutation Test of Symmetry for Ordinal Data</a></li>
<li class="chapter" data-level="7.4.5" data-path="non-parametric-test.html"><a href="non-parametric-test.html#permutation-tests-for-medians-and-percentiles"><i class="fa fa-check"></i><b>7.4.5</b> Permutation Tests for Medians and Percentiles</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html"><i class="fa fa-check"></i><b>8</b> Sample Size Calculation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#distribution"><i class="fa fa-check"></i><b>8.1</b> Distribution</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#quantile-function-in-sas"><i class="fa fa-check"></i><b>8.1.1</b> Quantile Function in SAS</a></li>
<li class="chapter" data-level="8.1.2" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#binomial-distribution"><i class="fa fa-check"></i><b>8.1.2</b> Binomial distribution</a></li>
<li class="chapter" data-level="8.1.3" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#negative-binomial-distribution"><i class="fa fa-check"></i><b>8.1.3</b> Negative binomial distribution</a></li>
<li class="chapter" data-level="8.1.4" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#multinomial-distribution"><i class="fa fa-check"></i><b>8.1.4</b> Multinomial distribution</a></li>
<li class="chapter" data-level="8.1.5" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#normal-distribution"><i class="fa fa-check"></i><b>8.1.5</b> Normal Distribution</a></li>
<li class="chapter" data-level="8.1.6" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#multivariate-normal-distribution"><i class="fa fa-check"></i><b>8.1.6</b> Multivariate normal distribution</a></li>
<li class="chapter" data-level="8.1.7" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#poisson-distribution"><i class="fa fa-check"></i><b>8.1.7</b> Poisson distribution</a></li>
<li class="chapter" data-level="8.1.8" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#exponential-distribution"><i class="fa fa-check"></i><b>8.1.8</b> Exponential distribution</a></li>
<li class="chapter" data-level="8.1.9" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#gamma-distribution"><i class="fa fa-check"></i><b>8.1.9</b> Gamma distribution</a></li>
<li class="chapter" data-level="8.1.10" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#weibull-distribution"><i class="fa fa-check"></i><b>8.1.10</b> Weibull Distribution</a></li>
<li class="chapter" data-level="8.1.11" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#chi-squared-distribution"><i class="fa fa-check"></i><b>8.1.11</b> Chi-Squared Distribution</a></li>
<li class="chapter" data-level="8.1.12" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#beta-distribution"><i class="fa fa-check"></i><b>8.1.12</b> Beta Distribution</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#binomial-ci"><i class="fa fa-check"></i><b>8.2</b> Binomial CI</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#point-estimates"><i class="fa fa-check"></i><b>8.2.1</b> Point Estimates</a></li>
<li class="chapter" data-level="8.2.2" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#binomial-ci-for-small-samples"><i class="fa fa-check"></i><b>8.2.2</b> Binomial CI for Small Samples</a></li>
<li class="chapter" data-level="8.2.3" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#r-packages"><i class="fa fa-check"></i><b>8.2.3</b> R Packages</a></li>
<li class="chapter" data-level="8.2.4" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#incidence-rate-ci"><i class="fa fa-check"></i><b>8.2.4</b> Incidence rate CI</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#test-1-sample-proportion"><i class="fa fa-check"></i><b>8.3</b> Test 1-Sample Proportion</a></li>
<li class="chapter" data-level="8.4" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#test-2-sample-proportions"><i class="fa fa-check"></i><b>8.4</b> Test 2-Sample Proportions</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#technical-details"><i class="fa fa-check"></i><b>8.4.1</b> Technical details</a></li>
<li class="chapter" data-level="8.4.2" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#fishers-exact-test-1"><i class="fa fa-check"></i><b>8.4.2</b> Fisher’s Exact Test</a></li>
<li class="chapter" data-level="8.4.3" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#z-test-or-chi-square-test-pooled-and-unpooled"><i class="fa fa-check"></i><b>8.4.3</b> Z Test (or Chi-Square Test) (Pooled and Unpooled)</a></li>
<li class="chapter" data-level="8.4.4" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#conditional-mantel-haenszel-test"><i class="fa fa-check"></i><b>8.4.4</b> Conditional Mantel-Haenszel Test</a></li>
<li class="chapter" data-level="8.4.5" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#paired-proportions-mcnemars-z-test"><i class="fa fa-check"></i><b>8.4.5</b> Paired Proportions: McNemar’s Z-test</a></li>
<li class="chapter" data-level="8.4.6" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#chi-square-test"><i class="fa fa-check"></i><b>8.4.6</b> Chi-square test</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#test-means"><i class="fa fa-check"></i><b>8.5</b> Test Mean(s)</a>
<ul>
<li class="chapter" data-level="8.5.1" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#one-sample-mean-2-sided-equality"><i class="fa fa-check"></i><b>8.5.1</b> One Sample Mean 2-Sided Equality</a></li>
<li class="chapter" data-level="8.5.2" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#sample-mean-non-inferiority-or-superiority"><i class="fa fa-check"></i><b>8.5.2</b> 1-Sample Mean Non-Inferiority or Superiority</a></li>
<li class="chapter" data-level="8.5.3" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#sample-mean-equivalence"><i class="fa fa-check"></i><b>8.5.3</b> 1-Sample Mean Equivalence</a></li>
<li class="chapter" data-level="8.5.4" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#sample-means-2-sided-equality"><i class="fa fa-check"></i><b>8.5.4</b> 2-Sample Means, 2-Sided Equality</a></li>
<li class="chapter" data-level="8.5.5" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#sample-means-1-sided"><i class="fa fa-check"></i><b>8.5.5</b> 2-Sample Means, 1-Sided</a></li>
<li class="chapter" data-level="8.5.6" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#sample-means-non-inferiority-or-superiority"><i class="fa fa-check"></i><b>8.5.6</b> 2-Sample Means Non-Inferiority or Superiority</a></li>
<li class="chapter" data-level="8.5.7" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#sample-means-equivalence"><i class="fa fa-check"></i><b>8.5.7</b> 2-Sample Means Equivalence</a></li>
<li class="chapter" data-level="8.5.8" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#t-test-using-pwr.t.test-or-pwr.t2n.test"><i class="fa fa-check"></i><b>8.5.8</b> T-Test using <code>pwr.t.test</code> or <code>pwr.t2n.test</code></a></li>
<li class="chapter" data-level="8.5.9" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#test-k-means"><i class="fa fa-check"></i><b>8.5.9</b> Test k Means</a></li>
<li class="chapter" data-level="8.5.10" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#anova-using-pwr.anova.test"><i class="fa fa-check"></i><b>8.5.10</b> Anova using <code>pwr.anova.test</code></a></li>
<li class="chapter" data-level="8.5.11" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#average-bioequivalence"><i class="fa fa-check"></i><b>8.5.11</b> Average Bioequivalence</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#other-methods"><i class="fa fa-check"></i><b>8.6</b> Other Methods</a>
<ul>
<li class="chapter" data-level="8.6.1" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#odds-ratio"><i class="fa fa-check"></i><b>8.6.1</b> Odds Ratio</a></li>
<li class="chapter" data-level="8.6.2" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#correlation"><i class="fa fa-check"></i><b>8.6.2</b> Correlation</a></li>
<li class="chapter" data-level="8.6.3" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#correlated-binary-data-ophthalmologic-studies"><i class="fa fa-check"></i><b>8.6.3</b> Correlated Binary Data: Ophthalmologic Studies</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#multiple-test"><i class="fa fa-check"></i><b>8.7</b> Multiple test</a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#definitions-of-power"><i class="fa fa-check"></i><b>8.7.1</b> Definitions of Power</a></li>
<li class="chapter" data-level="8.7.2" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#bonferroni-tests-individual-power"><i class="fa fa-check"></i><b>8.7.2</b> Bonferroni Tests (Individual Power)</a></li>
<li class="chapter" data-level="8.7.3" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#tukeys-method-individual-power"><i class="fa fa-check"></i><b>8.7.3</b> Tukey’s method (Individual Power)</a></li>
<li class="chapter" data-level="8.7.4" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#dunnetts-two-sided-tests-individual-power"><i class="fa fa-check"></i><b>8.7.4</b> Dunnett’s Two-Sided Tests (Individual Power)</a></li>
<li class="chapter" data-level="8.7.5" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#combined-power-simpower-and-plotsimpower"><i class="fa fa-check"></i><b>8.7.5</b> Combined Power %SimPower and %PlotSimPower</a></li>
<li class="chapter" data-level="8.7.6" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#dunnett-set-up-close-tests-in-r"><i class="fa fa-check"></i><b>8.7.6</b> Dunnett Set up Close Tests in R</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#time-to-event-data"><i class="fa fa-check"></i><b>8.8</b> Time-To-Event Data</a>
<ul>
<li class="chapter" data-level="8.8.1" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#cox-ph-2-sided-equality"><i class="fa fa-check"></i><b>8.8.1</b> Cox PH, 2-Sided Equality</a></li>
<li class="chapter" data-level="8.8.2" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#log-rank-tests-for-competing-risks"><i class="fa fa-check"></i><b>8.8.2</b> Log-Rank Tests for Competing Risks</a></li>
</ul></li>
<li class="chapter" data-level="8.9" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#estimation-in-diagnostic-test"><i class="fa fa-check"></i><b>8.9</b> Estimation in diagnostic test</a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#adequate-sensitivityspecificity"><i class="fa fa-check"></i><b>8.9.1</b> Adequate sensitivity/specificity</a></li>
<li class="chapter" data-level="8.9.2" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#testing-sensitivity-or-specificity"><i class="fa fa-check"></i><b>8.9.2</b> Testing sensitivity (or specificity)</a></li>
<li class="chapter" data-level="8.9.3" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#likelihood-ratio-estimation"><i class="fa fa-check"></i><b>8.9.3</b> Likelihood ratio estimation</a></li>
<li class="chapter" data-level="8.9.4" data-path="sample-size-calculation.html"><a href="sample-size-calculation.html#roc-index-of-accuracy"><i class="fa fa-check"></i><b>8.9.4</b> ROC index of accuracy</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="anova.html"><a href="anova.html"><i class="fa fa-check"></i><b>9</b> ANOVA</a>
<ul>
<li class="chapter" data-level="9.1" data-path="anova.html"><a href="anova.html#unstructured-models"><i class="fa fa-check"></i><b>9.1</b> Unstructured Models</a></li>
<li class="chapter" data-level="9.2" data-path="anova.html"><a href="anova.html#balanced-one-way-analysis-of-variance-anova"><i class="fa fa-check"></i><b>9.2</b> Balanced One-Way Analysis-of-Variance (ANOVA)</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="anova.html"><a href="anova.html#modeling-assumptions-and-basic-analysis"><i class="fa fa-check"></i><b>9.2.1</b> Modeling Assumptions and Basic Analysis</a></li>
<li class="chapter" data-level="9.2.2" data-path="anova.html"><a href="anova.html#parameter-estimates"><i class="fa fa-check"></i><b>9.2.2</b> Parameter Estimates</a></li>
<li class="chapter" data-level="9.2.3" data-path="anova.html"><a href="anova.html#r-implementation-6"><i class="fa fa-check"></i><b>9.2.3</b> R Implementation</a></li>
<li class="chapter" data-level="9.2.4" data-path="anova.html"><a href="anova.html#sas-implementation-5"><i class="fa fa-check"></i><b>9.2.4</b> SAS Implementation</a></li>
<li class="chapter" data-level="9.2.5" data-path="anova.html"><a href="anova.html#model-diagnosis"><i class="fa fa-check"></i><b>9.2.5</b> Model Diagnosis</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="anova.html"><a href="anova.html#unbalanced-one-way-anova-and-analysis-of-covariance-ancova"><i class="fa fa-check"></i><b>9.3</b> Unbalanced One-Way ANOVA and Analysis-of-Covariance (ANCOVA)</a></li>
<li class="chapter" data-level="9.4" data-path="anova.html"><a href="anova.html#two-ways-anova-test"><i class="fa fa-check"></i><b>9.4</b> Two-Ways ANOVA Test</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="anova.html"><a href="anova.html#introduction-4"><i class="fa fa-check"></i><b>9.4.1</b> Introduction</a></li>
<li class="chapter" data-level="9.4.2" data-path="anova.html"><a href="anova.html#r-implementation-7"><i class="fa fa-check"></i><b>9.4.2</b> R implementation</a></li>
<li class="chapter" data-level="9.4.3" data-path="anova.html"><a href="anova.html#sas-implementation-6"><i class="fa fa-check"></i><b>9.4.3</b> SAS Implementation</a></li>
<li class="chapter" data-level="9.4.4" data-path="anova.html"><a href="anova.html#unbalanced-design"><i class="fa fa-check"></i><b>9.4.4</b> Unbalanced design</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="anova.html"><a href="anova.html#heteroscedastic-responses"><i class="fa fa-check"></i><b>9.5</b> Heteroscedastic Responses</a></li>
<li class="chapter" data-level="9.6" data-path="anova.html"><a href="anova.html#repeated-measures-anova-data"><i class="fa fa-check"></i><b>9.6</b> Repeated Measures ANOVA Data</a></li>
<li class="chapter" data-level="9.7" data-path="anova.html"><a href="anova.html#multivariate-responses-with-normally-distributed-data"><i class="fa fa-check"></i><b>9.7</b> Multivariate Responses with Normally Distributed Data</a></li>
<li class="chapter" data-level="9.8" data-path="anova.html"><a href="anova.html#independent-observations-from-parametric-nonnormal-distributions"><i class="fa fa-check"></i><b>9.8</b> Independent Observations from Parametric Nonnormal Distributions</a></li>
<li class="chapter" data-level="9.9" data-path="anova.html"><a href="anova.html#dependent-observations-from-parametric-nonnormal-distributions"><i class="fa fa-check"></i><b>9.9</b> Dependent Observations from Parametric Nonnormal Distributions</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="multiple-comparison.html"><a href="multiple-comparison.html"><i class="fa fa-check"></i><b>10</b> Multiple-Comparison</a>
<ul>
<li class="chapter" data-level="10.1" data-path="multiple-comparison.html"><a href="multiple-comparison.html#introduction-5"><i class="fa fa-check"></i><b>10.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="multiple-comparison.html"><a href="multiple-comparison.html#multiplicity-problem"><i class="fa fa-check"></i><b>10.1.1</b> Multiplicity Problem</a></li>
<li class="chapter" data-level="10.1.2" data-path="multiple-comparison.html"><a href="multiple-comparison.html#error-rates"><i class="fa fa-check"></i><b>10.1.2</b> Error Rates</a></li>
<li class="chapter" data-level="10.1.3" data-path="multiple-comparison.html"><a href="multiple-comparison.html#the-adjusted-p"><i class="fa fa-check"></i><b>10.1.3</b> The adjusted P</a></li>
<li class="chapter" data-level="10.1.4" data-path="multiple-comparison.html"><a href="multiple-comparison.html#basic-statistical-concepts"><i class="fa fa-check"></i><b>10.1.4</b> Basic Statistical Concepts</a></li>
<li class="chapter" data-level="10.1.5" data-path="multiple-comparison.html"><a href="multiple-comparison.html#functions-in-glht-package-in-r"><i class="fa fa-check"></i><b>10.1.5</b> Functions in glht package in R</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="multiple-comparison.html"><a href="multiple-comparison.html#bonferroni-and-šidák-methods"><i class="fa fa-check"></i><b>10.2</b> Bonferroni and Šidák Methods</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="multiple-comparison.html"><a href="multiple-comparison.html#lsd-least-significance-difference"><i class="fa fa-check"></i><b>10.2.1</b> LSD (least significance difference)</a></li>
<li class="chapter" data-level="10.2.2" data-path="multiple-comparison.html"><a href="multiple-comparison.html#šidák"><i class="fa fa-check"></i><b>10.2.2</b> Šidák</a></li>
<li class="chapter" data-level="10.2.3" data-path="multiple-comparison.html"><a href="multiple-comparison.html#bonferroni"><i class="fa fa-check"></i><b>10.2.3</b> Bonferroni</a></li>
<li class="chapter" data-level="10.2.4" data-path="multiple-comparison.html"><a href="multiple-comparison.html#schweder-spjøtvoll-p-value-plot"><i class="fa fa-check"></i><b>10.2.4</b> Schweder-Spjøtvoll p-Value Plot</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="multiple-comparison.html"><a href="multiple-comparison.html#mcp-among-treatment-means-in-the-one-way-balanced-anova"><i class="fa fa-check"></i><b>10.3</b> MCP among Treatment Means in the One-Way Balanced ANOVA</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="multiple-comparison.html"><a href="multiple-comparison.html#ls-means"><i class="fa fa-check"></i><b>10.3.1</b> LS-Means</a></li>
<li class="chapter" data-level="10.3.2" data-path="multiple-comparison.html"><a href="multiple-comparison.html#the-multivariate-t-distribution"><i class="fa fa-check"></i><b>10.3.2</b> The Multivariate t Distribution</a></li>
<li class="chapter" data-level="10.3.3" data-path="multiple-comparison.html"><a href="multiple-comparison.html#calculating-the-critical-value-c_alpha"><i class="fa fa-check"></i><b>10.3.3</b> Calculating the Critical Value <span class="math inline">\(c_{\alpha}\)</span></a></li>
<li class="chapter" data-level="10.3.4" data-path="multiple-comparison.html"><a href="multiple-comparison.html#all-pairwise-comparisons-and-studentized-range-distribution"><i class="fa fa-check"></i><b>10.3.4</b> All Pairwise Comparisons and Studentized Range Distribution</a></li>
<li class="chapter" data-level="10.3.5" data-path="multiple-comparison.html"><a href="multiple-comparison.html#tukeys-method-for-all-pairwise-comparisons"><i class="fa fa-check"></i><b>10.3.5</b> Tukey’s Method for All Pairwise Comparisons</a></li>
<li class="chapter" data-level="10.3.6" data-path="multiple-comparison.html"><a href="multiple-comparison.html#displaying-pairwise-comparisons-graphically"><i class="fa fa-check"></i><b>10.3.6</b> Displaying Pairwise Comparisons Graphically</a></li>
<li class="chapter" data-level="10.3.7" data-path="multiple-comparison.html"><a href="multiple-comparison.html#dunnetts-two-sided-comparisons-with-a-control-and-dunnetts-two-sided-range-distribution"><i class="fa fa-check"></i><b>10.3.7</b> Dunnett’s Two-Sided Comparisons with a Control and Dunnett’s Two-Sided Range Distribution</a></li>
<li class="chapter" data-level="10.3.8" data-path="multiple-comparison.html"><a href="multiple-comparison.html#dunnetts-one-sided-comparisons-with-a-control"><i class="fa fa-check"></i><b>10.3.8</b> Dunnett’s One-Sided Comparisons with a Control</a></li>
<li class="chapter" data-level="10.3.9" data-path="multiple-comparison.html"><a href="multiple-comparison.html#maximum-modulus-distribution-multiple-inferences-for-independent-estimates"><i class="fa fa-check"></i><b>10.3.9</b> Maximum Modulus Distribution, Multiple Inferences for Independent Estimates</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="multiple-comparison.html"><a href="multiple-comparison.html#multiple-comparisons-among-treatment-means-in-the-one-way-unbalanced-anova"><i class="fa fa-check"></i><b>10.4</b> Multiple Comparisons among Treatment Means in the One-Way Unbalanced ANOVA</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="multiple-comparison.html"><a href="multiple-comparison.html#the-model-and-estimates"><i class="fa fa-check"></i><b>10.4.1</b> The Model and Estimates</a></li>
<li class="chapter" data-level="10.4.2" data-path="multiple-comparison.html"><a href="multiple-comparison.html#tukey-kramer-method"><i class="fa fa-check"></i><b>10.4.2</b> Tukey-Kramer Method</a></li>
<li class="chapter" data-level="10.4.3" data-path="multiple-comparison.html"><a href="multiple-comparison.html#alternative-simulation-based-method"><i class="fa fa-check"></i><b>10.4.3</b> Alternative Simulation-Based Method</a></li>
<li class="chapter" data-level="10.4.4" data-path="multiple-comparison.html"><a href="multiple-comparison.html#pairwise-comparisons-with-control"><i class="fa fa-check"></i><b>10.4.4</b> Pairwise Comparisons with Control</a></li>
<li class="chapter" data-level="10.4.5" data-path="multiple-comparison.html"><a href="multiple-comparison.html#comparisons-with-the-average-meananalysis-of-means-anom"><i class="fa fa-check"></i><b>10.4.5</b> Comparisons with the Average Mean–Analysis of Means (ANOM)</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="multiple-comparison.html"><a href="multiple-comparison.html#generalizations-for-the-analysis-of-covariance-ancova-model"><i class="fa fa-check"></i><b>10.5</b> Generalizations for the Analysis of Covariance (ANCOVA) model</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="multiple-comparison.html"><a href="multiple-comparison.html#dunnett-hsu-factor-analytic-approximation"><i class="fa fa-check"></i><b>10.5.1</b> Dunnett-Hsu Factor Analytic Approximation</a></li>
<li class="chapter" data-level="10.5.2" data-path="multiple-comparison.html"><a href="multiple-comparison.html#hsu-nelson-simulation-based-approximation-cvadjust-method"><i class="fa fa-check"></i><b>10.5.2</b> Hsu-Nelson Simulation-Based Approximation: CVADJUST Method</a></li>
<li class="chapter" data-level="10.5.3" data-path="multiple-comparison.html"><a href="multiple-comparison.html#comparisons-in-ancova-models-with-interaction"><i class="fa fa-check"></i><b>10.5.3</b> Comparisons in ANCOVA Models with Interaction</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="multiple-comparison.html"><a href="multiple-comparison.html#multiple-inferences-for-infinite-sets-of-parameters"><i class="fa fa-check"></i><b>10.6</b> Multiple Inferences for Infinite Sets of Parameters</a>
<ul>
<li class="chapter" data-level="10.6.1" data-path="multiple-comparison.html"><a href="multiple-comparison.html#scheffés-method"><i class="fa fa-check"></i><b>10.6.1</b> Scheffés Method</a></li>
<li class="chapter" data-level="10.6.2" data-path="multiple-comparison.html"><a href="multiple-comparison.html#finding-the-maximal-contrast"><i class="fa fa-check"></i><b>10.6.2</b> Finding the Maximal Contrast</a></li>
<li class="chapter" data-level="10.6.3" data-path="multiple-comparison.html"><a href="multiple-comparison.html#working-hotelling-method"><i class="fa fa-check"></i><b>10.6.3</b> Working-Hotelling method</a></li>
<li class="chapter" data-level="10.6.4" data-path="multiple-comparison.html"><a href="multiple-comparison.html#discrete-approximation-method"><i class="fa fa-check"></i><b>10.6.4</b> Discrete approximation method</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="multiple-comparison.html"><a href="multiple-comparison.html#multiple-comparisons-under-heteroscedasticity"><i class="fa fa-check"></i><b>10.7</b> Multiple Comparisons under Heteroscedasticity</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="multiple-comparison.html"><a href="multiple-comparison.html#introduction-of-heteroscedasticity"><i class="fa fa-check"></i><b>10.7.1</b> Introduction of heteroscedasticity</a></li>
<li class="chapter" data-level="10.7.2" data-path="multiple-comparison.html"><a href="multiple-comparison.html#satterthwaite-approximation"><i class="fa fa-check"></i><b>10.7.2</b> Satterthwaite Approximation</a></li>
<li class="chapter" data-level="10.7.3" data-path="multiple-comparison.html"><a href="multiple-comparison.html#maxt-method-under-heteroscedasticity"><i class="fa fa-check"></i><b>10.7.3</b> MaxT Method under Heteroscedasticity</a></li>
<li class="chapter" data-level="10.7.4" data-path="multiple-comparison.html"><a href="multiple-comparison.html#minp-method-under-heteroscedasticity"><i class="fa fa-check"></i><b>10.7.4</b> MinP Method under Heteroscedasticity</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="multiple-comparison.html"><a href="multiple-comparison.html#closed-and-stepwise-testing-methods"><i class="fa fa-check"></i><b>10.8</b> Closed and Stepwise Testing Methods</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="multiple-comparison.html"><a href="multiple-comparison.html#closed-family-of-hypotheses"><i class="fa fa-check"></i><b>10.8.1</b> Closed Family of Hypotheses</a></li>
<li class="chapter" data-level="10.8.2" data-path="multiple-comparison.html"><a href="multiple-comparison.html#bonferroni-holm-method"><i class="fa fa-check"></i><b>10.8.2</b> Bonferroni-Holm Method</a></li>
<li class="chapter" data-level="10.8.3" data-path="multiple-comparison.html"><a href="multiple-comparison.html#šidák-holm-method"><i class="fa fa-check"></i><b>10.8.3</b> Šidák-Holm Method</a></li>
<li class="chapter" data-level="10.8.4" data-path="multiple-comparison.html"><a href="multiple-comparison.html#closed-fisher-combination-method"><i class="fa fa-check"></i><b>10.8.4</b> Closed Fisher Combination Method</a></li>
<li class="chapter" data-level="10.8.5" data-path="multiple-comparison.html"><a href="multiple-comparison.html#simes-hommel-method"><i class="fa fa-check"></i><b>10.8.5</b> Simes-Hommel Method</a></li>
<li class="chapter" data-level="10.8.6" data-path="multiple-comparison.html"><a href="multiple-comparison.html#hochbergs-ok-step-up"><i class="fa fa-check"></i><b>10.8.6</b> Hochberg’s O(k) Step-Up</a></li>
<li class="chapter" data-level="10.8.7" data-path="multiple-comparison.html"><a href="multiple-comparison.html#sequential-testing-with-fixed-sequences"><i class="fa fa-check"></i><b>10.8.7</b> Sequential Testing with Fixed Sequences</a></li>
<li class="chapter" data-level="10.8.8" data-path="multiple-comparison.html"><a href="multiple-comparison.html#sequential-testing-using-gatekeeping-methods"><i class="fa fa-check"></i><b>10.8.8</b> Sequential Testing Using Gatekeeping Methods</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="multiple-comparison.html"><a href="multiple-comparison.html#closed-testing-of-pairwise-comparisons-and-general-contrasts"><i class="fa fa-check"></i><b>10.9</b> Closed Testing of Pairwise Comparisons and General Contrasts</a>
<ul>
<li class="chapter" data-level="10.9.1" data-path="multiple-comparison.html"><a href="multiple-comparison.html#incorporating-logical-constraints"><i class="fa fa-check"></i><b>10.9.1</b> Incorporating Logical Constraints</a></li>
<li class="chapter" data-level="10.9.2" data-path="multiple-comparison.html"><a href="multiple-comparison.html#shaffers-method"><i class="fa fa-check"></i><b>10.9.2</b> Shaffer’s Method</a></li>
<li class="chapter" data-level="10.9.3" data-path="multiple-comparison.html"><a href="multiple-comparison.html#extended-shaffer-royen-method"><i class="fa fa-check"></i><b>10.9.3</b> Extended Shaffer-Royen Method</a></li>
<li class="chapter" data-level="10.9.4" data-path="multiple-comparison.html"><a href="multiple-comparison.html#step-down-dunnett-test"><i class="fa fa-check"></i><b>10.9.4</b> Step-down Dunnett test</a></li>
</ul></li>
<li class="chapter" data-level="10.10" data-path="multiple-comparison.html"><a href="multiple-comparison.html#multiple-comparisons-with-binary-data"><i class="fa fa-check"></i><b>10.10</b> Multiple Comparisons with Binary Data</a>
<ul>
<li class="chapter" data-level="10.10.1" data-path="multiple-comparison.html"><a href="multiple-comparison.html#introduction-6"><i class="fa fa-check"></i><b>10.10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.10.2" data-path="multiple-comparison.html"><a href="multiple-comparison.html#multivariate-two-sample-binary-outcomes"><i class="fa fa-check"></i><b>10.10.2</b> Multivariate Two-Sample Binary Outcomes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html"><i class="fa fa-check"></i><b>11</b> Correlation and Regression</a>
<ul>
<li class="chapter" data-level="11.1" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#correlation-1"><i class="fa fa-check"></i><b>11.1</b> Correlation</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#pearson-correlation-coefficient"><i class="fa fa-check"></i><b>11.1.1</b> Pearson correlation coefficient</a></li>
<li class="chapter" data-level="11.1.2" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#spearmans-rank-correlation-coefficient"><i class="fa fa-check"></i><b>11.1.2</b> Spearman’s rank correlation coefficient</a></li>
<li class="chapter" data-level="11.1.3" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#kendall-rank-correlation-coefficient"><i class="fa fa-check"></i><b>11.1.3</b> Kendall rank correlation coefficient</a></li>
<li class="chapter" data-level="11.1.4" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#intraclass-correlation"><i class="fa fa-check"></i><b>11.1.4</b> Intraclass correlation</a></li>
<li class="chapter" data-level="11.1.5" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#visualize-the-correlation-in-r"><i class="fa fa-check"></i><b>11.1.5</b> Visualize the correlation in R</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#ordinary-least-squares-ols"><i class="fa fa-check"></i><b>11.2</b> Ordinary least squares (OLS)</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#assumpions"><i class="fa fa-check"></i><b>11.2.1</b> Assumpions</a></li>
<li class="chapter" data-level="11.2.2" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#interpretation"><i class="fa fa-check"></i><b>11.2.2</b> Interpretation</a></li>
<li class="chapter" data-level="11.2.3" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#matrix-solution"><i class="fa fa-check"></i><b>11.2.3</b> Matrix Solution</a></li>
<li class="chapter" data-level="11.2.4" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#gauss-markov-theorem"><i class="fa fa-check"></i><b>11.2.4</b> Gauss-Markov Theorem</a></li>
<li class="chapter" data-level="11.2.5" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#limitation"><i class="fa fa-check"></i><b>11.2.5</b> limitation</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#model-statistics"><i class="fa fa-check"></i><b>11.3</b> Model Statistics</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#residuals-standard-error"><i class="fa fa-check"></i><b>11.3.1</b> Residuals Standard Error</a></li>
<li class="chapter" data-level="11.3.2" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#r-squared-and-adjusted-r-squared"><i class="fa fa-check"></i><b>11.3.2</b> R-Squared and Adjusted R-Squared</a></li>
<li class="chapter" data-level="11.3.3" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#t-statistic"><i class="fa fa-check"></i><b>11.3.3</b> T Statistic</a></li>
<li class="chapter" data-level="11.3.4" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#f-statistic"><i class="fa fa-check"></i><b>11.3.4</b> F Statistic</a></li>
<li class="chapter" data-level="11.3.5" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#confidence-intervals"><i class="fa fa-check"></i><b>11.3.5</b> Confidence Intervals</a></li>
<li class="chapter" data-level="11.3.6" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#likelihood-ratio-test"><i class="fa fa-check"></i><b>11.3.6</b> Likelihood-ratio test</a></li>
<li class="chapter" data-level="11.3.7" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#accuracy"><i class="fa fa-check"></i><b>11.3.7</b> Accuracy</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#model-diagnostics"><i class="fa fa-check"></i><b>11.4</b> Model Diagnostics</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#checking-error-assumptions"><i class="fa fa-check"></i><b>11.4.1</b> Checking Error Assumptions</a></li>
<li class="chapter" data-level="11.4.2" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#finding-unusual-observations"><i class="fa fa-check"></i><b>11.4.2</b> Finding Unusual Observations</a></li>
<li class="chapter" data-level="11.4.3" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#checking-the-structure-of-the-model"><i class="fa fa-check"></i><b>11.4.3</b> Checking the Structure of the Model</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#sas-implementation-proc-reg"><i class="fa fa-check"></i><b>11.5</b> SAS implementation Proc Reg</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#options"><i class="fa fa-check"></i><b>11.5.1</b> Options</a></li>
<li class="chapter" data-level="11.5.2" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#diagnose"><i class="fa fa-check"></i><b>11.5.2</b> Diagnose</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="correlation-and-regression.html"><a href="correlation-and-regression.html#r-implementation-8"><i class="fa fa-check"></i><b>11.6</b> R implementation</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html"><i class="fa fa-check"></i><b>12</b> Advanced Linear Regression</a>
<ul>
<li class="chapter" data-level="12.1" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#model-selection"><i class="fa fa-check"></i><b>12.1</b> Model Selection</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#selection-methods"><i class="fa fa-check"></i><b>12.1.1</b> Selection Methods</a></li>
<li class="chapter" data-level="12.1.2" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#selection-criteria"><i class="fa fa-check"></i><b>12.1.2</b> Selection Criteria</a></li>
<li class="chapter" data-level="12.1.3" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#k--fold-cross-validation"><i class="fa fa-check"></i><b>12.1.3</b> k- Fold Cross validation</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#practical-difficulties-using-ols"><i class="fa fa-check"></i><b>12.2</b> Practical Difficulties using OLS</a></li>
<li class="chapter" data-level="12.3" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#skewness"><i class="fa fa-check"></i><b>12.3</b> Skewness</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#introduction-7"><i class="fa fa-check"></i><b>12.3.1</b> Introduction</a></li>
<li class="chapter" data-level="12.3.2" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#baisc-transformation"><i class="fa fa-check"></i><b>12.3.2</b> Baisc Transformation</a></li>
<li class="chapter" data-level="12.3.3" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#box-cox-power-transformation"><i class="fa fa-check"></i><b>12.3.3</b> Box-Cox Power Transformation</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#scale-1"><i class="fa fa-check"></i><b>12.4</b> Scale</a></li>
<li class="chapter" data-level="12.5" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#interaction"><i class="fa fa-check"></i><b>12.5</b> Interaction</a>
<ul>
<li class="chapter" data-level="12.5.1" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#simple-slopes-analysis"><i class="fa fa-check"></i><b>12.5.1</b> Simple slopes analysis</a></li>
<li class="chapter" data-level="12.5.2" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#plotting-interactions"><i class="fa fa-check"></i><b>12.5.2</b> Plotting Interactions</a></li>
<li class="chapter" data-level="12.5.3" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#check-linearity-assumption"><i class="fa fa-check"></i><b>12.5.3</b> Check linearity assumption</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#collinearity"><i class="fa fa-check"></i><b>12.6</b> Collinearity</a></li>
<li class="chapter" data-level="12.7" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#problems-with-the-error"><i class="fa fa-check"></i><b>12.7</b> Problems with the Error</a>
<ul>
<li class="chapter" data-level="12.7.1" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#generalized-least-squares"><i class="fa fa-check"></i><b>12.7.1</b> Generalized Least Squares</a></li>
<li class="chapter" data-level="12.7.2" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#weighted-least-squares"><i class="fa fa-check"></i><b>12.7.2</b> Weighted Least Squares</a></li>
<li class="chapter" data-level="12.7.3" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#robust-regression"><i class="fa fa-check"></i><b>12.7.3</b> Robust Regression</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#shrinkage-methods"><i class="fa fa-check"></i><b>12.8</b> Shrinkage Methods</a>
<ul>
<li class="chapter" data-level="12.8.1" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#principal-components-analzsis"><i class="fa fa-check"></i><b>12.8.1</b> Principal Components Analzsis</a></li>
<li class="chapter" data-level="12.8.2" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#partial-least-squares"><i class="fa fa-check"></i><b>12.8.2</b> Partial Least Squares</a></li>
<li class="chapter" data-level="12.8.3" data-path="advanced-linear-regression.html"><a href="advanced-linear-regression.html#ridge-regression"><i class="fa fa-check"></i><b>12.8.3</b> Ridge Regression</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="logistic-regression.html"><a href="logistic-regression.html"><i class="fa fa-check"></i><b>13</b> Logistic Regression</a>
<ul>
<li class="chapter" data-level="13.1" data-path="logistic-regression.html"><a href="logistic-regression.html#introduction-8"><i class="fa fa-check"></i><b>13.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="logistic-regression.html"><a href="logistic-regression.html#violation-of-assumptions-of-ordinary-least-squares-ols"><i class="fa fa-check"></i><b>13.1.1</b> Violation of assumptions of Ordinary least squares (OLS)</a></li>
<li class="chapter" data-level="13.1.2" data-path="logistic-regression.html"><a href="logistic-regression.html#more-fundamental-problem-outside-01"><i class="fa fa-check"></i><b>13.1.2</b> More fundamental problem outside [0,1]</a></li>
<li class="chapter" data-level="13.1.3" data-path="logistic-regression.html"><a href="logistic-regression.html#logistic-regression-model"><i class="fa fa-check"></i><b>13.1.3</b> Logistic Regression Model</a></li>
<li class="chapter" data-level="13.1.4" data-path="logistic-regression.html"><a href="logistic-regression.html#estimation-of-the-logistic-model"><i class="fa fa-check"></i><b>13.1.4</b> Estimation of the Logistic Model</a></li>
<li class="chapter" data-level="13.1.5" data-path="logistic-regression.html"><a href="logistic-regression.html#convergence-problems"><i class="fa fa-check"></i><b>13.1.5</b> Convergence Problems</a></li>
<li class="chapter" data-level="13.1.6" data-path="logistic-regression.html"><a href="logistic-regression.html#use-exact-methods."><i class="fa fa-check"></i><b>13.1.6</b> Use exact methods.</a></li>
<li class="chapter" data-level="13.1.7" data-path="logistic-regression.html"><a href="logistic-regression.html#use-penalized-likelihood"><i class="fa fa-check"></i><b>13.1.7</b> Use penalized likelihood</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="logistic-regression.html"><a href="logistic-regression.html#logit-modell"><i class="fa fa-check"></i><b>13.2</b> Logit Modell</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="logistic-regression.html"><a href="logistic-regression.html#introduction-9"><i class="fa fa-check"></i><b>13.2.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2.2" data-path="logistic-regression.html"><a href="logistic-regression.html#r-implementation-9"><i class="fa fa-check"></i><b>13.2.2</b> R Implementation</a></li>
<li class="chapter" data-level="13.2.3" data-path="logistic-regression.html"><a href="logistic-regression.html#sas-implementation-7"><i class="fa fa-check"></i><b>13.2.3</b> SAS Implementation</a></li>
<li class="chapter" data-level="13.2.4" data-path="logistic-regression.html"><a href="logistic-regression.html#multicollinearity"><i class="fa fa-check"></i><b>13.2.4</b> Multicollinearity</a></li>
<li class="chapter" data-level="13.2.5" data-path="logistic-regression.html"><a href="logistic-regression.html#goodness-of-fit-statistics-pearson-deviance"><i class="fa fa-check"></i><b>13.2.5</b> Goodness-of-Fit Statistics Pearson deviance</a></li>
<li class="chapter" data-level="13.2.6" data-path="logistic-regression.html"><a href="logistic-regression.html#hosmer-and-lemeshow-goodness-of-fit-test"><i class="fa fa-check"></i><b>13.2.6</b> Hosmer and Lemeshow Goodness-of-Fit Test</a></li>
<li class="chapter" data-level="13.2.7" data-path="logistic-regression.html"><a href="logistic-regression.html#statistics-measuring-predictive-power-r2"><i class="fa fa-check"></i><b>13.2.7</b> Statistics Measuring Predictive Power <span class="math inline">\(R^2\)</span></a></li>
<li class="chapter" data-level="13.2.8" data-path="logistic-regression.html"><a href="logistic-regression.html#roc-curves"><i class="fa fa-check"></i><b>13.2.8</b> ROC Curves</a></li>
<li class="chapter" data-level="13.2.9" data-path="logistic-regression.html"><a href="logistic-regression.html#predicted-values-residuals-and-influence-statistics"><i class="fa fa-check"></i><b>13.2.9</b> Predicted Values, Residuals, and Influence Statistics</a></li>
<li class="chapter" data-level="13.2.10" data-path="logistic-regression.html"><a href="logistic-regression.html#unobserved-heterogeneity"><i class="fa fa-check"></i><b>13.2.10</b> Unobserved Heterogeneity</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="logistic-regression.html"><a href="logistic-regression.html#illustration-in-sas"><i class="fa fa-check"></i><b>13.3</b> Illustration in SAS</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="logistic-regression.html"><a href="logistic-regression.html#effects-of-predictor-variables"><i class="fa fa-check"></i><b>13.3.1</b> Effects of Predictor Variables</a></li>
<li class="chapter" data-level="13.3.2" data-path="logistic-regression.html"><a href="logistic-regression.html#odds-ratio-plot"><i class="fa fa-check"></i><b>13.3.2</b> Odds ratio plot</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="logistic-regression.html"><a href="logistic-regression.html#probit-modell"><i class="fa fa-check"></i><b>13.4</b> Probit Modell</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="logistic-regression.html"><a href="logistic-regression.html#introduction-10"><i class="fa fa-check"></i><b>13.4.1</b> Introduction</a></li>
<li class="chapter" data-level="13.4.2" data-path="logistic-regression.html"><a href="logistic-regression.html#r-implemetation"><i class="fa fa-check"></i><b>13.4.2</b> R Implemetation</a></li>
<li class="chapter" data-level="13.4.3" data-path="logistic-regression.html"><a href="logistic-regression.html#sas-implementation-8"><i class="fa fa-check"></i><b>13.4.3</b> SAS Implementation</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="logistic-regression.html"><a href="logistic-regression.html#complementary-log-log-modell"><i class="fa fa-check"></i><b>13.5</b> Complementary log-log-Modell</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="logistic-regression.html"><a href="logistic-regression.html#r-implemetation-1"><i class="fa fa-check"></i><b>13.5.1</b> R Implemetation</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="logistic-regression.html"><a href="logistic-regression.html#multicategory-logit-models"><i class="fa fa-check"></i><b>13.6</b> Multicategory Logit Models</a>
<ul>
<li class="chapter" data-level="13.6.1" data-path="logistic-regression.html"><a href="logistic-regression.html#multinomialverteilung"><i class="fa fa-check"></i><b>13.6.1</b> Multinomialverteilung</a></li>
<li class="chapter" data-level="13.6.2" data-path="logistic-regression.html"><a href="logistic-regression.html#nominal-response"><i class="fa fa-check"></i><b>13.6.2</b> Nominal Response</a></li>
<li class="chapter" data-level="13.6.3" data-path="logistic-regression.html"><a href="logistic-regression.html#ordinal-response-cumulative-logit-model"><i class="fa fa-check"></i><b>13.6.3</b> Ordinal Response: Cumulative Logit Model</a></li>
</ul></li>
<li class="chapter" data-level="13.7" data-path="logistic-regression.html"><a href="logistic-regression.html#adjacent-categories-model"><i class="fa fa-check"></i><b>13.7</b> Adjacent Categories Model</a></li>
<li class="chapter" data-level="13.8" data-path="logistic-regression.html"><a href="logistic-regression.html#continuation-ratio-model"><i class="fa fa-check"></i><b>13.8</b> Continuation Ratio Model</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html"><i class="fa fa-check"></i><b>14</b> Advanced Logistic Regression</a>
<ul>
<li class="chapter" data-level="14.1" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#logit-analysis-of-contingency-tables"><i class="fa fa-check"></i><b>14.1</b> Logit Analysis of Contingency Tables</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#two-way-table"><i class="fa fa-check"></i><b>14.1.1</b> Two-Way Table</a></li>
<li class="chapter" data-level="14.1.2" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#three-way-table"><i class="fa fa-check"></i><b>14.1.2</b> Three-Way Table</a></li>
<li class="chapter" data-level="14.1.3" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#four-way-table"><i class="fa fa-check"></i><b>14.1.3</b> Four-Way Table</a></li>
<li class="chapter" data-level="14.1.4" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#overdispersion"><i class="fa fa-check"></i><b>14.1.4</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#loglinear-analysis-of-contingency-tables"><i class="fa fa-check"></i><b>14.2</b> Loglinear Analysis of Contingency Tables</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#two-way-table-1"><i class="fa fa-check"></i><b>14.2.1</b> Two-way Table</a></li>
<li class="chapter" data-level="14.2.2" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#problem-of-zeros"><i class="fa fa-check"></i><b>14.2.2</b> Problem of Zeros</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#discrete-choice-analysis"><i class="fa fa-check"></i><b>14.3</b> Discrete Choice Analysis</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#logistic-strata"><i class="fa fa-check"></i><b>14.3.1</b> Logistic Strata</a></li>
<li class="chapter" data-level="14.3.2" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#conditional-logit-model"><i class="fa fa-check"></i><b>14.3.2</b> Conditional logit model</a></li>
<li class="chapter" data-level="14.3.3" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#ranked-data"><i class="fa fa-check"></i><b>14.3.3</b> Ranked Data</a></li>
<li class="chapter" data-level="14.3.4" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#heteroscedastic-extreme-value-hev-model"><i class="fa fa-check"></i><b>14.3.4</b> Heteroscedastic extreme value (HEV) Model</a></li>
<li class="chapter" data-level="14.3.5" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#nested-logit-model"><i class="fa fa-check"></i><b>14.3.5</b> Nested logit model</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#longitudinal-and-other-clustered-data"><i class="fa fa-check"></i><b>14.4</b> Longitudinal and Other Clustered Data</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#introduction-11"><i class="fa fa-check"></i><b>14.4.1</b> Introduction</a></li>
<li class="chapter" data-level="14.4.2" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#robust-standard-errors"><i class="fa fa-check"></i><b>14.4.2</b> Robust Standard Errors</a></li>
<li class="chapter" data-level="14.4.3" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#gee-estimation-with-proc-genmod"><i class="fa fa-check"></i><b>14.4.3</b> GEE Estimation with PROC GENMOD</a></li>
<li class="chapter" data-level="14.4.4" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#mixed-models-with-proc-glimmix"><i class="fa fa-check"></i><b>14.4.4</b> Mixed Models with PROC GLIMMIX</a></li>
<li class="chapter" data-level="14.4.5" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#fixed-effects-with-conditional-logistic-regression"><i class="fa fa-check"></i><b>14.4.5</b> Fixed-Effects with Conditional Logistic Regression</a></li>
<li class="chapter" data-level="14.4.6" data-path="advanced-logistic-regression.html"><a href="advanced-logistic-regression.html#hybrid-method"><i class="fa fa-check"></i><b>14.4.6</b> Hybrid Method</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="count-data-regression.html"><a href="count-data-regression.html"><i class="fa fa-check"></i><b>15</b> Count Data Regression</a>
<ul>
<li class="chapter" data-level="15.1" data-path="count-data-regression.html"><a href="count-data-regression.html#introduction-of-count-data-regression"><i class="fa fa-check"></i><b>15.1</b> Introduction of Count Data Regression</a></li>
<li class="chapter" data-level="15.2" data-path="count-data-regression.html"><a href="count-data-regression.html#poisson-regression"><i class="fa fa-check"></i><b>15.2</b> Poisson regression</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="count-data-regression.html"><a href="count-data-regression.html#model"><i class="fa fa-check"></i><b>15.2.1</b> Model</a></li>
<li class="chapter" data-level="15.2.2" data-path="count-data-regression.html"><a href="count-data-regression.html#interpretation-of-estimated-coefficients"><i class="fa fa-check"></i><b>15.2.2</b> Interpretation of Estimated Coefficients</a></li>
<li class="chapter" data-level="15.2.3" data-path="count-data-regression.html"><a href="count-data-regression.html#overdispersion-1"><i class="fa fa-check"></i><b>15.2.3</b> Overdispersion</a></li>
<li class="chapter" data-level="15.2.4" data-path="count-data-regression.html"><a href="count-data-regression.html#adjustment-for-varying-time-spans"><i class="fa fa-check"></i><b>15.2.4</b> Adjustment for Varying Time Spans</a></li>
<li class="chapter" data-level="15.2.5" data-path="count-data-regression.html"><a href="count-data-regression.html#sas-implementation-using-proc-genmod"><i class="fa fa-check"></i><b>15.2.5</b> SAS implementation using PROC GENMOD</a></li>
<li class="chapter" data-level="15.2.6" data-path="count-data-regression.html"><a href="count-data-regression.html#r-implementation-12"><i class="fa fa-check"></i><b>15.2.6</b> R implementation</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="count-data-regression.html"><a href="count-data-regression.html#negative-binomial-regression"><i class="fa fa-check"></i><b>15.3</b> Negative Binomial Regression</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="count-data-regression.html"><a href="count-data-regression.html#model-1"><i class="fa fa-check"></i><b>15.3.1</b> Model</a></li>
<li class="chapter" data-level="15.3.2" data-path="count-data-regression.html"><a href="count-data-regression.html#sas-implementation-11"><i class="fa fa-check"></i><b>15.3.2</b> SAS implementation</a></li>
<li class="chapter" data-level="15.3.3" data-path="count-data-regression.html"><a href="count-data-regression.html#r-implementation-13"><i class="fa fa-check"></i><b>15.3.3</b> R implementation</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="count-data-regression.html"><a href="count-data-regression.html#zero-truncated-poisson-regression-model"><i class="fa fa-check"></i><b>15.4</b> Zero-truncated Poisson Regression Model</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="count-data-regression.html"><a href="count-data-regression.html#model-2"><i class="fa fa-check"></i><b>15.4.1</b> Model</a></li>
<li class="chapter" data-level="15.4.2" data-path="count-data-regression.html"><a href="count-data-regression.html#sas-implementation-12"><i class="fa fa-check"></i><b>15.4.2</b> SAS Implementation</a></li>
<li class="chapter" data-level="15.4.3" data-path="count-data-regression.html"><a href="count-data-regression.html#r-implementation-14"><i class="fa fa-check"></i><b>15.4.3</b> R implementation</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="count-data-regression.html"><a href="count-data-regression.html#zero-insflated-model"><i class="fa fa-check"></i><b>15.5</b> Zero-Insflated Model</a>
<ul>
<li class="chapter" data-level="15.5.1" data-path="count-data-regression.html"><a href="count-data-regression.html#model-3"><i class="fa fa-check"></i><b>15.5.1</b> Model</a></li>
<li class="chapter" data-level="15.5.2" data-path="count-data-regression.html"><a href="count-data-regression.html#interpretation-of-estimated-coefficients-1"><i class="fa fa-check"></i><b>15.5.2</b> Interpretation of Estimated Coefficients</a></li>
<li class="chapter" data-level="15.5.3" data-path="count-data-regression.html"><a href="count-data-regression.html#sas-implementation-13"><i class="fa fa-check"></i><b>15.5.3</b> SAS Implementation</a></li>
<li class="chapter" data-level="15.5.4" data-path="count-data-regression.html"><a href="count-data-regression.html#r-implementation-15"><i class="fa fa-check"></i><b>15.5.4</b> R Implementation</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="count-data-regression.html"><a href="count-data-regression.html#hurdle-models"><i class="fa fa-check"></i><b>15.6</b> Hurdle models</a>
<ul>
<li class="chapter" data-level="15.6.1" data-path="count-data-regression.html"><a href="count-data-regression.html#model-4"><i class="fa fa-check"></i><b>15.6.1</b> Model</a></li>
<li class="chapter" data-level="15.6.2" data-path="count-data-regression.html"><a href="count-data-regression.html#sas-implementation-14"><i class="fa fa-check"></i><b>15.6.2</b> SAS Implementation</a></li>
<li class="chapter" data-level="15.6.3" data-path="count-data-regression.html"><a href="count-data-regression.html#r-implementation-16"><i class="fa fa-check"></i><b>15.6.3</b> R Implementation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="proportion-response-regression.html"><a href="proportion-response-regression.html"><i class="fa fa-check"></i><b>16</b> Proportion Response Regression</a>
<ul>
<li class="chapter" data-level="16.1" data-path="proportion-response-regression.html"><a href="proportion-response-regression.html#beta-regression"><i class="fa fa-check"></i><b>16.1</b> Beta Regression</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="proportion-response-regression.html"><a href="proportion-response-regression.html#model-5"><i class="fa fa-check"></i><b>16.1.1</b> Model</a></li>
<li class="chapter" data-level="16.1.2" data-path="proportion-response-regression.html"><a href="proportion-response-regression.html#sas-implementation-15"><i class="fa fa-check"></i><b>16.1.2</b> SAS Implementation</a></li>
<li class="chapter" data-level="16.1.3" data-path="proportion-response-regression.html"><a href="proportion-response-regression.html#r-implementation-betareg"><i class="fa fa-check"></i><b>16.1.3</b> R Implementation “betareg”</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="proportion-response-regression.html"><a href="proportion-response-regression.html#zero-inflated-beta-regression"><i class="fa fa-check"></i><b>16.2</b> Zero-inflated Beta Regression</a></li>
<li class="chapter" data-level="16.3" data-path="proportion-response-regression.html"><a href="proportion-response-regression.html#one-inflated-beta-regression"><i class="fa fa-check"></i><b>16.3</b> One-inflated Beta Regression</a></li>
<li class="chapter" data-level="16.4" data-path="proportion-response-regression.html"><a href="proportion-response-regression.html#zero-one-inflated-beta-regression"><i class="fa fa-check"></i><b>16.4</b> Zero-one-inflated Beta Regression</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="survival-analysis.html"><a href="survival-analysis.html"><i class="fa fa-check"></i><b>17</b> Survival Analysis</a>
<ul>
<li class="chapter" data-level="17.1" data-path="survival-analysis.html"><a href="survival-analysis.html#preliminary"><i class="fa fa-check"></i><b>17.1</b> Preliminary</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="survival-analysis.html"><a href="survival-analysis.html#probability-density-function"><i class="fa fa-check"></i><b>17.1.1</b> Probability density function</a></li>
<li class="chapter" data-level="17.1.2" data-path="survival-analysis.html"><a href="survival-analysis.html#survival-function"><i class="fa fa-check"></i><b>17.1.2</b> Survival function</a></li>
<li class="chapter" data-level="17.1.3" data-path="survival-analysis.html"><a href="survival-analysis.html#hazard-function"><i class="fa fa-check"></i><b>17.1.3</b> Hazard function</a></li>
<li class="chapter" data-level="17.1.4" data-path="survival-analysis.html"><a href="survival-analysis.html#cumulative-hazard-function"><i class="fa fa-check"></i><b>17.1.4</b> Cumulative hazard function</a></li>
<li class="chapter" data-level="17.1.5" data-path="survival-analysis.html"><a href="survival-analysis.html#mean-residual-life"><i class="fa fa-check"></i><b>17.1.5</b> Mean Residual Life</a></li>
<li class="chapter" data-level="17.1.6" data-path="survival-analysis.html"><a href="survival-analysis.html#relation-between-functions"><i class="fa fa-check"></i><b>17.1.6</b> Relation between functions</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="survival-analysis.html"><a href="survival-analysis.html#kaplan-meier-estimator"><i class="fa fa-check"></i><b>17.2</b> Kaplan-Meier estimator</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="survival-analysis.html"><a href="survival-analysis.html#km-introduction"><i class="fa fa-check"></i><b>17.2.1</b> KM Introduction</a></li>
<li class="chapter" data-level="17.2.2" data-path="survival-analysis.html"><a href="survival-analysis.html#nelson-aalen-estimator-of-the-cumulative-hazard-function"><i class="fa fa-check"></i><b>17.2.2</b> Nelson-Aalen estimator of the cumulative hazard function</a></li>
<li class="chapter" data-level="17.2.3" data-path="survival-analysis.html"><a href="survival-analysis.html#survival-curve-in-sas"><i class="fa fa-check"></i><b>17.2.3</b> Survival curve in SAS</a></li>
<li class="chapter" data-level="17.2.4" data-path="survival-analysis.html"><a href="survival-analysis.html#km-survival-plots-in-sas"><i class="fa fa-check"></i><b>17.2.4</b> KM Survival Plots in SAS</a></li>
<li class="chapter" data-level="17.2.5" data-path="survival-analysis.html"><a href="survival-analysis.html#convert-personal-level-to-personal-period-in-r"><i class="fa fa-check"></i><b>17.2.5</b> Convert Personal-level to Personal-period in R</a></li>
<li class="chapter" data-level="17.2.6" data-path="survival-analysis.html"><a href="survival-analysis.html#package-survfit-in-r"><i class="fa fa-check"></i><b>17.2.6</b> Package survfit in R</a></li>
<li class="chapter" data-level="17.2.7" data-path="survival-analysis.html"><a href="survival-analysis.html#km-survival-plots-in-r"><i class="fa fa-check"></i><b>17.2.7</b> KM Survival Plots in R</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="survival-analysis.html"><a href="survival-analysis.html#compare-the-survival-function"><i class="fa fa-check"></i><b>17.3</b> Compare the survival function</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="survival-analysis.html"><a href="survival-analysis.html#tests-of-equality-of-the-survival-function"><i class="fa fa-check"></i><b>17.3.1</b> Tests of equality of the survival function</a></li>
<li class="chapter" data-level="17.3.2" data-path="survival-analysis.html"><a href="survival-analysis.html#other-nonparametric-tests-for-strata-statement"><i class="fa fa-check"></i><b>17.3.2</b> Other nonparametric tests for STRATA statement</a></li>
<li class="chapter" data-level="17.3.3" data-path="survival-analysis.html"><a href="survival-analysis.html#multiple-comparisons"><i class="fa fa-check"></i><b>17.3.3</b> Multiple comparisons</a></li>
<li class="chapter" data-level="17.3.4" data-path="survival-analysis.html"><a href="survival-analysis.html#comparing-survival-functions-using-log-rank-hr"><i class="fa fa-check"></i><b>17.3.4</b> Comparing survival functions using Log-Rank HR</a></li>
<li class="chapter" data-level="17.3.5" data-path="survival-analysis.html"><a href="survival-analysis.html#mantel-haenszel-hr"><i class="fa fa-check"></i><b>17.3.5</b> Mantel-Haenszel HR</a></li>
<li class="chapter" data-level="17.3.6" data-path="survival-analysis.html"><a href="survival-analysis.html#analysis-of-covariance-adjustment"><i class="fa fa-check"></i><b>17.3.6</b> Analysis-of-covariance (Adjustment)</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="survival-analysis.html"><a href="survival-analysis.html#accelerated-failure-time-aft-model"><i class="fa fa-check"></i><b>17.4</b> Accelerated failure time (AFT) model</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="survival-analysis.html"><a href="survival-analysis.html#introduction-12"><i class="fa fa-check"></i><b>17.4.1</b> Introduction</a></li>
<li class="chapter" data-level="17.4.2" data-path="survival-analysis.html"><a href="survival-analysis.html#proc-lifereg"><i class="fa fa-check"></i><b>17.4.2</b> PROC LIFEREG</a></li>
<li class="chapter" data-level="17.4.3" data-path="survival-analysis.html"><a href="survival-analysis.html#residuum-distribution"><i class="fa fa-check"></i><b>17.4.3</b> Residuum distribution</a></li>
<li class="chapter" data-level="17.4.4" data-path="survival-analysis.html"><a href="survival-analysis.html#exponential-model"><i class="fa fa-check"></i><b>17.4.4</b> Exponential Model</a></li>
<li class="chapter" data-level="17.4.5" data-path="survival-analysis.html"><a href="survival-analysis.html#weibull-model"><i class="fa fa-check"></i><b>17.4.5</b> Weibull Model</a></li>
<li class="chapter" data-level="17.4.6" data-path="survival-analysis.html"><a href="survival-analysis.html#log-normal-model"><i class="fa fa-check"></i><b>17.4.6</b> Log-Normal Model</a></li>
<li class="chapter" data-level="17.4.7" data-path="survival-analysis.html"><a href="survival-analysis.html#log-logistic-model"><i class="fa fa-check"></i><b>17.4.7</b> Log-Logistic Model</a></li>
<li class="chapter" data-level="17.4.8" data-path="survival-analysis.html"><a href="survival-analysis.html#fit-statistics-for-model-comparsion"><i class="fa fa-check"></i><b>17.4.8</b> Fit statistics for model comparsion</a></li>
<li class="chapter" data-level="17.4.9" data-path="survival-analysis.html"><a href="survival-analysis.html#graphical-method-for-distinguishing-different-distributions-in-sas"><i class="fa fa-check"></i><b>17.4.9</b> Graphical method for distinguishing different distributions in SAS</a></li>
<li class="chapter" data-level="17.4.10" data-path="survival-analysis.html"><a href="survival-analysis.html#prediction-and-hazard-function"><i class="fa fa-check"></i><b>17.4.10</b> Prediction and hazard function</a></li>
<li class="chapter" data-level="17.4.11" data-path="survival-analysis.html"><a href="survival-analysis.html#left-censoring-and-interval-censoring"><i class="fa fa-check"></i><b>17.4.11</b> Left Censoring and Interval Censoring</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="survival-analysis.html"><a href="survival-analysis.html#cox-proportional-hazards-model"><i class="fa fa-check"></i><b>17.5</b> Cox Proportional Hazards Model</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="survival-analysis.html"><a href="survival-analysis.html#introduction-13"><i class="fa fa-check"></i><b>17.5.1</b> Introduction</a></li>
<li class="chapter" data-level="17.5.2" data-path="survival-analysis.html"><a href="survival-analysis.html#parameter-estimate"><i class="fa fa-check"></i><b>17.5.2</b> Parameter estimate</a></li>
<li class="chapter" data-level="17.5.3" data-path="survival-analysis.html"><a href="survival-analysis.html#parameter-test"><i class="fa fa-check"></i><b>17.5.3</b> Parameter test</a></li>
<li class="chapter" data-level="17.5.4" data-path="survival-analysis.html"><a href="survival-analysis.html#graphs-of-the-survival-and-baseline-hazard-function-in-sas"><i class="fa fa-check"></i><b>17.5.4</b> Graphs of the survival and baseline hazard function in SAS</a></li>
<li class="chapter" data-level="17.5.5" data-path="survival-analysis.html"><a href="survival-analysis.html#check-proportional-hazards"><i class="fa fa-check"></i><b>17.5.5</b> Check proportional hazards</a></li>
<li class="chapter" data-level="17.5.6" data-path="survival-analysis.html"><a href="survival-analysis.html#dealing-with-nonproportionality"><i class="fa fa-check"></i><b>17.5.6</b> Dealing with nonproportionality</a></li>
<li class="chapter" data-level="17.5.7" data-path="survival-analysis.html"><a href="survival-analysis.html#model-diagnostics-1"><i class="fa fa-check"></i><b>17.5.7</b> Model Diagnostics</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html"><i class="fa fa-check"></i><b>18</b> Advanced Survival Analysis</a>
<ul>
<li class="chapter" data-level="18.1" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#tied-or-discrete-data"><i class="fa fa-check"></i><b>18.1</b> Tied or Discrete Data</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#introduction-of-tie"><i class="fa fa-check"></i><b>18.1.1</b> Introduction of Tie</a></li>
<li class="chapter" data-level="18.1.2" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#discrete-time"><i class="fa fa-check"></i><b>18.1.2</b> Discrete time</a></li>
<li class="chapter" data-level="18.1.3" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#notation-and-definitions"><i class="fa fa-check"></i><b>18.1.3</b> Notation and Definitions</a></li>
<li class="chapter" data-level="18.1.4" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#discrete-cox-regression"><i class="fa fa-check"></i><b>18.1.4</b> Discrete Cox Regression</a></li>
<li class="chapter" data-level="18.1.5" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#discrete-time-regression-models"><i class="fa fa-check"></i><b>18.1.5</b> Discrete-Time Regression Models</a></li>
<li class="chapter" data-level="18.1.6" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#the-glm-framework-and-person-period-data"><i class="fa fa-check"></i><b>18.1.6</b> The GLM Framework and Person-Period Data</a></li>
<li class="chapter" data-level="18.1.7" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#discrete-time-survival-analysis-in-r"><i class="fa fa-check"></i><b>18.1.7</b> Discrete-Time Survival Analysis in R</a></li>
<li class="chapter" data-level="18.1.8" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#discrete-cause-specific-hazards-model"><i class="fa fa-check"></i><b>18.1.8</b> Discrete Cause-Specific Hazards Model</a></li>
<li class="chapter" data-level="18.1.9" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#discrete-subdistribution-hazard-model"><i class="fa fa-check"></i><b>18.1.9</b> Discrete Subdistribution Hazard Model</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#time-dependent-covariates"><i class="fa fa-check"></i><b>18.2</b> Time-dependent covariates</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#introduction-14"><i class="fa fa-check"></i><b>18.2.1</b> Introduction</a></li>
<li class="chapter" data-level="18.2.2" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#programming-method"><i class="fa fa-check"></i><b>18.2.2</b> Programming method</a></li>
<li class="chapter" data-level="18.2.3" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#counting-method"><i class="fa fa-check"></i><b>18.2.3</b> Counting method</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#additive-hazards-regression-model"><i class="fa fa-check"></i><b>18.3</b> Additive Hazards Regression Model</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#preview"><i class="fa fa-check"></i><b>18.3.1</b> Preview</a></li>
<li class="chapter" data-level="18.3.2" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#lin-and-yings-additive-hazards-model"><i class="fa fa-check"></i><b>18.3.2</b> Lin and Ying’s additive hazards model</a></li>
<li class="chapter" data-level="18.3.3" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#aalens-additive-hazards-model"><i class="fa fa-check"></i><b>18.3.3</b> Aalen’s Additive Hazards Model</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#competing-risk"><i class="fa fa-check"></i><b>18.4</b> Competing Risk</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#introduction-15"><i class="fa fa-check"></i><b>18.4.1</b> Introduction</a></li>
<li class="chapter" data-level="18.4.2" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#type-specific-hazards"><i class="fa fa-check"></i><b>18.4.2</b> Type-specific hazards</a></li>
<li class="chapter" data-level="18.4.3" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#covariate-effects-via-cox-model"><i class="fa fa-check"></i><b>18.4.3</b> Covariate effects via COX model</a></li>
<li class="chapter" data-level="18.4.4" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#cumulative-incidence-function-cif"><i class="fa fa-check"></i><b>18.4.4</b> Cumulative incidence function (CIF)</a></li>
<li class="chapter" data-level="18.4.5" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#subdistribution-hazard-function"><i class="fa fa-check"></i><b>18.4.5</b> Subdistribution hazard function</a></li>
<li class="chapter" data-level="18.4.6" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#cause-specific-hazard-function"><i class="fa fa-check"></i><b>18.4.6</b> Cause-specific hazard function</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#clustered-events"><i class="fa fa-check"></i><b>18.5</b> Clustered Events</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#introduction-18"><i class="fa fa-check"></i><b>18.5.1</b> Introduction</a></li>
<li class="chapter" data-level="18.5.2" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#proportional-hazards-model-which-adopts-a-marginal"><i class="fa fa-check"></i><b>18.5.2</b> Proportional hazards model which adopts a marginal</a></li>
<li class="chapter" data-level="18.5.3" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#frailty-model"><i class="fa fa-check"></i><b>18.5.3</b> Frailty model</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#non-proportional-hazards-in-survival-analysis"><i class="fa fa-check"></i><b>18.6</b> Non-proportional hazards in Survival Analysis</a>
<ul>
<li class="chapter" data-level="18.6.1" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#weighted-piecewise-log-rank-test"><i class="fa fa-check"></i><b>18.6.1</b> Weighted Piecewise Log-Rank Test</a></li>
<li class="chapter" data-level="18.6.2" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#restricted-mean-survival-time-difference"><i class="fa fa-check"></i><b>18.6.2</b> Restricted Mean Survival Time Difference</a></li>
<li class="chapter" data-level="18.6.3" data-path="advanced-survival-analysis.html"><a href="advanced-survival-analysis.html#maximum-combination-test"><i class="fa fa-check"></i><b>18.6.3</b> Maximum Combination Test</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="mixed-model.html"><a href="mixed-model.html"><i class="fa fa-check"></i><b>19</b> Mixed Model</a>
<ul>
<li class="chapter" data-level="19.1" data-path="mixed-model.html"><a href="mixed-model.html#introduction-19"><i class="fa fa-check"></i><b>19.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="19.1.1" data-path="mixed-model.html"><a href="mixed-model.html#correlated-response-data"><i class="fa fa-check"></i><b>19.1.1</b> Correlated response data</a></li>
<li class="chapter" data-level="19.1.2" data-path="mixed-model.html"><a href="mixed-model.html#hierarchical-and-marginal-model"><i class="fa fa-check"></i><b>19.1.2</b> Hierarchical and Marginal model</a></li>
<li class="chapter" data-level="19.1.3" data-path="mixed-model.html"><a href="mixed-model.html#mle-in-marginal-model"><i class="fa fa-check"></i><b>19.1.3</b> MLE in Marginal model</a></li>
<li class="chapter" data-level="19.1.4" data-path="mixed-model.html"><a href="mixed-model.html#reml-in-marginal-model"><i class="fa fa-check"></i><b>19.1.4</b> REML in Marginal model</a></li>
<li class="chapter" data-level="19.1.5" data-path="mixed-model.html"><a href="mixed-model.html#variance-correction"><i class="fa fa-check"></i><b>19.1.5</b> Variance correction</a></li>
<li class="chapter" data-level="19.1.6" data-path="mixed-model.html"><a href="mixed-model.html#hypothesis-tests"><i class="fa fa-check"></i><b>19.1.6</b> Hypothesis tests</a></li>
<li class="chapter" data-level="19.1.7" data-path="mixed-model.html"><a href="mixed-model.html#residual-structure"><i class="fa fa-check"></i><b>19.1.7</b> Residual Structure</a></li>
<li class="chapter" data-level="19.1.8" data-path="mixed-model.html"><a href="mixed-model.html#converge-problems"><i class="fa fa-check"></i><b>19.1.8</b> Converge problems</a></li>
<li class="chapter" data-level="19.1.9" data-path="mixed-model.html"><a href="mixed-model.html#preface-of-linear-mixed-model"><i class="fa fa-check"></i><b>19.1.9</b> Preface of linear mixed model</a></li>
</ul></li>
<li class="chapter" data-level="19.2" data-path="mixed-model.html"><a href="mixed-model.html#random-slope-and-intercept-model"><i class="fa fa-check"></i><b>19.2</b> Random Slope and Intercept Model</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="mixed-model.html"><a href="mixed-model.html#model-introduction"><i class="fa fa-check"></i><b>19.2.1</b> Model Introduction</a></li>
<li class="chapter" data-level="19.2.2" data-path="mixed-model.html"><a href="mixed-model.html#sas-implementation-16"><i class="fa fa-check"></i><b>19.2.2</b> SAS Implementation</a></li>
<li class="chapter" data-level="19.2.3" data-path="mixed-model.html"><a href="mixed-model.html#r-implementation-17"><i class="fa fa-check"></i><b>19.2.3</b> R Implementation</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="mixed-model.html"><a href="mixed-model.html#random-slope-and-intercept-model-with-covariance-structure"><i class="fa fa-check"></i><b>19.3</b> Random Slope and Intercept Model with Covariance Structure</a>
<ul>
<li class="chapter" data-level="19.3.1" data-path="mixed-model.html"><a href="mixed-model.html#model-introduction-1"><i class="fa fa-check"></i><b>19.3.1</b> Model Introduction</a></li>
<li class="chapter" data-level="19.3.2" data-path="mixed-model.html"><a href="mixed-model.html#sas-implementation-17"><i class="fa fa-check"></i><b>19.3.2</b> SAS Implementation</a></li>
<li class="chapter" data-level="19.3.3" data-path="mixed-model.html"><a href="mixed-model.html#r-implementation-18"><i class="fa fa-check"></i><b>19.3.3</b> R Implementation</a></li>
</ul></li>
<li class="chapter" data-level="19.4" data-path="mixed-model.html"><a href="mixed-model.html#generalized-estimating-equation"><i class="fa fa-check"></i><b>19.4</b> Generalized Estimating Equation</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="mixed-model.html"><a href="mixed-model.html#introduction-of-gee"><i class="fa fa-check"></i><b>19.4.1</b> Introduction of GEE</a></li>
<li class="chapter" data-level="19.4.2" data-path="mixed-model.html"><a href="mixed-model.html#sas-implementation-18"><i class="fa fa-check"></i><b>19.4.2</b> SAS Implementation</a></li>
<li class="chapter" data-level="19.4.3" data-path="mixed-model.html"><a href="mixed-model.html#r-implementation-19"><i class="fa fa-check"></i><b>19.4.3</b> R Implementation</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="mixed-model.html"><a href="mixed-model.html#hierarchical-regression-model-for-normal-response"><i class="fa fa-check"></i><b>19.5</b> Hierarchical Regression Model for Normal Response</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="mixed-model.html"><a href="mixed-model.html#introduction-20"><i class="fa fa-check"></i><b>19.5.1</b> Introduction</a></li>
<li class="chapter" data-level="19.5.2" data-path="mixed-model.html"><a href="mixed-model.html#sas-implementation-19"><i class="fa fa-check"></i><b>19.5.2</b> SAS Implementation</a></li>
<li class="chapter" data-level="19.5.3" data-path="mixed-model.html"><a href="mixed-model.html#r-implementation-20"><i class="fa fa-check"></i><b>19.5.3</b> R Implementation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="generalized-additive-models.html"><a href="generalized-additive-models.html"><i class="fa fa-check"></i><b>20</b> Generalized Additive Models</a>
<ul>
<li class="chapter" data-level="20.1" data-path="generalized-additive-models.html"><a href="generalized-additive-models.html#generalized-linear-mixed-model"><i class="fa fa-check"></i><b>20.1</b> Generalized Linear (Mixed) Model</a></li>
<li class="chapter" data-level="20.2" data-path="generalized-additive-models.html"><a href="generalized-additive-models.html#generalized-linear-model"><i class="fa fa-check"></i><b>20.2</b> Generalized Linear Model</a></li>
<li class="chapter" data-level="20.3" data-path="generalized-additive-models.html"><a href="generalized-additive-models.html#generalized-additive-models-1"><i class="fa fa-check"></i><b>20.3</b> Generalized Additive Models</a>
<ul>
<li class="chapter" data-level="20.3.1" data-path="generalized-additive-models.html"><a href="generalized-additive-models.html#concept"><i class="fa fa-check"></i><b>20.3.1</b> Concept</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="missing-data.html"><a href="missing-data.html"><i class="fa fa-check"></i><b>21</b> Missing Data</a>
<ul>
<li class="chapter" data-level="21.1" data-path="missing-data.html"><a href="missing-data.html#missing-mechanisms"><i class="fa fa-check"></i><b>21.1</b> Missing mechanisms</a></li>
<li class="chapter" data-level="21.2" data-path="missing-data.html"><a href="missing-data.html#compatibility-and-congeniality"><i class="fa fa-check"></i><b>21.2</b> Compatibility and Congeniality</a></li>
<li class="chapter" data-level="21.3" data-path="missing-data.html"><a href="missing-data.html#single-imputation"><i class="fa fa-check"></i><b>21.3</b> Single imputation</a></li>
<li class="chapter" data-level="21.4" data-path="missing-data.html"><a href="missing-data.html#multiple-imputation"><i class="fa fa-check"></i><b>21.4</b> Multiple imputation</a>
<ul>
<li class="chapter" data-level="21.4.1" data-path="missing-data.html"><a href="missing-data.html#joint-modeling-jm"><i class="fa fa-check"></i><b>21.4.1</b> Joint Modeling (JM)</a></li>
<li class="chapter" data-level="21.4.2" data-path="missing-data.html"><a href="missing-data.html#multiple-imputation-by-fully-conditional-specification-mice"><i class="fa fa-check"></i><b>21.4.2</b> Multiple Imputation by Fully Conditional Specification (MICE)</a></li>
</ul></li>
<li class="chapter" data-level="21.5" data-path="missing-data.html"><a href="missing-data.html#bayesian-linear-regression"><i class="fa fa-check"></i><b>21.5</b> Bayesian linear regression</a></li>
<li class="chapter" data-level="21.6" data-path="missing-data.html"><a href="missing-data.html#predictive-mean-matching"><i class="fa fa-check"></i><b>21.6</b> Predictive Mean Matching</a></li>
<li class="chapter" data-level="21.7" data-path="missing-data.html"><a href="missing-data.html#mice-using-random-forest"><i class="fa fa-check"></i><b>21.7</b> MICE using random forest</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="meta-analysis.html"><a href="meta-analysis.html"><i class="fa fa-check"></i><b>22</b> Meta Analysis</a>
<ul>
<li class="chapter" data-level="22.1" data-path="meta-analysis.html"><a href="meta-analysis.html#introduction-21"><i class="fa fa-check"></i><b>22.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="22.1.1" data-path="meta-analysis.html"><a href="meta-analysis.html#meta-analysis-for-different-data"><i class="fa fa-check"></i><b>22.1.1</b> meta-analysis for different data</a></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="meta-analysis.html"><a href="meta-analysis.html#fixed-effect-model-for-continuous-outcomes"><i class="fa fa-check"></i><b>22.2</b> Fixed Effect Model for Continuous Outcomes</a>
<ul>
<li class="chapter" data-level="22.2.1" data-path="meta-analysis.html"><a href="meta-analysis.html#effect-measures"><i class="fa fa-check"></i><b>22.2.1</b> Effect Measures</a></li>
<li class="chapter" data-level="22.2.2" data-path="meta-analysis.html"><a href="meta-analysis.html#standardized-mean-difference"><i class="fa fa-check"></i><b>22.2.2</b> Standardized Mean Difference</a></li>
<li class="chapter" data-level="22.2.3" data-path="meta-analysis.html"><a href="meta-analysis.html#inverse-variance-weighted-average-method"><i class="fa fa-check"></i><b>22.2.3</b> Inverse variance-weighted average method</a></li>
<li class="chapter" data-level="22.2.4" data-path="meta-analysis.html"><a href="meta-analysis.html#generic-inverse-variance-meta-analysis-metagen"><i class="fa fa-check"></i><b>22.2.4</b> Generic inverse variance meta-analysis <code>metagen</code></a></li>
<li class="chapter" data-level="22.2.5" data-path="meta-analysis.html"><a href="meta-analysis.html#weighted-sum-of-z-scores"><i class="fa fa-check"></i><b>22.2.5</b> Weighted Sum of Z-Scores</a></li>
</ul></li>
<li class="chapter" data-level="22.3" data-path="meta-analysis.html"><a href="meta-analysis.html#random-effects-model-for-continuous-outcomes"><i class="fa fa-check"></i><b>22.3</b> Random Effects Model for Continuous Outcomes</a>
<ul>
<li class="chapter" data-level="22.3.1" data-path="meta-analysis.html"><a href="meta-analysis.html#introduction-22"><i class="fa fa-check"></i><b>22.3.1</b> Introduction</a></li>
<li class="chapter" data-level="22.3.2" data-path="meta-analysis.html"><a href="meta-analysis.html#implementation"><i class="fa fa-check"></i><b>22.3.2</b> Implementation</a></li>
<li class="chapter" data-level="22.3.3" data-path="meta-analysis.html"><a href="meta-analysis.html#hartung-knapp-adjustment"><i class="fa fa-check"></i><b>22.3.3</b> Hartung-Knapp Adjustment</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="meta-analysis.html"><a href="meta-analysis.html#meta-regression"><i class="fa fa-check"></i><b>22.4</b> Meta-Regression</a></li>
<li class="chapter" data-level="22.5" data-path="meta-analysis.html"><a href="meta-analysis.html#meta-analysis-with-binary-outcomes"><i class="fa fa-check"></i><b>22.5</b> Meta-Analysis with Binary Outcomes</a>
<ul>
<li class="chapter" data-level="22.5.1" data-path="meta-analysis.html"><a href="meta-analysis.html#effect-measures-1"><i class="fa fa-check"></i><b>22.5.1</b> Effect Measures</a></li>
<li class="chapter" data-level="22.5.2" data-path="meta-analysis.html"><a href="meta-analysis.html#implementation-1"><i class="fa fa-check"></i><b>22.5.2</b> Implementation</a></li>
<li class="chapter" data-level="22.5.3" data-path="meta-analysis.html"><a href="meta-analysis.html#estimation-in-sparse-data---continuity-correction"><i class="fa fa-check"></i><b>22.5.3</b> Estimation in Sparse Data - Continuity correction</a></li>
<li class="chapter" data-level="22.5.4" data-path="meta-analysis.html"><a href="meta-analysis.html#peto-odds-ratio"><i class="fa fa-check"></i><b>22.5.4</b> Peto Odds Ratio</a></li>
<li class="chapter" data-level="22.5.5" data-path="meta-analysis.html"><a href="meta-analysis.html#fixed-effect-model-inverse-variance-method"><i class="fa fa-check"></i><b>22.5.5</b> Fixed Effect Model: Inverse Variance Method</a></li>
<li class="chapter" data-level="22.5.6" data-path="meta-analysis.html"><a href="meta-analysis.html#fixed-effect-model-mantelhaenszel-method"><i class="fa fa-check"></i><b>22.5.6</b> Fixed Effect Model: Mantel–Haenszel Method</a></li>
<li class="chapter" data-level="22.5.7" data-path="meta-analysis.html"><a href="meta-analysis.html#random-effects-model"><i class="fa fa-check"></i><b>22.5.7</b> Random Effects Model</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="time-series-analysis.html"><a href="time-series-analysis.html"><i class="fa fa-check"></i><b>23</b> Time Series Analysis</a>
<ul>
<li class="chapter" data-level="23.1" data-path="time-series-analysis.html"><a href="time-series-analysis.html#fundational-concepts"><i class="fa fa-check"></i><b>23.1</b> Fundational Concepts</a>
<ul>
<li class="chapter" data-level="23.1.1" data-path="time-series-analysis.html"><a href="time-series-analysis.html#means-variances-and-covariances"><i class="fa fa-check"></i><b>23.1.1</b> Means, Variances, and Covariances</a></li>
<li class="chapter" data-level="23.1.2" data-path="time-series-analysis.html"><a href="time-series-analysis.html#properties-of-covariance"><i class="fa fa-check"></i><b>23.1.2</b> Properties of covariance</a></li>
<li class="chapter" data-level="23.1.3" data-path="time-series-analysis.html"><a href="time-series-analysis.html#properties-of-expectation"><i class="fa fa-check"></i><b>23.1.3</b> Properties of Expectation</a></li>
<li class="chapter" data-level="23.1.4" data-path="time-series-analysis.html"><a href="time-series-analysis.html#properties-of-variance"><i class="fa fa-check"></i><b>23.1.4</b> Properties of Variance</a></li>
<li class="chapter" data-level="23.1.5" data-path="time-series-analysis.html"><a href="time-series-analysis.html#the-random-walk"><i class="fa fa-check"></i><b>23.1.5</b> The Random Walk</a></li>
<li class="chapter" data-level="23.1.6" data-path="time-series-analysis.html"><a href="time-series-analysis.html#a-moving-average"><i class="fa fa-check"></i><b>23.1.6</b> A Moving Average</a></li>
<li class="chapter" data-level="23.1.7" data-path="time-series-analysis.html"><a href="time-series-analysis.html#strictly-stationarity"><i class="fa fa-check"></i><b>23.1.7</b> Strictly Stationarity</a></li>
<li class="chapter" data-level="23.1.8" data-path="time-series-analysis.html"><a href="time-series-analysis.html#weakly-or-second-order-stationary"><i class="fa fa-check"></i><b>23.1.8</b> Weakly (or second-order) stationary</a></li>
<li class="chapter" data-level="23.1.9" data-path="time-series-analysis.html"><a href="time-series-analysis.html#white-noise"><i class="fa fa-check"></i><b>23.1.9</b> White Noise</a></li>
<li class="chapter" data-level="23.1.10" data-path="time-series-analysis.html"><a href="time-series-analysis.html#deterministic-versus-stochastic-trends"><i class="fa fa-check"></i><b>23.1.10</b> Deterministic Versus Stochastic Trends</a></li>
</ul></li>
<li class="chapter" data-level="23.2" data-path="time-series-analysis.html"><a href="time-series-analysis.html#arma-models"><i class="fa fa-check"></i><b>23.2</b> ARMA models</a>
<ul>
<li class="chapter" data-level="23.2.1" data-path="time-series-analysis.html"><a href="time-series-analysis.html#general-linear-processes"><i class="fa fa-check"></i><b>23.2.1</b> General Linear Processes</a></li>
<li class="chapter" data-level="23.2.2" data-path="time-series-analysis.html"><a href="time-series-analysis.html#moving-average-processes"><i class="fa fa-check"></i><b>23.2.2</b> Moving Average Processes</a></li>
<li class="chapter" data-level="23.2.3" data-path="time-series-analysis.html"><a href="time-series-analysis.html#autoregressive-processes"><i class="fa fa-check"></i><b>23.2.3</b> Autoregressive Processes</a></li>
<li class="chapter" data-level="23.2.4" data-path="time-series-analysis.html"><a href="time-series-analysis.html#the-mixed-autoregressive-moving-average-model"><i class="fa fa-check"></i><b>23.2.4</b> The Mixed Autoregressive Moving Average Model</a></li>
<li class="chapter" data-level="23.2.5" data-path="time-series-analysis.html"><a href="time-series-analysis.html#invertibility"><i class="fa fa-check"></i><b>23.2.5</b> Invertibility</a></li>
<li class="chapter" data-level="23.2.6" data-path="time-series-analysis.html"><a href="time-series-analysis.html#fit-the-ar-model-in-r"><i class="fa fa-check"></i><b>23.2.6</b> Fit the AR model in R</a></li>
<li class="chapter" data-level="23.2.7" data-path="time-series-analysis.html"><a href="time-series-analysis.html#fit-the-ma-model-in-r"><i class="fa fa-check"></i><b>23.2.7</b> Fit the MA model in R</a></li>
</ul></li>
<li class="chapter" data-level="23.3" data-path="time-series-analysis.html"><a href="time-series-analysis.html#arima-models"><i class="fa fa-check"></i><b>23.3</b> ARIMA Models</a>
<ul>
<li class="chapter" data-level="23.3.1" data-path="time-series-analysis.html"><a href="time-series-analysis.html#stationarity-through-differencing"><i class="fa fa-check"></i><b>23.3.1</b> Stationarity Through Differencing</a></li>
<li class="chapter" data-level="23.3.2" data-path="time-series-analysis.html"><a href="time-series-analysis.html#arima-integrated-autoregressive-moving-average-model"><i class="fa fa-check"></i><b>23.3.2</b> ARIMA: Integrated autoregressive moving average model</a></li>
<li class="chapter" data-level="23.3.3" data-path="time-series-analysis.html"><a href="time-series-analysis.html#ima11-model"><i class="fa fa-check"></i><b>23.3.3</b> IMA(1,1) Model</a></li>
<li class="chapter" data-level="23.3.4" data-path="time-series-analysis.html"><a href="time-series-analysis.html#ari11-model"><i class="fa fa-check"></i><b>23.3.4</b> ARI(1,1) Model</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="time-series-analysis.html"><a href="time-series-analysis.html#time-series-graphics"><i class="fa fa-check"></i><b>23.4</b> Time series graphics</a>
<ul>
<li class="chapter" data-level="23.4.1" data-path="time-series-analysis.html"><a href="time-series-analysis.html#time-plots"><i class="fa fa-check"></i><b>23.4.1</b> Time plots</a></li>
<li class="chapter" data-level="23.4.2" data-path="time-series-analysis.html"><a href="time-series-analysis.html#seasonal-plots"><i class="fa fa-check"></i><b>23.4.2</b> Seasonal plots</a></li>
<li class="chapter" data-level="23.4.3" data-path="time-series-analysis.html"><a href="time-series-analysis.html#scatterplots-matrices"><i class="fa fa-check"></i><b>23.4.3</b> Scatterplots matrices</a></li>
<li class="chapter" data-level="23.4.4" data-path="time-series-analysis.html"><a href="time-series-analysis.html#lag-plots"><i class="fa fa-check"></i><b>23.4.4</b> Lag plots</a></li>
<li class="chapter" data-level="23.4.5" data-path="time-series-analysis.html"><a href="time-series-analysis.html#autocorrelation"><i class="fa fa-check"></i><b>23.4.5</b> Autocorrelation</a></li>
<li class="chapter" data-level="23.4.6" data-path="time-series-analysis.html"><a href="time-series-analysis.html#white-noise-1"><i class="fa fa-check"></i><b>23.4.6</b> White noise</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24" data-path="propensity-score.html"><a href="propensity-score.html"><i class="fa fa-check"></i><b>24</b> Propensity Score</a>
<ul>
<li class="chapter" data-level="24.1" data-path="propensity-score.html"><a href="propensity-score.html#ate-att"><i class="fa fa-check"></i><b>24.1</b> ATE &amp; ATT</a></li>
<li class="chapter" data-level="24.2" data-path="propensity-score.html"><a href="propensity-score.html#propensity-score-matching"><i class="fa fa-check"></i><b>24.2</b> Propensity Score Matching</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="clinic-trail.html"><a href="clinic-trail.html"><i class="fa fa-check"></i><b>25</b> Clinic Trail</a>
<ul>
<li class="chapter" data-level="25.1" data-path="clinic-trail.html"><a href="clinic-trail.html#regulation"><i class="fa fa-check"></i><b>25.1</b> Regulation</a></li>
<li class="chapter" data-level="25.2" data-path="clinic-trail.html"><a href="clinic-trail.html#randomization"><i class="fa fa-check"></i><b>25.2</b> Randomization</a>
<ul>
<li class="chapter" data-level="25.2.1" data-path="clinic-trail.html"><a href="clinic-trail.html#simple-randomization"><i class="fa fa-check"></i><b>25.2.1</b> Simple randomization</a></li>
<li class="chapter" data-level="25.2.2" data-path="clinic-trail.html"><a href="clinic-trail.html#block-randomization"><i class="fa fa-check"></i><b>25.2.2</b> Block randomization</a></li>
<li class="chapter" data-level="25.2.3" data-path="clinic-trail.html"><a href="clinic-trail.html#stratified-randomization"><i class="fa fa-check"></i><b>25.2.3</b> Stratified randomization</a></li>
</ul></li>
<li class="chapter" data-level="25.3" data-path="clinic-trail.html"><a href="clinic-trail.html#phase-i-trials-design"><i class="fa fa-check"></i><b>25.3</b> Phase I Trials Design</a>
<ul>
<li class="chapter" data-level="25.3.1" data-path="clinic-trail.html"><a href="clinic-trail.html#introduction-23"><i class="fa fa-check"></i><b>25.3.1</b> Introduction</a></li>
</ul></li>
<li class="chapter" data-level="25.4" data-path="clinic-trail.html"><a href="clinic-trail.html#phase-ii-trials-design"><i class="fa fa-check"></i><b>25.4</b> Phase II Trials Design</a>
<ul>
<li class="chapter" data-level="25.4.1" data-path="clinic-trail.html"><a href="clinic-trail.html#introduction-24"><i class="fa fa-check"></i><b>25.4.1</b> Introduction</a></li>
</ul></li>
<li class="chapter" data-level="25.5" data-path="clinic-trail.html"><a href="clinic-trail.html#pharmacokinetic-analysis-pk-package"><i class="fa fa-check"></i><b>25.5</b> Pharmacokinetic Analysis (PK package）</a>
<ul>
<li class="chapter" data-level="25.5.1" data-path="clinic-trail.html"><a href="clinic-trail.html#auc"><i class="fa fa-check"></i><b>25.5.1</b> AUC</a></li>
<li class="chapter" data-level="25.5.2" data-path="clinic-trail.html"><a href="clinic-trail.html#auc-in-complete-data-design"><i class="fa fa-check"></i><b>25.5.2</b> AUC in complete data design</a></li>
<li class="chapter" data-level="25.5.3" data-path="clinic-trail.html"><a href="clinic-trail.html#auc-in-repeated-complete-data-design"><i class="fa fa-check"></i><b>25.5.3</b> AUC in repeated complete data design</a></li>
<li class="chapter" data-level="25.5.4" data-path="clinic-trail.html"><a href="clinic-trail.html#bioequivalence-between-aucs"><i class="fa fa-check"></i><b>25.5.4</b> Bioequivalence between AUCs</a></li>
<li class="chapter" data-level="25.5.5" data-path="clinic-trail.html"><a href="clinic-trail.html#two-phase-half-life-estimation-by-biexponential-model"><i class="fa fa-check"></i><b>25.5.5</b> Two-phase half-life estimation by biexponential model</a></li>
<li class="chapter" data-level="25.5.6" data-path="clinic-trail.html"><a href="clinic-trail.html#two-phase-half-life-estimation-by-linear-fitting"><i class="fa fa-check"></i><b>25.5.6</b> Two-phase half-life estimation by linear fitting</a></li>
<li class="chapter" data-level="25.5.7" data-path="clinic-trail.html"><a href="clinic-trail.html#estimation-of-various-pk-parameters"><i class="fa fa-check"></i><b>25.5.7</b> Estimation of various PK parameters</a></li>
</ul></li>
<li class="chapter" data-level="25.6" data-path="clinic-trail.html"><a href="clinic-trail.html#phase-iii-trials-design"><i class="fa fa-check"></i><b>25.6</b> Phase III Trials Design</a>
<ul>
<li class="chapter" data-level="25.6.1" data-path="clinic-trail.html"><a href="clinic-trail.html#introduction-25"><i class="fa fa-check"></i><b>25.6.1</b> Introduction</a></li>
<li class="chapter" data-level="25.6.2" data-path="clinic-trail.html"><a href="clinic-trail.html#mediwound1"><i class="fa fa-check"></i><b>25.6.2</b> MEDIWOUND1</a></li>
</ul></li>
<li class="chapter" data-level="25.7" data-path="clinic-trail.html"><a href="clinic-trail.html#postmarketing-surveillance-pms"><i class="fa fa-check"></i><b>25.7</b> Postmarketing Surveillance (PMS)</a>
<ul>
<li class="chapter" data-level="25.7.1" data-path="clinic-trail.html"><a href="clinic-trail.html#phase-iv-and-pms"><i class="fa fa-check"></i><b>25.7.1</b> Phase IV and PMS</a></li>
</ul></li>
<li class="chapter" data-level="25.8" data-path="clinic-trail.html"><a href="clinic-trail.html#trials-design-phase-ii"><i class="fa fa-check"></i><b>25.8</b> Trials Design (Phase II)</a>
<ul>
<li class="chapter" data-level="25.8.1" data-path="clinic-trail.html"><a href="clinic-trail.html#flemings-two-stage-design"><i class="fa fa-check"></i><b>25.8.1</b> Fleming’s Two-Stage design</a></li>
<li class="chapter" data-level="25.8.2" data-path="clinic-trail.html"><a href="clinic-trail.html#simons-2-stage-design"><i class="fa fa-check"></i><b>25.8.2</b> Simon’s 2-Stage Design</a></li>
<li class="chapter" data-level="25.8.3" data-path="clinic-trail.html"><a href="clinic-trail.html#jonckheere-terptsra-jt-trend-test"><i class="fa fa-check"></i><b>25.8.3</b> Jonckheere-Terptsra (JT) trend test</a></li>
<li class="chapter" data-level="25.8.4" data-path="clinic-trail.html"><a href="clinic-trail.html#cochran-armitage-ca-trend-test"><i class="fa fa-check"></i><b>25.8.4</b> Cochran-Armitage (CA) trend test</a></li>
<li class="chapter" data-level="25.8.5" data-path="clinic-trail.html"><a href="clinic-trail.html#mcp-mod"><i class="fa fa-check"></i><b>25.8.5</b> MCP-Mod</a></li>
<li class="chapter" data-level="25.8.6" data-path="clinic-trail.html"><a href="clinic-trail.html#two-stage-phase-ii-design-for-response-and-toxicity"><i class="fa fa-check"></i><b>25.8.6</b> Two Stage Phase II Design for Response and Toxicity</a></li>
</ul></li>
<li class="chapter" data-level="25.9" data-path="clinic-trail.html"><a href="clinic-trail.html#trials-design-phase-iii"><i class="fa fa-check"></i><b>25.9</b> Trials Design (Phase III)</a>
<ul>
<li class="chapter" data-level="25.9.1" data-path="clinic-trail.html"><a href="clinic-trail.html#non-inferiority-and-equivalence-three-armed-trial"><i class="fa fa-check"></i><b>25.9.1</b> Non-inferiority and Equivalence Three-armed trial</a></li>
<li class="chapter" data-level="25.9.2" data-path="clinic-trail.html"><a href="clinic-trail.html#pigeot-method"><i class="fa fa-check"></i><b>25.9.2</b> Pigeot Method</a></li>
</ul></li>
<li class="chapter" data-level="25.10" data-path="clinic-trail.html"><a href="clinic-trail.html#sample-size-calculation-1"><i class="fa fa-check"></i><b>25.10</b> Sample Size Calculation</a>
<ul>
<li class="chapter" data-level="25.10.1" data-path="clinic-trail.html"><a href="clinic-trail.html#chi-square-test-1"><i class="fa fa-check"></i><b>25.10.1</b> <strong>Chi-Square Test</strong></a></li>
<li class="chapter" data-level="25.10.2" data-path="clinic-trail.html"><a href="clinic-trail.html#fishers-exact-test-2"><i class="fa fa-check"></i><b>25.10.2</b> Fisher’s Exact Test</a></li>
<li class="chapter" data-level="25.10.3" data-path="clinic-trail.html"><a href="clinic-trail.html#t-test"><i class="fa fa-check"></i><b>25.10.3</b> T-Test</a></li>
</ul></li>
<li class="chapter" data-level="25.11" data-path="clinic-trail.html"><a href="clinic-trail.html#data-analysis-methods-in-cip"><i class="fa fa-check"></i><b>25.11</b> Data Analysis Methods in CIP</a>
<ul>
<li class="chapter" data-level="25.11.1" data-path="clinic-trail.html"><a href="clinic-trail.html#statistical-analysis-plan"><i class="fa fa-check"></i><b>25.11.1</b> Statistical Analysis Plan</a></li>
<li class="chapter" data-level="25.11.2" data-path="clinic-trail.html"><a href="clinic-trail.html#general-considerations"><i class="fa fa-check"></i><b>25.11.2</b> General Considerations</a></li>
<li class="chapter" data-level="25.11.3" data-path="clinic-trail.html"><a href="clinic-trail.html#analysis-sets"><i class="fa fa-check"></i><b>25.11.3</b> Analysis Sets</a></li>
<li class="chapter" data-level="25.11.4" data-path="clinic-trail.html"><a href="clinic-trail.html#patient-disposition"><i class="fa fa-check"></i><b>25.11.4</b> Patient Disposition</a></li>
<li class="chapter" data-level="25.11.5" data-path="clinic-trail.html"><a href="clinic-trail.html#patient-characteristics"><i class="fa fa-check"></i><b>25.11.5</b> Patient Characteristics</a></li>
<li class="chapter" data-level="25.11.6" data-path="clinic-trail.html"><a href="clinic-trail.html#treatment-compliance"><i class="fa fa-check"></i><b>25.11.6</b> Treatment Compliance</a></li>
<li class="chapter" data-level="25.11.7" data-path="clinic-trail.html"><a href="clinic-trail.html#descriptiveexplorative-analysis"><i class="fa fa-check"></i><b>25.11.7</b> Descriptive/Explorative Analysis</a></li>
<li class="chapter" data-level="25.11.8" data-path="clinic-trail.html"><a href="clinic-trail.html#efficacy-survival-analyses"><i class="fa fa-check"></i><b>25.11.8</b> Efficacy Survival Analyses</a></li>
<li class="chapter" data-level="25.11.9" data-path="clinic-trail.html"><a href="clinic-trail.html#efficacy-anova-analyses"><i class="fa fa-check"></i><b>25.11.9</b> Efficacy ANOVA Analyses</a></li>
<li class="chapter" data-level="25.11.10" data-path="clinic-trail.html"><a href="clinic-trail.html#efficacy-glm-dose-finding-study"><i class="fa fa-check"></i><b>25.11.10</b> Efficacy GLM Dose-Finding Study</a></li>
<li class="chapter" data-level="25.11.11" data-path="clinic-trail.html"><a href="clinic-trail.html#safety-analyses"><i class="fa fa-check"></i><b>25.11.11</b> Safety Analyses</a></li>
<li class="chapter" data-level="25.11.12" data-path="clinic-trail.html"><a href="clinic-trail.html#pharmacokinetic-and-pharmacodynamic-analyses"><i class="fa fa-check"></i><b>25.11.12</b> Pharmacokinetic and Pharmacodynamic Analyses</a></li>
<li class="chapter" data-level="25.11.13" data-path="clinic-trail.html"><a href="clinic-trail.html#immunogenicity-data-analyses"><i class="fa fa-check"></i><b>25.11.13</b> Immunogenicity Data Analyses</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html"><i class="fa fa-check"></i><b>26</b> Group (Adaptive) Sequential Design</a>
<ul>
<li class="chapter" data-level="26.1" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#classical-designs-without-futility-stopping"><i class="fa fa-check"></i><b>26.1</b> Classical Designs without futility stopping</a>
<ul>
<li class="chapter" data-level="26.1.1" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#pocock-method"><i class="fa fa-check"></i><b>26.1.1</b> Pocock Method</a></li>
<li class="chapter" data-level="26.1.2" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#obrien-fleming-method"><i class="fa fa-check"></i><b>26.1.2</b> O’Brien &amp; Fleming Method</a></li>
<li class="chapter" data-level="26.1.3" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#wang-tsiatis-method"><i class="fa fa-check"></i><b>26.1.3</b> Wang &amp; Tsiatis Method</a></li>
<li class="chapter" data-level="26.1.4" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#rejection-bounds-and-local-p-values"><i class="fa fa-check"></i><b>26.1.4</b> Rejection bounds and Local p-values</a></li>
<li class="chapter" data-level="26.1.5" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#power-and-sample-size"><i class="fa fa-check"></i><b>26.1.5</b> Power and Sample size</a></li>
</ul></li>
<li class="chapter" data-level="26.2" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#classical-designs-with-binding-futility-stopping"><i class="fa fa-check"></i><b>26.2</b> Classical Designs with binding futility stopping</a>
<ul>
<li class="chapter" data-level="26.2.1" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#symmetric-designs"><i class="fa fa-check"></i><b>26.2.1</b> Symmetric designs</a></li>
<li class="chapter" data-level="26.2.2" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#one-sided-designs"><i class="fa fa-check"></i><b>26.2.2</b> One-Sided Designs</a></li>
</ul></li>
<li class="chapter" data-level="26.3" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#alpha-spending-function-approach"><i class="fa fa-check"></i><b>26.3</b> Alpha Spending Function Approach</a></li>
<li class="chapter" data-level="26.4" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#r-implementation-using-rpact"><i class="fa fa-check"></i><b>26.4</b> R Implementation using rpact</a>
<ul>
<li class="chapter" data-level="26.4.1" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#basic-functions"><i class="fa fa-check"></i><b>26.4.1</b> Basic Functions</a></li>
<li class="chapter" data-level="26.4.2" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#getdesigngroupsequential-defining-efficacy-boundaries"><i class="fa fa-check"></i><b>26.4.2</b> <code>getDesignGroupSequential</code> defining efficacy boundaries</a></li>
</ul></li>
<li class="chapter" data-level="26.5" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#sample-size"><i class="fa fa-check"></i><b>26.5</b> Sample Size</a>
<ul>
<li class="chapter" data-level="26.5.1" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#sample-sizes-for-different-types-of-endpoints-without-ia"><i class="fa fa-check"></i><b>26.5.1</b> Sample Sizes for Different Types of Endpoints without IA</a></li>
<li class="chapter" data-level="26.5.2" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#two-groups-continuous-endpoint-without-ia"><i class="fa fa-check"></i><b>26.5.2</b> Two groups continuous endpoint (without IA)</a></li>
<li class="chapter" data-level="26.5.3" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#two-groups-binary-endpoint-without-ia"><i class="fa fa-check"></i><b>26.5.3</b> Two groups binary endpoint (without IA)</a></li>
<li class="chapter" data-level="26.5.4" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#group-sequential-designs-for-conti-and-binary"><i class="fa fa-check"></i><b>26.5.4</b> Group-sequential designs for conti and binary</a></li>
<li class="chapter" data-level="26.5.5" data-path="group-adaptive-sequential-design.html"><a href="group-adaptive-sequential-design.html#survival-endpoint"><i class="fa fa-check"></i><b>26.5.5</b> Survival endpoint</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27" data-path="adaptive-designs.html"><a href="adaptive-designs.html"><i class="fa fa-check"></i><b>27</b> Adaptive designs</a>
<ul>
<li class="chapter" data-level="27.1" data-path="adaptive-designs.html"><a href="adaptive-designs.html#introduction-26"><i class="fa fa-check"></i><b>27.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="27.1.1" data-path="adaptive-designs.html"><a href="adaptive-designs.html#early-termination-due-to-futility"><i class="fa fa-check"></i><b>27.1.1</b> Early termination due to futility</a></li>
<li class="chapter" data-level="27.1.2" data-path="adaptive-designs.html"><a href="adaptive-designs.html#early-termination-due-to-efficacy"><i class="fa fa-check"></i><b>27.1.2</b> Early termination due to efficacy</a></li>
<li class="chapter" data-level="27.1.3" data-path="adaptive-designs.html"><a href="adaptive-designs.html#sample-size-reassessment"><i class="fa fa-check"></i><b>27.1.3</b> Sample size reassessment</a></li>
<li class="chapter" data-level="27.1.4" data-path="adaptive-designs.html"><a href="adaptive-designs.html#change-or-modification-of-the-primary-endpoint"><i class="fa fa-check"></i><b>27.1.4</b> Change or modification of the primary endpoint</a></li>
<li class="chapter" data-level="27.1.5" data-path="adaptive-designs.html"><a href="adaptive-designs.html#discontinuing-treatment-arms"><i class="fa fa-check"></i><b>27.1.5</b> Discontinuing treatment arms</a></li>
<li class="chapter" data-level="27.1.6" data-path="adaptive-designs.html"><a href="adaptive-designs.html#switching-between-superiority-and-non-inferiority"><i class="fa fa-check"></i><b>27.1.6</b> Switching between superiority and non-inferiority</a></li>
<li class="chapter" data-level="27.1.7" data-path="adaptive-designs.html"><a href="adaptive-designs.html#selection-of-the-patient-population"><i class="fa fa-check"></i><b>27.1.7</b> Selection of the patient population</a></li>
<li class="chapter" data-level="27.1.8" data-path="adaptive-designs.html"><a href="adaptive-designs.html#methods"><i class="fa fa-check"></i><b>27.1.8</b> Methods</a></li>
</ul></li>
<li class="chapter" data-level="27.2" data-path="adaptive-designs.html"><a href="adaptive-designs.html#general-theory"><i class="fa fa-check"></i><b>27.2</b> General Theory</a>
<ul>
<li class="chapter" data-level="27.2.1" data-path="adaptive-designs.html"><a href="adaptive-designs.html#stopping-boundary"><i class="fa fa-check"></i><b>27.2.1</b> Stopping Boundary</a></li>
<li class="chapter" data-level="27.2.2" data-path="adaptive-designs.html"><a href="adaptive-designs.html#power-and-adjusted-p-value"><i class="fa fa-check"></i><b>27.2.2</b> Power and Adjusted p-value</a></li>
<li class="chapter" data-level="27.2.3" data-path="adaptive-designs.html"><a href="adaptive-designs.html#stopping-probabilities-design-evaluation"><i class="fa fa-check"></i><b>27.2.3</b> Stopping Probabilities (Design Evaluation)</a></li>
<li class="chapter" data-level="27.2.4" data-path="adaptive-designs.html"><a href="adaptive-designs.html#expected-duration-of-an-adaptive-trial-design-evaluation"><i class="fa fa-check"></i><b>27.2.4</b> Expected Duration of an Adaptive Trial (Design Evaluation)</a></li>
<li class="chapter" data-level="27.2.5" data-path="adaptive-designs.html"><a href="adaptive-designs.html#expected-sample-sizes-design-evaluation"><i class="fa fa-check"></i><b>27.2.5</b> Expected Sample Sizes (Design Evaluation)</a></li>
<li class="chapter" data-level="27.2.6" data-path="adaptive-designs.html"><a href="adaptive-designs.html#conditional-power-and-futility-index"><i class="fa fa-check"></i><b>27.2.6</b> Conditional Power and futility index</a></li>
</ul></li>
<li class="chapter" data-level="27.3" data-path="adaptive-designs.html"><a href="adaptive-designs.html#method-with-direct-combination-of-p-values"><i class="fa fa-check"></i><b>27.3</b> Method with Direct Combination of p-values</a>
<ul>
<li class="chapter" data-level="27.3.1" data-path="adaptive-designs.html"><a href="adaptive-designs.html#method-based-on-individual-p-values"><i class="fa fa-check"></i><b>27.3.1</b> Method Based on Individual p-values</a></li>
<li class="chapter" data-level="27.3.2" data-path="adaptive-designs.html"><a href="adaptive-designs.html#method-based-on-the-sum-of-p-values"><i class="fa fa-check"></i><b>27.3.2</b> Method Based on the Sum of p-values</a></li>
<li class="chapter" data-level="27.3.3" data-path="adaptive-designs.html"><a href="adaptive-designs.html#method-with-product-of-p-values"><i class="fa fa-check"></i><b>27.3.3</b> Method with Product of p-values</a></li>
<li class="chapter" data-level="27.3.4" data-path="adaptive-designs.html"><a href="adaptive-designs.html#fishers-product-test-combination-tests"><i class="fa fa-check"></i><b>27.3.4</b> Fisher’s product test (Combination Tests)</a></li>
<li class="chapter" data-level="27.3.5" data-path="adaptive-designs.html"><a href="adaptive-designs.html#fishers-product-test-with-early-rej.-and-acceptance"><i class="fa fa-check"></i><b>27.3.5</b> Fisher’s product test with early rej. and acceptance</a></li>
</ul></li>
<li class="chapter" data-level="27.4" data-path="adaptive-designs.html"><a href="adaptive-designs.html#inverse-normal-combination-function"><i class="fa fa-check"></i><b>27.4</b> Inverse normal combination function</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="cdisc-1.html"><a href="cdisc-1.html"><i class="fa fa-check"></i><b>28</b> CDISC</a>
<ul>
<li class="chapter" data-level="28.1" data-path="cdisc-1.html"><a href="cdisc-1.html#study-data-tabulation-model-sdtm"><i class="fa fa-check"></i><b>28.1</b> Study Data Tabulation Model (SDTM)</a>
<ul>
<li class="chapter" data-level="28.1.1" data-path="cdisc-1.html"><a href="cdisc-1.html#introduction-27"><i class="fa fa-check"></i><b>28.1.1</b> Introduction</a></li>
<li class="chapter" data-level="28.1.2" data-path="cdisc-1.html"><a href="cdisc-1.html#model-concepts-and-terms"><i class="fa fa-check"></i><b>28.1.2</b> Model Concepts and Terms</a></li>
<li class="chapter" data-level="28.1.3" data-path="cdisc-1.html"><a href="cdisc-1.html#sdtm-domains"><i class="fa fa-check"></i><b>28.1.3</b> SDTM Domains</a></li>
<li class="chapter" data-level="28.1.4" data-path="cdisc-1.html"><a href="cdisc-1.html#sdtm-define.xml"><i class="fa fa-check"></i><b>28.1.4</b> SDTM Define.xml</a></li>
<li class="chapter" data-level="28.1.5" data-path="cdisc-1.html"><a href="cdisc-1.html#extract-transform-load-etl"><i class="fa fa-check"></i><b>28.1.5</b> Extract-Transform-Load (ETL)</a></li>
<li class="chapter" data-level="28.1.6" data-path="cdisc-1.html"><a href="cdisc-1.html#example-building-dm-sdtm-datasets"><i class="fa fa-check"></i><b>28.1.6</b> Example Building DM SDTM Datasets</a></li>
</ul></li>
<li class="chapter" data-level="28.2" data-path="cdisc-1.html"><a href="cdisc-1.html#sdtm-sas-macro"><i class="fa fa-check"></i><b>28.2</b> SDTM SAS Macro</a>
<ul>
<li class="chapter" data-level="28.2.1" data-path="cdisc-1.html"><a href="cdisc-1.html#make_define"><i class="fa fa-check"></i><b>28.2.1</b> ％make_define</a></li>
<li class="chapter" data-level="28.2.2" data-path="cdisc-1.html"><a href="cdisc-1.html#make_codelist_formats"><i class="fa fa-check"></i><b>28.2.2</b> %make_codelist_formats</a></li>
<li class="chapter" data-level="28.2.3" data-path="cdisc-1.html"><a href="cdisc-1.html#make_empty_dataset"><i class="fa fa-check"></i><b>28.2.3</b> %make_empty_dataset</a></li>
<li class="chapter" data-level="28.2.4" data-path="cdisc-1.html"><a href="cdisc-1.html#make_dtc_date"><i class="fa fa-check"></i><b>28.2.4</b> %make_dtc_date</a></li>
<li class="chapter" data-level="28.2.5" data-path="cdisc-1.html"><a href="cdisc-1.html#make_sdtm_dy"><i class="fa fa-check"></i><b>28.2.5</b> %make_sdtm_dy</a></li>
<li class="chapter" data-level="28.2.6" data-path="cdisc-1.html"><a href="cdisc-1.html#make_sort_order"><i class="fa fa-check"></i><b>28.2.6</b> %make_sort_order</a></li>
<li class="chapter" data-level="28.2.7" data-path="cdisc-1.html"><a href="cdisc-1.html#create_stdm_domains"><i class="fa fa-check"></i><b>28.2.7</b> %Create_STDM_Domains</a></li>
<li class="chapter" data-level="28.2.8" data-path="cdisc-1.html"><a href="cdisc-1.html#detectduplicates"><i class="fa fa-check"></i><b>28.2.8</b> %DetectDuplicates</a></li>
<li class="chapter" data-level="28.2.9" data-path="cdisc-1.html"><a href="cdisc-1.html#changedvarnum"><i class="fa fa-check"></i><b>28.2.9</b> %ChangedVarNum</a></li>
<li class="chapter" data-level="28.2.10" data-path="cdisc-1.html"><a href="cdisc-1.html#changedlabels"><i class="fa fa-check"></i><b>28.2.10</b> %ChangedLabels</a></li>
</ul></li>
<li class="chapter" data-level="28.3" data-path="cdisc-1.html"><a href="cdisc-1.html#analysis-data-model-adam"><i class="fa fa-check"></i><b>28.3</b> Analysis Data Model (ADaM)</a>
<ul>
<li class="chapter" data-level="28.3.1" data-path="cdisc-1.html"><a href="cdisc-1.html#compare-with-sdtm"><i class="fa fa-check"></i><b>28.3.1</b> Compare with SDTM</a></li>
<li class="chapter" data-level="28.3.2" data-path="cdisc-1.html"><a href="cdisc-1.html#types-of-adam-metadata"><i class="fa fa-check"></i><b>28.3.2</b> 4 types of ADaM metadata</a></li>
<li class="chapter" data-level="28.3.3" data-path="cdisc-1.html"><a href="cdisc-1.html#subject-level-analysis-dataset-adsl"><i class="fa fa-check"></i><b>28.3.3</b> Subject-Level Analysis Dataset (ADSL)</a></li>
<li class="chapter" data-level="28.3.4" data-path="cdisc-1.html"><a href="cdisc-1.html#the-basic-data-structure-bds"><i class="fa fa-check"></i><b>28.3.4</b> The Basic Data Structure (BDS)</a></li>
<li class="chapter" data-level="28.3.5" data-path="cdisc-1.html"><a href="cdisc-1.html#example-building-adsl-datasets"><i class="fa fa-check"></i><b>28.3.5</b> Example Building ADSL Datasets</a></li>
<li class="chapter" data-level="28.3.6" data-path="cdisc-1.html"><a href="cdisc-1.html#example-building-adam-basic-data-structure-bds-datasets"><i class="fa fa-check"></i><b>28.3.6</b> Example Building ADaM Basic Data Structure (BDS) Datasets</a></li>
<li class="chapter" data-level="28.3.7" data-path="cdisc-1.html"><a href="cdisc-1.html#example-building-adae-adverse-event-analysis-datasets-datasets"><i class="fa fa-check"></i><b>28.3.7</b> Example Building ADAE – Adverse Event Analysis Datasets Datasets</a></li>
</ul></li>
<li class="chapter" data-level="28.4" data-path="cdisc-1.html"><a href="cdisc-1.html#adam-sas-macro"><i class="fa fa-check"></i><b>28.4</b> ADaM SAS Macro</a>
<ul>
<li class="chapter" data-level="28.4.1" data-path="cdisc-1.html"><a href="cdisc-1.html#make_define-1"><i class="fa fa-check"></i><b>28.4.1</b> %make_define</a></li>
<li class="chapter" data-level="28.4.2" data-path="cdisc-1.html"><a href="cdisc-1.html#create_adam_dataset"><i class="fa fa-check"></i><b>28.4.2</b> %create_adam_dataset</a></li>
<li class="chapter" data-level="28.4.3" data-path="cdisc-1.html"><a href="cdisc-1.html#make_empty_dataset-1"><i class="fa fa-check"></i><b>28.4.3</b> %make_empty_dataset</a></li>
<li class="chapter" data-level="28.4.4" data-path="cdisc-1.html"><a href="cdisc-1.html#dtc2dt"><i class="fa fa-check"></i><b>28.4.4</b> %DTC2DT</a></li>
<li class="chapter" data-level="28.4.5" data-path="cdisc-1.html"><a href="cdisc-1.html#imputed_date"><i class="fa fa-check"></i><b>28.4.5</b> %imputed_date</a></li>
<li class="chapter" data-level="28.4.6" data-path="cdisc-1.html"><a href="cdisc-1.html#mergsupp"><i class="fa fa-check"></i><b>28.4.6</b> %mergsupp</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="29" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html"><i class="fa fa-check"></i><b>29</b> Regularization Penalized Regression</a>
<ul>
<li class="chapter" data-level="29.1" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html#introduction-28"><i class="fa fa-check"></i><b>29.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="29.1.1" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html#motivation"><i class="fa fa-check"></i><b>29.1.1</b> Motivation</a></li>
<li class="chapter" data-level="29.1.2" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html#data-preparation"><i class="fa fa-check"></i><b>29.1.2</b> Data preparation</a></li>
<li class="chapter" data-level="29.1.3" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html#best-subset-regression"><i class="fa fa-check"></i><b>29.1.3</b> Best subset regression</a></li>
</ul></li>
<li class="chapter" data-level="29.2" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html#ridge-regression-1"><i class="fa fa-check"></i><b>29.2</b> Ridge Regression</a>
<ul>
<li class="chapter" data-level="29.2.1" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html#modeling"><i class="fa fa-check"></i><b>29.2.1</b> Modeling</a></li>
</ul></li>
<li class="chapter" data-level="29.3" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html#lasso-regression"><i class="fa fa-check"></i><b>29.3</b> Lasso Regression</a>
<ul>
<li class="chapter" data-level="29.3.1" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html#modelling"><i class="fa fa-check"></i><b>29.3.1</b> Modelling</a></li>
<li class="chapter" data-level="29.3.2" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html#glmnet-cross-validation"><i class="fa fa-check"></i><b>29.3.2</b> glmnet cross validation</a></li>
</ul></li>
<li class="chapter" data-level="29.4" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html#elasticnet"><i class="fa fa-check"></i><b>29.4</b> ElasticNet</a>
<ul>
<li class="chapter" data-level="29.4.1" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html#modelling-1"><i class="fa fa-check"></i><b>29.4.1</b> Modelling</a></li>
<li class="chapter" data-level="29.4.2" data-path="regularization-penalized-regression.html"><a href="regularization-penalized-regression.html#classification"><i class="fa fa-check"></i><b>29.4.2</b> Classification</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="30" data-path="bayesian-theory.html"><a href="bayesian-theory.html"><i class="fa fa-check"></i><b>30</b> Bayesian Theory</a>
<ul>
<li class="chapter" data-level="30.1" data-path="bayesian-theory.html"><a href="bayesian-theory.html#introduction-of-bayesian"><i class="fa fa-check"></i><b>30.1</b> Introduction of Bayesian</a>
<ul>
<li class="chapter" data-level="30.1.1" data-path="bayesian-theory.html"><a href="bayesian-theory.html#frequency-and-bayesian"><i class="fa fa-check"></i><b>30.1.1</b> Frequency and Bayesian</a></li>
<li class="chapter" data-level="30.1.2" data-path="bayesian-theory.html"><a href="bayesian-theory.html#bayes-rule"><i class="fa fa-check"></i><b>30.1.2</b> Bayes’ Rule</a></li>
<li class="chapter" data-level="30.1.3" data-path="bayesian-theory.html"><a href="bayesian-theory.html#bootstrap"><i class="fa fa-check"></i><b>30.1.3</b> Bootstrap</a></li>
<li class="chapter" data-level="30.1.4" data-path="bayesian-theory.html"><a href="bayesian-theory.html#posterior-distribution"><i class="fa fa-check"></i><b>30.1.4</b> Posterior Distribution</a></li>
<li class="chapter" data-level="30.1.5" data-path="bayesian-theory.html"><a href="bayesian-theory.html#bayesian-credible-intervals"><i class="fa fa-check"></i><b>30.1.5</b> Bayesian Credible Intervals</a></li>
</ul></li>
<li class="chapter" data-level="30.2" data-path="bayesian-theory.html"><a href="bayesian-theory.html#parameter-estimates-1"><i class="fa fa-check"></i><b>30.2</b> Parameter estimates</a>
<ul>
<li class="chapter" data-level="30.2.1" data-path="bayesian-theory.html"><a href="bayesian-theory.html#maximum-likelihood-estimation"><i class="fa fa-check"></i><b>30.2.1</b> Maximum likelihood estimation</a></li>
<li class="chapter" data-level="30.2.2" data-path="bayesian-theory.html"><a href="bayesian-theory.html#maximum-a-posteriori-estimation"><i class="fa fa-check"></i><b>30.2.2</b> Maximum a posteriori estimation</a></li>
<li class="chapter" data-level="30.2.3" data-path="bayesian-theory.html"><a href="bayesian-theory.html#bayesian-model"><i class="fa fa-check"></i><b>30.2.3</b> Bayesian Model</a></li>
</ul></li>
<li class="chapter" data-level="30.3" data-path="bayesian-theory.html"><a href="bayesian-theory.html#prior-distributions"><i class="fa fa-check"></i><b>30.3</b> Prior Distributions</a>
<ul>
<li class="chapter" data-level="30.3.1" data-path="bayesian-theory.html"><a href="bayesian-theory.html#conjugate-prior-distributions"><i class="fa fa-check"></i><b>30.3.1</b> Conjugate Prior Distributions</a></li>
<li class="chapter" data-level="30.3.2" data-path="bayesian-theory.html"><a href="bayesian-theory.html#non-informativ-priors"><i class="fa fa-check"></i><b>30.3.2</b> Non-informativ Priors</a></li>
</ul></li>
<li class="chapter" data-level="30.4" data-path="bayesian-theory.html"><a href="bayesian-theory.html#mcmc"><i class="fa fa-check"></i><b>30.4</b> MCMC</a>
<ul>
<li class="chapter" data-level="30.4.1" data-path="bayesian-theory.html"><a href="bayesian-theory.html#monte-carlo-simulation"><i class="fa fa-check"></i><b>30.4.1</b> Monte Carlo Simulation</a></li>
<li class="chapter" data-level="30.4.2" data-path="bayesian-theory.html"><a href="bayesian-theory.html#概率分布采样"><i class="fa fa-check"></i><b>30.4.2</b> 概率分布采样</a></li>
<li class="chapter" data-level="30.4.3" data-path="bayesian-theory.html"><a href="bayesian-theory.html#接受-拒绝采样"><i class="fa fa-check"></i><b>30.4.3</b> 接受-拒绝采样</a></li>
<li class="chapter" data-level="30.4.4" data-path="bayesian-theory.html"><a href="bayesian-theory.html#markov-chain"><i class="fa fa-check"></i><b>30.4.4</b> Markov Chain</a></li>
<li class="chapter" data-level="30.4.5" data-path="bayesian-theory.html"><a href="bayesian-theory.html#马尔科夫链的收敛性质"><i class="fa fa-check"></i><b>30.4.5</b> 马尔科夫链的收敛性质</a></li>
<li class="chapter" data-level="30.4.6" data-path="bayesian-theory.html"><a href="bayesian-theory.html#基于马尔科夫链采样"><i class="fa fa-check"></i><b>30.4.6</b> 基于马尔科夫链采样</a></li>
<li class="chapter" data-level="30.4.7" data-path="bayesian-theory.html"><a href="bayesian-theory.html#mcmc采样"><i class="fa fa-check"></i><b>30.4.7</b> MCMC采样</a></li>
<li class="chapter" data-level="30.4.8" data-path="bayesian-theory.html"><a href="bayesian-theory.html#m-h采样"><i class="fa fa-check"></i><b>30.4.8</b> M-H采样</a></li>
<li class="chapter" data-level="30.4.9" data-path="bayesian-theory.html"><a href="bayesian-theory.html#gibbs采样"><i class="fa fa-check"></i><b>30.4.9</b> Gibbs采样</a></li>
</ul></li>
<li class="chapter" data-level="30.5" data-path="bayesian-theory.html"><a href="bayesian-theory.html#regression-and-variable-selection"><i class="fa fa-check"></i><b>30.5</b> Regression and Variable Selection</a>
<ul>
<li class="chapter" data-level="30.5.1" data-path="bayesian-theory.html"><a href="bayesian-theory.html#classical-least-squares-estimator"><i class="fa fa-check"></i><b>30.5.1</b> Classical Least Squares Estimator</a></li>
<li class="chapter" data-level="30.5.2" data-path="bayesian-theory.html"><a href="bayesian-theory.html#the-jeffreys-prior-analysis"><i class="fa fa-check"></i><b>30.5.2</b> The Jeffreys Prior Analysis</a></li>
<li class="chapter" data-level="30.5.3" data-path="bayesian-theory.html"><a href="bayesian-theory.html#zellners-g-prior-analysis"><i class="fa fa-check"></i><b>30.5.3</b> Zellner’s G-prior analysis</a></li>
</ul></li>
<li class="chapter" data-level="30.6" data-path="bayesian-theory.html"><a href="bayesian-theory.html#bayesian-linear-regression-1"><i class="fa fa-check"></i><b>30.6</b> Bayesian linear regression</a></li>
<li class="chapter" data-level="30.7" data-path="bayesian-theory.html"><a href="bayesian-theory.html#introduction-29"><i class="fa fa-check"></i><b>30.7</b> Introduction</a>
<ul>
<li class="chapter" data-level="30.7.1" data-path="bayesian-theory.html"><a href="bayesian-theory.html#item-response-theory"><i class="fa fa-check"></i><b>30.7.1</b> Item response theory</a></li>
<li class="chapter" data-level="30.7.2" data-path="bayesian-theory.html"><a href="bayesian-theory.html#em-algorithm"><i class="fa fa-check"></i><b>30.7.2</b> EM algorithm</a></li>
<li class="chapter" data-level="30.7.3" data-path="bayesian-theory.html"><a href="bayesian-theory.html#mcmc-algorithm"><i class="fa fa-check"></i><b>30.7.3</b> MCMC algorithm</a></li>
<li class="chapter" data-level="30.7.4" data-path="bayesian-theory.html"><a href="bayesian-theory.html#unidimensional-irt-models"><i class="fa fa-check"></i><b>30.7.4</b> Unidimensional IRT Models</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="31" data-path="smoothing.html"><a href="smoothing.html"><i class="fa fa-check"></i><b>31</b> Smoothing</a>
<ul>
<li class="chapter" data-level="31.1" data-path="smoothing.html"><a href="smoothing.html#smoothing-1"><i class="fa fa-check"></i><b>31.1</b> Smoothing</a>
<ul>
<li class="chapter" data-level="31.1.1" data-path="smoothing.html"><a href="smoothing.html#bin-smoothing"><i class="fa fa-check"></i><b>31.1.1</b> Bin smoothing</a></li>
<li class="chapter" data-level="31.1.2" data-path="smoothing.html"><a href="smoothing.html#kernels"><i class="fa fa-check"></i><b>31.1.2</b> Kernels</a></li>
<li class="chapter" data-level="31.1.3" data-path="smoothing.html"><a href="smoothing.html#local-weighted-regression-loess"><i class="fa fa-check"></i><b>31.1.3</b> Local weighted regression (loess)</a></li>
</ul></li>
<li class="chapter" data-level="31.2" data-path="smoothing.html"><a href="smoothing.html#loess-regression"><i class="fa fa-check"></i><b>31.2</b> Loess Regression</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="knn.html"><a href="knn.html"><i class="fa fa-check"></i><b>32</b> KNN</a>
<ul>
<li class="chapter" data-level="32.1" data-path="knn.html"><a href="knn.html#introduction-30"><i class="fa fa-check"></i><b>32.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="32.1.1" data-path="knn.html"><a href="knn.html#idee"><i class="fa fa-check"></i><b>32.1.1</b> Idee</a></li>
<li class="chapter" data-level="32.1.2" data-path="knn.html"><a href="knn.html#加权最近邻法"><i class="fa fa-check"></i><b>32.1.2</b> 加权最近邻法</a></li>
<li class="chapter" data-level="32.1.3" data-path="knn.html"><a href="knn.html#knn算法三要素"><i class="fa fa-check"></i><b>32.1.3</b> KNN算法三要素</a></li>
<li class="chapter" data-level="32.1.4" data-path="knn.html"><a href="knn.html#优缺点"><i class="fa fa-check"></i><b>32.1.4</b> 优缺点</a></li>
</ul></li>
<li class="chapter" data-level="32.2" data-path="knn.html"><a href="knn.html#knn算法的实现方式"><i class="fa fa-check"></i><b>32.2</b> KNN算法的实现方式</a>
<ul>
<li class="chapter" data-level="32.2.1" data-path="knn.html"><a href="knn.html#brute-force"><i class="fa fa-check"></i><b>32.2.1</b> Brute-force</a></li>
<li class="chapter" data-level="32.2.2" data-path="knn.html"><a href="knn.html#kd树实现"><i class="fa fa-check"></i><b>32.2.2</b> KD树实现</a></li>
<li class="chapter" data-level="32.2.3" data-path="knn.html"><a href="knn.html#球树实现"><i class="fa fa-check"></i><b>32.2.3</b> 球树实现</a></li>
</ul></li>
<li class="chapter" data-level="32.3" data-path="knn.html"><a href="knn.html#application"><i class="fa fa-check"></i><b>32.3</b> Application</a>
<ul>
<li class="chapter" data-level="32.3.1" data-path="knn.html"><a href="knn.html#data-preparation-1"><i class="fa fa-check"></i><b>32.3.1</b> Data Preparation</a></li>
<li class="chapter" data-level="32.3.2" data-path="knn.html"><a href="knn.html#knn-modelling"><i class="fa fa-check"></i><b>32.3.2</b> KNN Modelling</a></li>
<li class="chapter" data-level="32.3.3" data-path="knn.html"><a href="knn.html#加权最近邻法-1"><i class="fa fa-check"></i><b>32.3.3</b> 加权最近邻法</a></li>
<li class="chapter" data-level="32.3.4" data-path="knn.html"><a href="knn.html#over-training"><i class="fa fa-check"></i><b>32.3.4</b> Over-training</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="33" data-path="svm.html"><a href="svm.html"><i class="fa fa-check"></i><b>33</b> SVM</a>
<ul>
<li class="chapter" data-level="33.1" data-path="svm.html"><a href="svm.html#introduction-31"><i class="fa fa-check"></i><b>33.1</b> Introduction</a></li>
<li class="chapter" data-level="33.2" data-path="svm.html"><a href="svm.html#introduction-32"><i class="fa fa-check"></i><b>33.2</b> Introduction</a>
<ul>
<li class="chapter" data-level="33.2.1" data-path="svm.html"><a href="svm.html#perceptron"><i class="fa fa-check"></i><b>33.2.1</b> Perceptron</a></li>
<li class="chapter" data-level="33.2.2" data-path="svm.html"><a href="svm.html#函数间隔与几何间隔"><i class="fa fa-check"></i><b>33.2.2</b> 函数间隔与几何间隔</a></li>
<li class="chapter" data-level="33.2.3" data-path="svm.html"><a href="svm.html#svm支持向量"><i class="fa fa-check"></i><b>33.2.3</b> SVM支持向量</a></li>
<li class="chapter" data-level="33.2.4" data-path="svm.html"><a href="svm.html#svm模型目标函数与优化"><i class="fa fa-check"></i><b>33.2.4</b> SVM模型目标函数与优化</a></li>
<li class="chapter" data-level="33.2.5" data-path="svm.html"><a href="svm.html#线性可分svm的算法过程"><i class="fa fa-check"></i><b>33.2.5</b> 线性可分SVM的算法过程</a></li>
<li class="chapter" data-level="33.2.6" data-path="svm.html"><a href="svm.html#线性svm的软间隔最大化"><i class="fa fa-check"></i><b>33.2.6</b> 线性SVM的软间隔最大化</a></li>
<li class="chapter" data-level="33.2.7" data-path="svm.html"><a href="svm.html#线性不可分支持向量机与核函数"><i class="fa fa-check"></i><b>33.2.7</b> 线性不可分支持向量机与核函数</a></li>
</ul></li>
<li class="chapter" data-level="33.3" data-path="svm.html"><a href="svm.html#application-1"><i class="fa fa-check"></i><b>33.3</b> Application</a>
<ul>
<li class="chapter" data-level="33.3.1" data-path="svm.html"><a href="svm.html#data-preparation-2"><i class="fa fa-check"></i><b>33.3.1</b> Data Preparation</a></li>
<li class="chapter" data-level="33.3.2" data-path="svm.html"><a href="svm.html#svm-modelling"><i class="fa fa-check"></i><b>33.3.2</b> SVM Modelling</a></li>
<li class="chapter" data-level="33.3.3" data-path="svm.html"><a href="svm.html#model-selection-1"><i class="fa fa-check"></i><b>33.3.3</b> Model Selection</a></li>
<li class="chapter" data-level="33.3.4" data-path="svm.html"><a href="svm.html#character-selection"><i class="fa fa-check"></i><b>33.3.4</b> Character selection</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="tree-models.html"><a href="tree-models.html"><i class="fa fa-check"></i><b>34</b> Tree models</a>
<ul>
<li class="chapter" data-level="34.1" data-path="tree-models.html"><a href="tree-models.html#decision-tree-model"><i class="fa fa-check"></i><b>34.1</b> Decision Tree Model</a>
<ul>
<li class="chapter" data-level="34.1.1" data-path="tree-models.html"><a href="tree-models.html#decision-tree-algorithm"><i class="fa fa-check"></i><b>34.1.1</b> Decision tree algorithm</a></li>
<li class="chapter" data-level="34.1.2" data-path="tree-models.html"><a href="tree-models.html#id3-algorithm"><i class="fa fa-check"></i><b>34.1.2</b> ID3 Algorithm</a></li>
<li class="chapter" data-level="34.1.3" data-path="tree-models.html"><a href="tree-models.html#c4.5-algorithm"><i class="fa fa-check"></i><b>34.1.3</b> C4.5 Algorithm</a></li>
<li class="chapter" data-level="34.1.4" data-path="tree-models.html"><a href="tree-models.html#cart-algorithm"><i class="fa fa-check"></i><b>34.1.4</b> CART Algorithm</a></li>
<li class="chapter" data-level="34.1.5" data-path="tree-models.html"><a href="tree-models.html#pruning"><i class="fa fa-check"></i><b>34.1.5</b> Pruning</a></li>
<li class="chapter" data-level="34.1.6" data-path="tree-models.html"><a href="tree-models.html#package-rpart"><i class="fa fa-check"></i><b>34.1.6</b> Package ‘rpart’</a></li>
</ul></li>
<li class="chapter" data-level="34.2" data-path="tree-models.html"><a href="tree-models.html#random-forest"><i class="fa fa-check"></i><b>34.2</b> Random Forest</a>
<ul>
<li class="chapter" data-level="34.2.1" data-path="tree-models.html"><a href="tree-models.html#bootstrap-bagging"><i class="fa fa-check"></i><b>34.2.1</b> Bootstrap (Bagging)</a></li>
<li class="chapter" data-level="34.2.2" data-path="tree-models.html"><a href="tree-models.html#bagging算法流程"><i class="fa fa-check"></i><b>34.2.2</b> bagging算法流程</a></li>
<li class="chapter" data-level="34.2.3" data-path="tree-models.html"><a href="tree-models.html#random-forest-algorithm"><i class="fa fa-check"></i><b>34.2.3</b> Random Forest Algorithm</a></li>
<li class="chapter" data-level="34.2.4" data-path="tree-models.html"><a href="tree-models.html#random-forest-promotion"><i class="fa fa-check"></i><b>34.2.4</b> Random forest promotion</a></li>
<li class="chapter" data-level="34.2.5" data-path="tree-models.html"><a href="tree-models.html#package-randomforest"><i class="fa fa-check"></i><b>34.2.5</b> Package ‘randomForest’</a></li>
</ul></li>
<li class="chapter" data-level="34.3" data-path="tree-models.html"><a href="tree-models.html#modelling-2"><i class="fa fa-check"></i><b>34.3</b> Modelling</a>
<ul>
<li class="chapter" data-level="34.3.1" data-path="tree-models.html"><a href="tree-models.html#data-preparation-3"><i class="fa fa-check"></i><b>34.3.1</b> Data preparation</a></li>
<li class="chapter" data-level="34.3.2" data-path="tree-models.html"><a href="tree-models.html#regression-tree-1"><i class="fa fa-check"></i><b>34.3.2</b> Regression tree</a></li>
<li class="chapter" data-level="34.3.3" data-path="tree-models.html"><a href="tree-models.html#classification-tree-1"><i class="fa fa-check"></i><b>34.3.3</b> Classification tree</a></li>
<li class="chapter" data-level="34.3.4" data-path="tree-models.html"><a href="tree-models.html#random-forest-for-regression"><i class="fa fa-check"></i><b>34.3.4</b> Random forest for regression</a></li>
<li class="chapter" data-level="34.3.5" data-path="tree-models.html"><a href="tree-models.html#random-forest-for-classification"><i class="fa fa-check"></i><b>34.3.5</b> Random forest for classification</a></li>
<li class="chapter" data-level="34.3.6" data-path="tree-models.html"><a href="tree-models.html#皮玛印第安人糖尿病数据集"><i class="fa fa-check"></i><b>34.3.6</b> 皮玛印第安人糖尿病数据集</a></li>
<li class="chapter" data-level="34.3.7" data-path="tree-models.html"><a href="tree-models.html#使用随机森林进行特征选择"><i class="fa fa-check"></i><b>34.3.7</b> 使用随机森林进行特征选择</a></li>
</ul></li>
<li class="chapter" data-level="34.4" data-path="tree-models.html"><a href="tree-models.html#gradient-boosting"><i class="fa fa-check"></i><b>34.4</b> Gradient Boosting</a>
<ul>
<li class="chapter" data-level="34.4.1" data-path="tree-models.html"><a href="tree-models.html#gbdt概述"><i class="fa fa-check"></i><b>34.4.1</b> GBDT概述</a></li>
<li class="chapter" data-level="34.4.2" data-path="tree-models.html"><a href="tree-models.html#gbdt的负梯度拟合"><i class="fa fa-check"></i><b>34.4.2</b> GBDT的负梯度拟合</a></li>
<li class="chapter" data-level="34.4.3" data-path="tree-models.html"><a href="tree-models.html#gbdt回归算法"><i class="fa fa-check"></i><b>34.4.3</b> GBDT回归算法</a></li>
<li class="chapter" data-level="34.4.4" data-path="tree-models.html"><a href="tree-models.html#gbdt分类算法"><i class="fa fa-check"></i><b>34.4.4</b> GBDT分类算法</a></li>
<li class="chapter" data-level="34.4.5" data-path="tree-models.html"><a href="tree-models.html#package-gbm"><i class="fa fa-check"></i><b>34.4.5</b> Package ‘gbm’</a></li>
<li class="chapter" data-level="34.4.6" data-path="tree-models.html"><a href="tree-models.html#极限梯度提升分类"><i class="fa fa-check"></i><b>34.4.6</b> 极限梯度提升——分类</a></li>
</ul></li>
<li class="chapter" data-level="34.5" data-path="tree-models.html"><a href="tree-models.html#cubist-model"><i class="fa fa-check"></i><b>34.5</b> Cubist Model</a>
<ul>
<li class="chapter" data-level="34.5.1" data-path="tree-models.html"><a href="tree-models.html#introduction-33"><i class="fa fa-check"></i><b>34.5.1</b> Introduction</a></li>
<li class="chapter" data-level="34.5.2" data-path="tree-models.html"><a href="tree-models.html#application-data-preparation"><i class="fa fa-check"></i><b>34.5.2</b> Application Data Preparation</a></li>
<li class="chapter" data-level="34.5.3" data-path="tree-models.html"><a href="tree-models.html#fit-continious-outcome"><i class="fa fa-check"></i><b>34.5.3</b> Fit Continious Outcome</a></li>
<li class="chapter" data-level="34.5.4" data-path="tree-models.html"><a href="tree-models.html#variable-importance"><i class="fa fa-check"></i><b>34.5.4</b> Variable Importance</a></li>
<li class="chapter" data-level="34.5.5" data-path="tree-models.html"><a href="tree-models.html#summary-display"><i class="fa fa-check"></i><b>34.5.5</b> Summary display</a></li>
<li class="chapter" data-level="34.5.6" data-path="tree-models.html"><a href="tree-models.html#specific-parts"><i class="fa fa-check"></i><b>34.5.6</b> specific parts</a></li>
<li class="chapter" data-level="34.5.7" data-path="tree-models.html"><a href="tree-models.html#ensembles-by-committees"><i class="fa fa-check"></i><b>34.5.7</b> Ensembles By Committees</a></li>
<li class="chapter" data-level="34.5.8" data-path="tree-models.html"><a href="tree-models.html#nearestneighbors-adjustmemt"><i class="fa fa-check"></i><b>34.5.8</b> Nearest–neighbors Adjustmemt</a></li>
<li class="chapter" data-level="34.5.9" data-path="tree-models.html"><a href="tree-models.html#optimize-parameters"><i class="fa fa-check"></i><b>34.5.9</b> Optimize parameters</a></li>
<li class="chapter" data-level="34.5.10" data-path="tree-models.html"><a href="tree-models.html#logistic-cv"><i class="fa fa-check"></i><b>34.5.10</b> Logistic CV</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="35" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>35</b> PCA</a>
<ul>
<li class="chapter" data-level="35.1" data-path="pca.html"><a href="pca.html#introduction-34"><i class="fa fa-check"></i><b>35.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="35.1.1" data-path="pca.html"><a href="pca.html#component"><i class="fa fa-check"></i><b>35.1.1</b> Component</a></li>
<li class="chapter" data-level="35.1.2" data-path="pca.html"><a href="pca.html#pca算法"><i class="fa fa-check"></i><b>35.1.2</b> PCA算法</a></li>
<li class="chapter" data-level="35.1.3" data-path="pca.html"><a href="pca.html#主成分旋转"><i class="fa fa-check"></i><b>35.1.3</b> 主成分旋转</a></li>
<li class="chapter" data-level="35.1.4" data-path="pca.html"><a href="pca.html#kernelized-pca"><i class="fa fa-check"></i><b>35.1.4</b> Kernelized PCA</a></li>
</ul></li>
<li class="chapter" data-level="35.2" data-path="pca.html"><a href="pca.html#application-2"><i class="fa fa-check"></i><b>35.2</b> Application</a>
<ul>
<li class="chapter" data-level="35.2.1" data-path="pca.html"><a href="pca.html#data-preparation-4"><i class="fa fa-check"></i><b>35.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="35.2.2" data-path="pca.html"><a href="pca.html#modeling-1"><i class="fa fa-check"></i><b>35.2.2</b> Modeling</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="36" data-path="cluster-analysis.html"><a href="cluster-analysis.html"><i class="fa fa-check"></i><b>36</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="36.1" data-path="cluster-analysis.html"><a href="cluster-analysis.html#hierarchical-clustering"><i class="fa fa-check"></i><b>36.1</b> Hierarchical Clustering</a>
<ul>
<li class="chapter" data-level="36.1.1" data-path="cluster-analysis.html"><a href="cluster-analysis.html#introduction-35"><i class="fa fa-check"></i><b>36.1.1</b> Introduction</a></li>
<li class="chapter" data-level="36.1.2" data-path="cluster-analysis.html"><a href="cluster-analysis.html#hierarchical-clustering-algorithms"><i class="fa fa-check"></i><b>36.1.2</b> Hierarchical clustering algorithms</a></li>
<li class="chapter" data-level="36.1.3" data-path="cluster-analysis.html"><a href="cluster-analysis.html#measure-the-dissimilarity-between-two-clusters-of-observations"><i class="fa fa-check"></i><b>36.1.3</b> Measure the dissimilarity between two clusters of observations</a></li>
</ul></li>
<li class="chapter" data-level="36.2" data-path="cluster-analysis.html"><a href="cluster-analysis.html#k-means-clustering"><i class="fa fa-check"></i><b>36.2</b> K-means Clustering</a>
<ul>
<li class="chapter" data-level="36.2.1" data-path="cluster-analysis.html"><a href="cluster-analysis.html#algorithm"><i class="fa fa-check"></i><b>36.2.1</b> Algorithm</a></li>
<li class="chapter" data-level="36.2.2" data-path="cluster-analysis.html"><a href="cluster-analysis.html#k-means"><i class="fa fa-check"></i><b>36.2.2</b> K-Means++</a></li>
<li class="chapter" data-level="36.2.3" data-path="cluster-analysis.html"><a href="cluster-analysis.html#elkan-k-means"><i class="fa fa-check"></i><b>36.2.3</b> elkan K-Means</a></li>
<li class="chapter" data-level="36.2.4" data-path="cluster-analysis.html"><a href="cluster-analysis.html#mini-batch-k-means"><i class="fa fa-check"></i><b>36.2.4</b> Mini Batch K-Means</a></li>
</ul></li>
<li class="chapter" data-level="36.3" data-path="cluster-analysis.html"><a href="cluster-analysis.html#gowers-coefficient-and-pam"><i class="fa fa-check"></i><b>36.3</b> Gower’s coefficient and PAM</a>
<ul>
<li class="chapter" data-level="36.3.1" data-path="cluster-analysis.html"><a href="cluster-analysis.html#gowers-coefficient"><i class="fa fa-check"></i><b>36.3.1</b> Gower’s coefficient</a></li>
<li class="chapter" data-level="36.3.2" data-path="cluster-analysis.html"><a href="cluster-analysis.html#不同数据类型的相异度计算-距离法"><i class="fa fa-check"></i><b>36.3.2</b> 不同数据类型的相异度计算 (距离法)</a></li>
<li class="chapter" data-level="36.3.3" data-path="cluster-analysis.html"><a href="cluster-analysis.html#pam"><i class="fa fa-check"></i><b>36.3.3</b> PAM</a></li>
</ul></li>
<li class="chapter" data-level="36.4" data-path="cluster-analysis.html"><a href="cluster-analysis.html#birch-clustering"><i class="fa fa-check"></i><b>36.4</b> BIRCH Clustering</a>
<ul>
<li class="chapter" data-level="36.4.1" data-path="cluster-analysis.html"><a href="cluster-analysis.html#birch-introduction"><i class="fa fa-check"></i><b>36.4.1</b> BIRCH Introduction</a></li>
<li class="chapter" data-level="36.4.2" data-path="cluster-analysis.html"><a href="cluster-analysis.html#聚类特征cf与聚类特征树cf-tree"><i class="fa fa-check"></i><b>36.4.2</b> 聚类特征CF与聚类特征树CF Tree</a></li>
<li class="chapter" data-level="36.4.3" data-path="cluster-analysis.html"><a href="cluster-analysis.html#cf-tree的生成"><i class="fa fa-check"></i><b>36.4.3</b> CF Tree的生成</a></li>
<li class="chapter" data-level="36.4.4" data-path="cluster-analysis.html"><a href="cluster-analysis.html#birch算法"><i class="fa fa-check"></i><b>36.4.4</b> BIRCH算法</a></li>
</ul></li>
<li class="chapter" data-level="36.5" data-path="cluster-analysis.html"><a href="cluster-analysis.html#application-3"><i class="fa fa-check"></i><b>36.5</b> Application</a>
<ul>
<li class="chapter" data-level="36.5.1" data-path="cluster-analysis.html"><a href="cluster-analysis.html#data-preparation-5"><i class="fa fa-check"></i><b>36.5.1</b> Data preparation</a></li>
<li class="chapter" data-level="36.5.2" data-path="cluster-analysis.html"><a href="cluster-analysis.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>36.5.2</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="36.5.3" data-path="cluster-analysis.html"><a href="cluster-analysis.html#k-means-clustering-1"><i class="fa fa-check"></i><b>36.5.3</b> K-means Clustering</a></li>
<li class="chapter" data-level="36.5.4" data-path="cluster-analysis.html"><a href="cluster-analysis.html#gowers-coefficient-and-pam-1"><i class="fa fa-check"></i><b>36.5.4</b> Gower’s coefficient and PAM</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="37" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html"><i class="fa fa-check"></i><b>37</b> linear discriminant analysis (LDA)</a>
<ul>
<li class="chapter" data-level="37.1" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html#introduction-36"><i class="fa fa-check"></i><b>37.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="37.1.1" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html#naive-bayes"><i class="fa fa-check"></i><b>37.1.1</b> Naive Bayes</a></li>
<li class="chapter" data-level="37.1.2" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html#controlling-prevalence"><i class="fa fa-check"></i><b>37.1.2</b> Controlling prevalence</a></li>
<li class="chapter" data-level="37.1.3" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html#qda"><i class="fa fa-check"></i><b>37.1.3</b> QDA</a></li>
<li class="chapter" data-level="37.1.4" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html#lda"><i class="fa fa-check"></i><b>37.1.4</b> LDA</a></li>
</ul></li>
<li class="chapter" data-level="37.2" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html#discriminant-analysis-algorithm"><i class="fa fa-check"></i><b>37.2</b> Discriminant analysis algorithm</a>
<ul>
<li class="chapter" data-level="37.2.1" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html#idee-1"><i class="fa fa-check"></i><b>37.2.1</b> Idee</a></li>
<li class="chapter" data-level="37.2.2" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html#瑞利商rayleigh-quotient"><i class="fa fa-check"></i><b>37.2.2</b> 瑞利商（Rayleigh quotient）</a></li>
<li class="chapter" data-level="37.2.3" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html#广义瑞利商-genralized-rayleigh-quotient"><i class="fa fa-check"></i><b>37.2.3</b> 广义瑞利商 genralized Rayleigh quotient</a></li>
<li class="chapter" data-level="37.2.4" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html#lda算法流程"><i class="fa fa-check"></i><b>37.2.4</b> LDA算法流程</a></li>
<li class="chapter" data-level="37.2.5" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html#lda-application"><i class="fa fa-check"></i><b>37.2.5</b> LDA Application</a></li>
<li class="chapter" data-level="37.2.6" data-path="linear-discriminant-analysis-lda.html"><a href="linear-discriminant-analysis-lda.html#qda-1"><i class="fa fa-check"></i><b>37.2.6</b> QDA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="38" data-path="neural-network.html"><a href="neural-network.html"><i class="fa fa-check"></i><b>38</b> Neural Network</a>
<ul>
<li class="chapter" data-level="38.1" data-path="neural-network.html"><a href="neural-network.html#introduction-37"><i class="fa fa-check"></i><b>38.1</b> Introduction</a></li>
<li class="chapter" data-level="38.2" data-path="neural-network.html"><a href="neural-network.html#反向传播方法进行训练的前馈神经网络"><i class="fa fa-check"></i><b>38.2</b> 反向传播方法进行训练的前馈神经网络</a></li>
<li class="chapter" data-level="38.3" data-path="neural-network.html"><a href="neural-network.html#application-4"><i class="fa fa-check"></i><b>38.3</b> Application</a>
<ul>
<li class="chapter" data-level="38.3.1" data-path="neural-network.html"><a href="neural-network.html#数据准备"><i class="fa fa-check"></i><b>38.3.1</b> 数据准备</a></li>
<li class="chapter" data-level="38.3.2" data-path="neural-network.html"><a href="neural-network.html#模型构建"><i class="fa fa-check"></i><b>38.3.2</b> 模型构建</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">As a Statistician</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="survival-analysis" class="section level1" number="17">
<h1><span class="header-section-number">Chapter 17</span> Survival Analysis</h1>
<div id="htmlwidget-8d9534e324db9aa9d38c" style="width:672px;height:480px;" class="markmap html-widget"></div>
<script type="application/json" data-for="htmlwidget-8d9534e324db9aa9d38c">{"x":{"data":"# \n## Survival Analysis \n### Preliminary\n#### Probability density function\n#### Cumulative distribution function\n#### Survival function\n#### Hazard function\n#### Cumulative hazard function\n#### Mean Residual Life\n#### Relation between functions\n### Kaplan-Meier estimator\n#### KM Introduction\n#### Nelson-Aalen estimator of the cumulative hazard function\n#### Survival curve in SAS\n#### KM Survival Plots in SAS\n#### Convert Personal-level to Personal-period in R\n#### Package survfit in R\n#### KM Survival Plots in R\n### Compare the survival function\n#### Tests of equality of the survival function\n#### Other nonparametric tests for STRATA statement \n#### Multiple comparisons\n#### Comparing survival functions using Log-Rank HR\n#### Mantel-Haenszel HR\n#### Analysis-of-covariance (Adjustment)\n### Accelerated failure time (AFT) model\n#### Introduction\n#### PROC LIFEREG\n#### Residuum distribution\n#### Exponential Model \n#### Weibull Model\n#### Log-Normal Model\n#### Log-Logistic Model\n#### Fit statistics for model comparsion\n#### Graphical method for distinguishing different distributions in SAS\n#### Prediction and hazard function\n#### Left Censoring and Interval Censoring\n### Cox Proportional Hazards Model\n#### Introduction\n#### Parameter estimate\n##### NR iteration for parameter estimate\n##### Model fitting in SAS\n##### Model fitting in R\n#### Parameter test\n#### Graphs of the survival and baseline hazard function in SAS\n#### Check proportional hazards\n#### Dealing with nonproportionality\n#### Model Diagnostics\n##### Cox-Snell Residuals\n##### Martingale residuals\n##### Deviance Residuals\n##### Schoenfeld residuals\n##### Testing influential observations","options":{"preset":"colorful","autoFit":true}},"evals":[],"jsHooks":[]}</script>
<div id="preliminary" class="section level2" number="17.1">
<h2><span class="header-section-number">17.1</span> Preliminary</h2>
<p>Survival analysis mainly focuses on processing a special kind of time data, and the time data may be partially censored. The shortcomings of the ordinary least squares regression method are that the event occurrence time is usually not normally distributed, and the model cannot handle Censor. The non-parametric method can simply and quickly check the survival experience, while the Cox proportional hazard regression model is still the main Analytical method.</p>
<div id="probability-density-function" class="section level3" number="17.1.1">
<h3><span class="header-section-number">17.1.1</span> Probability density function</h3>
<p>The function that describes likelihood of observing Time at time t relative to all other survival times is known as the probability density function
the probability of observing a survival time within the interval <span class="math inline">\([a, b]\)</span> is
<span class="math display">\[
\operatorname{Pr}(a \leq \text { Time } \leq b)=\int_{a}^{b} f(t) d t=\int_{a}^{b} \lambda e^{-\lambda t} d t
\]</span>
### Cumulative distribution function</p>
<p>Describes the probability of observing Time less than or equal to some timet <span class="math inline">\(t\)</span> <span class="math inline">\(\operatorname{Pr}(\)</span> Time <span class="math inline">\(\leq t)\)</span>
<span class="math display">\[
\begin{array}{c}
F(t)=\int_{0}^{t} f(t) d t \\
f(t)=\frac{d F(t)}{d t}
\end{array}
\]</span>
In SAS, we can graph an estimate of the cdf using proc univariate.</p>
<pre><code>proc univariate data = whas500(where=(fstat=1));
    var lenfol;
    cdfplot lenfol;
run;</code></pre>
</div>
<div id="survival-function" class="section level3" number="17.1.2">
<h3><span class="header-section-number">17.1.2</span> Survival function</h3>
<p><span class="math display">\[
S(t)=P(T&gt;t)=1-F(t)
\]</span>
The survival function gives the probability that a person survives longer than some specified time <span class="math inline">\(t\)</span> : that
is, <span class="math inline">\(S(t)\)</span> gives the probability that the random variable <span class="math inline">\(T\)</span> exceeds the specified time <span class="math inline">\(t .\)</span> And here, some
important characteristics:</p>
<ul>
<li>It is nonincreasing; that is, it heads downward as <span class="math inline">\(t\)</span> increases.</li>
<li>At time <span class="math inline">\(t=0, S(t)=S(0)=1\)</span>; that is, at the start of the study, since no one has gotten the event
yet, the probability of surviving past time zero is one.</li>
<li>At time <span class="math inline">\(t=\inf , S(t)=S(\mathrm{inf})=0 ;\)</span> that is, theoretically, if the study period increased without limit.
eventually nobody would survive, so the survival curve must eventually fall to zero.</li>
</ul>
<p>Here we can use proc lifetest to graph <span class="math inline">\(S_(t)\)</span>.</p>
<pre><code>proc lifetest data=whas500(where=(fstat=1)) plots=survival(atrisk);
time lenfol*fstat(0);
run; </code></pre>
</div>
<div id="hazard-function" class="section level3" number="17.1.3">
<h3><span class="header-section-number">17.1.3</span> Hazard function</h3>
<p>The hazard function <span class="math inline">\(h(t)\)</span>, is given by the formula:
<span class="math display">\[
h(t)=\lim _{\Delta_{t} \rightarrow 0} \frac{P(t \leq T&lt;t+\Delta t \mid T \geq t)}{\Delta t}
\]</span>
We could say that the hazard function
is the probability that if you survive to time <span class="math inline">\(t\)</span>, you will experience the event in the next instant, or in other
words, the hazard function gives the instantaneous potential per unit time for the event to occur, given
that the individual has survived up to time <span class="math inline">\(t\)</span>. Because of the given sign here, the hazard function is
sometimes called a conditional failure rate.</p>
<p>We can estimate the hazard function is SAS as well using proc lifetest</p>
<pre><code>proc lifetest data=whas500(where=(fstat=1)) plots=hazard(bw=200);
    time lenfol*fstat(0);
run;</code></pre>
</div>
<div id="cumulative-hazard-function" class="section level3" number="17.1.4">
<h3><span class="header-section-number">17.1.4</span> Cumulative hazard function</h3>
<p>Calculated by integrating the hazard function over an interval of time:
<span class="math display">\[H(t) = \int_0^th(u)du\]</span></p>
<p>The cumulative hazard function H(t) and the survival function S(t) have a simple monotonic relationship. Therefore, when the survival function reaches the maximum at the beginning of the analysis time, the cumulative hazard function is the smallest. As time goes by, the survival function advances toward its minimum, and the cumulative hazard function advances toward its maximum. You can use proc lifetest to estimate the cumulative hazard function, and then send the results to proc sgplot for plotting.</p>
<pre><code>ods output ProductLimitEstimates = ple;
proc lifetest data=whas500(where=(fstat=1))  nelson outs=outwhas500;
    time lenfol*fstat(0);
run;
proc sgplot data = ple;
    series x = lenfol y = CumHaz;
run;</code></pre>
</div>
<div id="mean-residual-life" class="section level3" number="17.1.5">
<h3><span class="header-section-number">17.1.5</span> Mean Residual Life</h3>
<p><span class="math display">\[r(t)=E(T-t \mid T \geq t)=\frac{\int_{t}^{\infty} S(u) d u}{S(t)},\]</span></p>
<p>Intuitively, this is as simple as when I know that I have lived to the time point t and how many years I have to live.</p>
</div>
<div id="relation-between-functions" class="section level3" number="17.1.6">
<h3><span class="header-section-number">17.1.6</span> Relation between functions</h3>
<p>The survival function can be ascertained from the probability density function by integrating
over the probability density function from time <span class="math inline">\(t\)</span> to infinity, or by calculating the difference between one and the cumulative distribution function <span class="math inline">\(F(t)\)</span>. The hazard can then be found by dividing the negative derivative of the survival function by the survival function. Note that the functions <span class="math inline">\(f(t), F(t), h(t)\)</span>, and
<span class="math inline">\(H(t)\)</span> are all related.</p>
<p><strong>Assume that <span class="math inline">\(T\)</span> is non-negative and continuos:</strong></p>
<ul>
<li>Probability density function:
<span class="math display">\[
f(t)=F^{\prime}(t)=\frac{d F(t)}{d t}
\]</span></li>
<li>Cumulative distribution function:
<span class="math display">\[
F(t)=P(T \leq t)=\int_{0}^{t} f(u) d u
\]</span></li>
<li>Survival function
<span class="math display">\[
\begin{array}{l}
S(t)=1-F(t)\\
S(t)=P(T&gt;t)=\int_{t}^{+\infty} f(u) d u \\
S(t)=\exp \left(-\int_{0}^{t} h(u) d u\right) \\
S(t)=\exp (-H(t))
\end{array}
\]</span></li>
<li>Hazard function
<span class="math display">\[h(t) = \frac{ f(t)}{S(t)}= \frac{ -d[S(t)]/dt}{S(t)}\]</span></li>
<li>Cumulative hazard function
o Cumulative hazard function
<span class="math display">\[
H(t)=\int_{0}^{t} h(u) d u
\]</span></li>
</ul>
<p><strong>Assume that <span class="math inline">\(T\)</span> is non-negative and discrete,</strong></p>
<ul>
<li>Probability mass function:
<span class="math display">\[
\begin{aligned}
p\left(t_{i}\right) &amp;=P\left(T=t_{i}\right) \\
p\left(t_{i}\right) &amp;=S\left(t_{i-1}\right)-S\left(t_{i}\right) \\
p\left(t_{i}\right) &amp;=F\left(t_{i}\right)-F\left(t_{i-1}\right)
\end{aligned}
\]</span></li>
<li>Cumulative distribution function:
<span class="math display">\[
F(t)=P(T \leq t)=\sum_{t_{i} \leq t} p\left(t_{i}\right)
\]</span></li>
<li>Survival function
<span class="math display">\[
S(t)=\prod_{t_{i} \leq t}\left(1-h\left(t_{i}\right)\right)
\]</span></li>
<li>Hazard function
<span class="math display">\[
\begin{aligned}
h(t) &amp;=\frac{p\left(t_{i}\right)}{S\left(t_{i-1}\right)}=\frac{-d[S(t)] / d t}{S(t)} \\
h(t) &amp;=1-\frac{S\left(t_{i}\right)}{S\left(t_{i-1}\right)}
\end{aligned}
\]</span></li>
<li>Cumulative hazard function
<span class="math display">\[
H(t)=\sum_{t_{i} \leq t} h\left(t_{i}\right)
\]</span></li>
</ul>
</div>
</div>
<div id="kaplan-meier-estimator" class="section level2" number="17.2">
<h2><span class="header-section-number">17.2</span> Kaplan-Meier estimator</h2>
<div id="km-introduction" class="section level3" number="17.2.1">
<h3><span class="header-section-number">17.2.1</span> KM Introduction</h3>
<p><span class="math display">\[\hat S(t)=\prod_{t_i\leq t}\frac{n_i – d_i}{n_i},\]</span></p>
<ul>
<li><span class="math inline">\(n_{i}\)</span> is the number of subjects at risk</li>
<li><span class="math inline">\(d_{i}\)</span> is the number of subjects who fail, both at time <span class="math inline">\(t_{i}\)</span>, the number who failed out
of <span class="math inline">\(n_{i}\)</span></li>
</ul>
</div>
<div id="nelson-aalen-estimator-of-the-cumulative-hazard-function" class="section level3" number="17.2.2">
<h3><span class="header-section-number">17.2.2</span> Nelson-Aalen estimator of the cumulative hazard function</h3>
<p>Since it and the survival function <span class="math inline">\(S(t)=e^{-H(t)}\)</span> Nelson-Aalen estimator is a non-parametric estimator of the cumulative hazard function</p>
<p><span class="math display">\[
\hat{H}(t)=\sum_{t_{i} l e q t} \frac{d_{i}}{n_{i}}
\]</span></p>
<ul>
<li>The Nelson-Aalen estimator is requested in SAS through the nelson option on
the proc lifetest statement. SAS will output both Kaplan Meier estimates of
the survival function and Nelson-Aalen estimates of the cumulative hazard
function in one table.</li>
<li>Quartile Estimates: Calculating median, mean, and other survival times</li>
</ul>
<pre><code>proc lifetest data=whas500 atrisk nelson;
    time lenfol*fstat(0);
run;</code></pre>
</div>
<div id="survival-curve-in-sas" class="section level3" number="17.2.3">
<h3><span class="header-section-number">17.2.3</span> Survival curve in SAS</h3>
<pre><code>proc lifetest data=whas500 atrisk plots=survival(cb) outs=outwhas500;
  time lenfol*fstat(0);
run;</code></pre>
<p><strong>Graphing the Kaplan-Meier estimate</strong></p>
<ul>
<li>By default, proc lifetest graphs the Kaplan Meier estimate, even without the
plot= option on the proc lifetest statement, so we could have used the
same code from above that produced the table of Kaplan-Meier estimates to
generate the graph.</li>
<li>However, we would like to add confidence bands and the number at risk to the
graph, so we add plots=survival(atrisk cb)</li>
</ul>
<blockquote>
<p>生存曲线周围的蓝色阴影区域代表95%置信带，此处为Hall-Wellner置信带。为整 个生存函数计算该置信带，在任何给定的时间间隔内，其宽度必须大于点状置信区
间 (单个间隔附近的置信度间隔)， 以确保所有点状置信区间的95%都包含在该带 中。survivor 函数的许多转换可用于通过conftype选项计算置信区间的替代方法</p>
</blockquote>
<p><strong>Life Table method</strong></p>
<p>生命表方法如果观察的次数很多，并且如果精确地测量了事件时间，那么将有很多唯一的事件时间。然后，KM方法会生成较长的表，这些表可能难以呈现和解释。</p>
<ol style="list-style-type: decimal">
<li>解决此问题的一种方法是使用TIMELIST选项（在PROC LIFEREG语句中），该选项仅在指定的时间点报告KM估计。</li>
<li>另一种解决方案是切换到生命周期表方法 life-table method ，在该方法中，事件时间被分组为可以随意设置的时间间隔。此外，寿命表方法（也称为精算方法 actuarial method）可以生成危害函数的估计值和曲线图。寿命表方法的缺点是间隔的选择通常有些随意，从而导致结果的
随意性以及如何选择间隔的不确定性。不可避免地也会丢失一些信息。但是注意，PROC LIFETEST根据未分组的数据（如果可用）计算对数秩和Wilcoxon统计信息（以及其他可选的测试统计信息），
因此它们不受寿命表方法的间隔选择的影响。</li>
</ol>
<pre><code>ODS GRAPHICS ON;
PROC LIFETEST DATA=recid METHOD=LIFE PLOTS=(S,H);
 TIME week*arrest(0);
RUN;
ODS GRAPHICS OFF;

    Interval    Number   Number    Sample   Probability   Standard
 [Lower, Upper) Failed   Censored  Size     of Failure    Error
  0      10     14       0         432.0    0.0324        0.00852
 10      20     21       0         418.0    0.0502        0.0107
 20      30     23       0         397.0    0.0579        0.0117
 30      40     23       0         374.0    0.0615        0.0124
 40      50     26       0         351.0    0.0741        0.0140
 50      60      7       318       166.0    0.0422        0.0156</code></pre>
</div>
<div id="km-survival-plots-in-sas" class="section level3" number="17.2.4">
<h3><span class="header-section-number">17.2.4</span> KM Survival Plots in SAS</h3>
<ul>
<li>在任何被检查的观察结果中，图形都包含加号 (看起来像刻度线)。 当数据集很大且包含大量经过审 查的观察结果时, 这些标记可能会使人分心。可以使用NOCENSOR选项抑制它们.</li>
<li>另一个有用的选项是ATRISK，可以将仍然处于危险中 (尚未死亡或受到审查) 的人数添加到图表中。</li>
<li>要获得具有幸存者功能95%置信度限制的图形, 请使用CL选项。置信度极限是逐点极限 pointwise limits, 这意味着对于每个指定的生存时间，我们有95%的置信度到那个时间生存的概率在那些极限之 内。置信度限制仅扩展到最大事件时间.</li>
<li>假设我们需要置信带 confidence bands that can be interpreted by saying that we are <span class="math inline">\(95 \%\)</span> confident that the entire survivor function falls within the upper curve and the lower curve. More complex
methods are needed to produce such bands, and PROC LIFETEST offers two: the HallWellner method
and the equal precision (EP) method. 有95%的信心说整个幸存者功能都落在上曲线和下曲线之间。 产生这样的频带需要更复杂的方法, 而 PROC LIFETEST提供两种方法：HallWellner方法和等精度 (EP) 方法。
<ul>
<li>EP方法 倾向于产生在尾部更稳定的置信带。要实现此方法, 该选项将变为PLOTS <span class="math inline">\(=\mathrm{S}(\mathrm{CB}=\)</span> EP) 。 要同时获得逐点频带和EP频带, 请使用选项PLOTS <span class="math inline">\(=\mathrm{S}(\mathrm{CL} \mathrm{CB}=\mathrm{EP})\)</span> 置信带始终比点状 置信极限宽。</li>
<li>Other transformations are available, the most attractive being the logit <span class="math inline">\(\log [\hat{S}(t) /(1-\hat{S}(t))]\)</span> To switch to this transform, include the CONFTYPE=LOGIT option in the PROC statement.</li>
</ul></li>
</ul>
<pre><code>proc lifetest data=whas500 atrisk  PLOTS=S(NOCENSOR ATRISK CL) OUTSURV=outwhas500;
  time lenfol*fstat(0);
run;</code></pre>
<p><strong>Test in plot</strong></p>
<pre><code>ODS GRAPHICS ON;
PROC LIFETEST DATA=myel PLOTS=S(TEST);
 TIME dur*status(0);
 STRATA treat;
RUN;
ODS GRAPHICS OFF;</code></pre>
<p><strong>log-log survival plots</strong></p>
<p>LS produces a plot of <span class="math inline">\(-\log \hat{S}(t)\)</span> versus <span class="math inline">\(t\)</span>.
<span class="math display">\[
-\log S(t)=\int_{0}^{t} h(u) d u
\]</span></p>
<p>The LLS keyword produces a plot of <span class="math inline">\(\log [-\log \hat{S}(t)]\)</span> versus <span class="math inline">\(\log t\)</span>. If survival times follow a Weibull distribution, which has a hazard given by <span class="math inline">\(\log h(t)=\alpha+\beta \log t\)</span>, then the log-log survival plot (log
cumulative hazard plot) should be a straight line with a slope of <span class="math inline">\(\beta\)</span>.
If the hazards are proportional, the log-log survivor functions should be strictly parallel.</p>
<pre><code>PROC LIFETEST DATA=COMBINE PLOTS=LLS;
 TIME years*event(0);
 STRATA type;
RUN; </code></pre>
<p><strong>Hazard plots</strong></p>
<p>Examine smoothed hazard plots using the kernel smoothing option</p>
<pre><code>ODS GRAPHICS ON;
PROC LIFETEST DATA=combine PLOTS=H(BW=10);
 TIME years*event(0);
 STRATA type;
RUN;
ODS GRAPHICS OFF;

## confidence limits around the hazard function;
PROC LIFETEST DATA=recid PLOTS=H(CL);</code></pre>
</div>
<div id="convert-personal-level-to-personal-period-in-r" class="section level3" number="17.2.5">
<h3><span class="header-section-number">17.2.5</span> Convert Personal-level to Personal-period in R</h3>
<p>In studies of survival or modeling discrete-time events, one compact way to store data is in what may be called, “person-level” or generally “observation-level.” For example, you could have three variables, one indicating the observation, one indicating the time period the event occurred or the last follow-up period and one indicating whether the observation was censored.</p>
<p>The PLPP function takes five arguments. The first, data is the data set to be converted. The second, id is the name of the variable containing the identifier for each observation. The third, period is the name of the variable that indicates how many periods the person or observation was in. The fourth, event is the name of the variable that indicates whether the event occurred or not or whether the observation was censored (depending on which direction you are converting). The fifth, direction indicates whether the function should go from person-level to person-period or from person-period to person-level. There are two options, “period” to go to person-period or “level” to go to person-level. Now let’s try it out. For the examples that follow to work, you need to source the function into R.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="survival-analysis.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Person-Level Person-Period Converter Function</span></span>
<span id="cb12-2"><a href="survival-analysis.html#cb12-2" aria-hidden="true" tabindex="-1"></a>PLPP <span class="ot">&lt;-</span> <span class="cf">function</span>(data, id, period, event, <span class="at">direction =</span> <span class="fu">c</span>(<span class="st">&quot;period&quot;</span>, <span class="st">&quot;level&quot;</span>)){</span>
<span id="cb12-3"><a href="survival-analysis.html#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Data Checking and Verification Steps</span></span>
<span id="cb12-4"><a href="survival-analysis.html#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">is.matrix</span>(data) <span class="sc">||</span> <span class="fu">is.data.frame</span>(data))</span>
<span id="cb12-5"><a href="survival-analysis.html#cb12-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">c</span>(id, period, event) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="fu">colnames</span>(data), <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(data)))</span>
<span id="cb12-6"><a href="survival-analysis.html#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="survival-analysis.html#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">any</span>(<span class="fu">is.na</span>(data[, <span class="fu">c</span>(id, period, event)]))) {</span>
<span id="cb12-8"><a href="survival-analysis.html#cb12-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;PLPP cannot currently handle missing data in the id, period, or event variables&quot;</span>)</span>
<span id="cb12-9"><a href="survival-analysis.html#cb12-9" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb12-10"><a href="survival-analysis.html#cb12-10" aria-hidden="true" tabindex="-1"></a>   <span class="do">## Do the conversion</span></span>
<span id="cb12-11"><a href="survival-analysis.html#cb12-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">switch</span>(<span class="fu">match.arg</span>(direction),</span>
<span id="cb12-12"><a href="survival-analysis.html#cb12-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">period =</span> {</span>
<span id="cb12-13"><a href="survival-analysis.html#cb12-13" aria-hidden="true" tabindex="-1"></a>      index <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), data[, period])</span>
<span id="cb12-14"><a href="survival-analysis.html#cb12-14" aria-hidden="true" tabindex="-1"></a>      idmax <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(data[, period])</span>
<span id="cb12-15"><a href="survival-analysis.html#cb12-15" aria-hidden="true" tabindex="-1"></a>      reve <span class="ot">&lt;-</span> <span class="sc">!</span>data[, event]</span>
<span id="cb12-16"><a href="survival-analysis.html#cb12-16" aria-hidden="true" tabindex="-1"></a>      dat <span class="ot">&lt;-</span> data[index, ]</span>
<span id="cb12-17"><a href="survival-analysis.html#cb12-17" aria-hidden="true" tabindex="-1"></a>      dat[, period] <span class="ot">&lt;-</span> <span class="fu">ave</span>(dat[, period], dat[, id], <span class="at">FUN =</span> seq_along)</span>
<span id="cb12-18"><a href="survival-analysis.html#cb12-18" aria-hidden="true" tabindex="-1"></a>      dat[, event] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb12-19"><a href="survival-analysis.html#cb12-19" aria-hidden="true" tabindex="-1"></a>      dat[idmax, event] <span class="ot">&lt;-</span> reve},</span>
<span id="cb12-20"><a href="survival-analysis.html#cb12-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">level =</span> {</span>
<span id="cb12-21"><a href="survival-analysis.html#cb12-21" aria-hidden="true" tabindex="-1"></a>      tmp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(data[, <span class="fu">c</span>(period, id)], <span class="at">i =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data))</span>
<span id="cb12-22"><a href="survival-analysis.html#cb12-22" aria-hidden="true" tabindex="-1"></a>      index <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">by</span>(tmp, tmp[, id],</span>
<span id="cb12-23"><a href="survival-analysis.html#cb12-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">FUN =</span> <span class="cf">function</span>(x) x[<span class="fu">which.max</span>(x[, period]), <span class="st">&quot;i&quot;</span>]))</span>
<span id="cb12-24"><a href="survival-analysis.html#cb12-24" aria-hidden="true" tabindex="-1"></a>      dat <span class="ot">&lt;-</span> data[index, ]</span>
<span id="cb12-25"><a href="survival-analysis.html#cb12-25" aria-hidden="true" tabindex="-1"></a>      dat[, event] <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="sc">!</span>dat[, event])</span>
<span id="cb12-26"><a href="survival-analysis.html#cb12-26" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb12-27"><a href="survival-analysis.html#cb12-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-28"><a href="survival-analysis.html#cb12-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(dat) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-29"><a href="survival-analysis.html#cb12-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(dat)</span>
<span id="cb12-30"><a href="survival-analysis.html#cb12-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb12-31"><a href="survival-analysis.html#cb12-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-32"><a href="survival-analysis.html#cb12-32" aria-hidden="true" tabindex="-1"></a><span class="do">## Read in the person-level dataset</span></span>
<span id="cb12-33"><a href="survival-analysis.html#cb12-33" aria-hidden="true" tabindex="-1"></a>teachers <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;https://stats.idre.ucla.edu/stat/examples/alda/teachers.csv&quot;</span>)</span>
<span id="cb12-34"><a href="survival-analysis.html#cb12-34" aria-hidden="true" tabindex="-1"></a><span class="do">## Look at a subset of the cases</span></span>
<span id="cb12-35"><a href="survival-analysis.html#cb12-35" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(teachers, id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">20</span>, <span class="dv">126</span>, <span class="dv">129</span>))</span></code></pre></div>
<pre><code>##      id  t censor
## 19   20  3      0
## 125 126 12      0
## 128 129 12      1</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="survival-analysis.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Uses PLPP to convert to person-period and store in object, &#39;tpp&#39;</span></span>
<span id="cb14-2"><a href="survival-analysis.html#cb14-2" aria-hidden="true" tabindex="-1"></a>tpp <span class="ot">&lt;-</span> <span class="fu">PLPP</span>(<span class="at">data =</span> teachers, <span class="at">id =</span> <span class="st">&quot;id&quot;</span>, <span class="at">period =</span> <span class="st">&quot;t&quot;</span>, <span class="at">event =</span> <span class="st">&quot;censor&quot;</span>, <span class="at">direction =</span> <span class="st">&quot;period&quot;</span>)</span>
<span id="cb14-3"><a href="survival-analysis.html#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Look at a subset of the cases</span></span>
<span id="cb14-4"><a href="survival-analysis.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(tpp, id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">20</span>, <span class="dv">126</span>, <span class="dv">129</span>))</span></code></pre></div>
<pre><code>##      id  t censor
## 95   20  1      0
## 96   20  2      0
## 97   20  3      1
## 740 126  1      0
## 741 126  2      0
## 742 126  3      0
## 743 126  4      0
## 744 126  5      0
## 745 126  6      0
## 746 126  7      0
## 747 126  8      0
## 748 126  9      0
## 749 126 10      0
## 750 126 11      0
## 751 126 12      1
## 760 129  1      0
## 761 129  2      0
## 762 129  3      0
## 763 129  4      0
## 764 129  5      0
## 765 129  6      0
## 766 129  7      0
## 767 129  8      0
## 768 129  9      0
## 769 129 10      0
## 770 129 11      0
## 771 129 12      0</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="survival-analysis.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Convert person-period to person-level</span></span>
<span id="cb16-2"><a href="survival-analysis.html#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Read in person-period dataset</span></span>
<span id="cb16-3"><a href="survival-analysis.html#cb16-3" aria-hidden="true" tabindex="-1"></a>teachers.pp <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;https://stats.idre.ucla.edu/stat/examples/alda/teachers_pp.csv&quot;</span>)</span>
<span id="cb16-4"><a href="survival-analysis.html#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Look at a subset of the cases</span></span>
<span id="cb16-5"><a href="survival-analysis.html#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(teachers.pp, id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">20</span>, <span class="dv">126</span>, <span class="dv">129</span>))</span></code></pre></div>
<pre><code>##      id period event
## 95   20      1     0
## 96   20      2     0
## 97   20      3     1
## 740 126      1     0
## 741 126      2     0
## 742 126      3     0
## 743 126      4     0
## 744 126      5     0
## 745 126      6     0
## 746 126      7     0
## 747 126      8     0
## 748 126      9     0
## 749 126     10     0
## 750 126     11     0
## 751 126     12     1
## 760 129      1     0
## 761 129      2     0
## 762 129      3     0
## 763 129      4     0
## 764 129      5     0
## 765 129      6     0
## 766 129      7     0
## 767 129      8     0
## 768 129      9     0
## 769 129     10     0
## 770 129     11     0
## 771 129     12     0</code></pre>
</div>
<div id="package-survfit-in-r" class="section level3" number="17.2.6">
<h3><span class="header-section-number">17.2.6</span> Package survfit in R</h3>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="survival-analysis.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Compute survival curves</span></span>
<span id="cb18-2"><a href="survival-analysis.html#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;survival&quot;</span>)</span>
<span id="cb18-3"><a href="survival-analysis.html#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;survminer&quot;</span>)</span>
<span id="cb18-4"><a href="survival-analysis.html#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="survival-analysis.html#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;lung&quot;</span>)</span>
<span id="cb18-6"><a href="survival-analysis.html#cb18-6" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">survfit</span>(<span class="fu">Surv</span>(time, status) <span class="sc">~</span> <span class="dv">1</span>, <span class="at">data =</span> lung)</span>
<span id="cb18-7"><a href="survival-analysis.html#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Stratification</span></span>
<span id="cb18-8"><a href="survival-analysis.html#cb18-8" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">survfit</span>(<span class="fu">Surv</span>(time, status) <span class="sc">~</span> sex, <span class="at">data =</span> lung)</span>
<span id="cb18-9"><a href="survival-analysis.html#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(fit)</span></code></pre></div>
<pre><code>## Call: survfit(formula = Surv(time, status) ~ sex, data = lung)
## 
##                 sex=1 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##    11    138       3   0.9783  0.0124       0.9542        1.000
##    12    135       1   0.9710  0.0143       0.9434        0.999
##    13    134       2   0.9565  0.0174       0.9231        0.991
##    15    132       1   0.9493  0.0187       0.9134        0.987
##    26    131       1   0.9420  0.0199       0.9038        0.982
##    30    130       1   0.9348  0.0210       0.8945        0.977
##    31    129       1   0.9275  0.0221       0.8853        0.972
##    53    128       2   0.9130  0.0240       0.8672        0.961
##    54    126       1   0.9058  0.0249       0.8583        0.956
##    59    125       1   0.8986  0.0257       0.8496        0.950
##    60    124       1   0.8913  0.0265       0.8409        0.945
##    65    123       2   0.8768  0.0280       0.8237        0.933
##    71    121       1   0.8696  0.0287       0.8152        0.928
##    81    120       1   0.8623  0.0293       0.8067        0.922
##    88    119       2   0.8478  0.0306       0.7900        0.910
##    92    117       1   0.8406  0.0312       0.7817        0.904
##    93    116       1   0.8333  0.0317       0.7734        0.898
##    95    115       1   0.8261  0.0323       0.7652        0.892
##   105    114       1   0.8188  0.0328       0.7570        0.886
##   107    113       1   0.8116  0.0333       0.7489        0.880
##   110    112       1   0.8043  0.0338       0.7408        0.873
##   116    111       1   0.7971  0.0342       0.7328        0.867
##   118    110       1   0.7899  0.0347       0.7247        0.861
##   131    109       1   0.7826  0.0351       0.7167        0.855
##   132    108       2   0.7681  0.0359       0.7008        0.842
##   135    106       1   0.7609  0.0363       0.6929        0.835
##   142    105       1   0.7536  0.0367       0.6851        0.829
##   144    104       1   0.7464  0.0370       0.6772        0.823
##   147    103       1   0.7391  0.0374       0.6694        0.816
##   156    102       2   0.7246  0.0380       0.6538        0.803
##   163    100       3   0.7029  0.0389       0.6306        0.783
##   166     97       1   0.6957  0.0392       0.6230        0.777
##   170     96       1   0.6884  0.0394       0.6153        0.770
##   175     94       1   0.6811  0.0397       0.6076        0.763
##   176     93       1   0.6738  0.0399       0.5999        0.757
##   177     92       1   0.6664  0.0402       0.5922        0.750
##   179     91       2   0.6518  0.0406       0.5769        0.736
##   180     89       1   0.6445  0.0408       0.5693        0.730
##   181     88       2   0.6298  0.0412       0.5541        0.716
##   183     86       1   0.6225  0.0413       0.5466        0.709
##   189     83       1   0.6150  0.0415       0.5388        0.702
##   197     80       1   0.6073  0.0417       0.5309        0.695
##   202     78       1   0.5995  0.0419       0.5228        0.687
##   207     77       1   0.5917  0.0420       0.5148        0.680
##   210     76       1   0.5839  0.0422       0.5068        0.673
##   212     75       1   0.5762  0.0424       0.4988        0.665
##   218     74       1   0.5684  0.0425       0.4909        0.658
##   222     72       1   0.5605  0.0426       0.4829        0.651
##   223     70       1   0.5525  0.0428       0.4747        0.643
##   229     67       1   0.5442  0.0429       0.4663        0.635
##   230     66       1   0.5360  0.0431       0.4579        0.627
##   239     64       1   0.5276  0.0432       0.4494        0.619
##   246     63       1   0.5192  0.0433       0.4409        0.611
##   267     61       1   0.5107  0.0434       0.4323        0.603
##   269     60       1   0.5022  0.0435       0.4238        0.595
##   270     59       1   0.4937  0.0436       0.4152        0.587
##   283     57       1   0.4850  0.0437       0.4065        0.579
##   284     56       1   0.4764  0.0438       0.3979        0.570
##   285     54       1   0.4676  0.0438       0.3891        0.562
##   286     53       1   0.4587  0.0439       0.3803        0.553
##   288     52       1   0.4499  0.0439       0.3716        0.545
##   291     51       1   0.4411  0.0439       0.3629        0.536
##   301     48       1   0.4319  0.0440       0.3538        0.527
##   303     46       1   0.4225  0.0440       0.3445        0.518
##   306     44       1   0.4129  0.0440       0.3350        0.509
##   310     43       1   0.4033  0.0441       0.3256        0.500
##   320     42       1   0.3937  0.0440       0.3162        0.490
##   329     41       1   0.3841  0.0440       0.3069        0.481
##   337     40       1   0.3745  0.0439       0.2976        0.471
##   353     39       2   0.3553  0.0437       0.2791        0.452
##   363     37       1   0.3457  0.0436       0.2700        0.443
##   364     36       1   0.3361  0.0434       0.2609        0.433
##   371     35       1   0.3265  0.0432       0.2519        0.423
##   387     34       1   0.3169  0.0430       0.2429        0.413
##   390     33       1   0.3073  0.0428       0.2339        0.404
##   394     32       1   0.2977  0.0425       0.2250        0.394
##   428     29       1   0.2874  0.0423       0.2155        0.383
##   429     28       1   0.2771  0.0420       0.2060        0.373
##   442     27       1   0.2669  0.0417       0.1965        0.362
##   455     25       1   0.2562  0.0413       0.1868        0.351
##   457     24       1   0.2455  0.0410       0.1770        0.341
##   460     22       1   0.2344  0.0406       0.1669        0.329
##   477     21       1   0.2232  0.0402       0.1569        0.318
##   519     20       1   0.2121  0.0397       0.1469        0.306
##   524     19       1   0.2009  0.0391       0.1371        0.294
##   533     18       1   0.1897  0.0385       0.1275        0.282
##   558     17       1   0.1786  0.0378       0.1179        0.270
##   567     16       1   0.1674  0.0371       0.1085        0.258
##   574     15       1   0.1562  0.0362       0.0992        0.246
##   583     14       1   0.1451  0.0353       0.0900        0.234
##   613     13       1   0.1339  0.0343       0.0810        0.221
##   624     12       1   0.1228  0.0332       0.0722        0.209
##   643     11       1   0.1116  0.0320       0.0636        0.196
##   655     10       1   0.1004  0.0307       0.0552        0.183
##   689      9       1   0.0893  0.0293       0.0470        0.170
##   707      8       1   0.0781  0.0276       0.0390        0.156
##   791      7       1   0.0670  0.0259       0.0314        0.143
##   814      5       1   0.0536  0.0239       0.0223        0.128
##   883      3       1   0.0357  0.0216       0.0109        0.117
## 
##                 sex=2 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##     5     90       1   0.9889  0.0110       0.9675        1.000
##    60     89       1   0.9778  0.0155       0.9478        1.000
##    61     88       1   0.9667  0.0189       0.9303        1.000
##    62     87       1   0.9556  0.0217       0.9139        0.999
##    79     86       1   0.9444  0.0241       0.8983        0.993
##    81     85       1   0.9333  0.0263       0.8832        0.986
##    95     83       1   0.9221  0.0283       0.8683        0.979
##   107     81       1   0.9107  0.0301       0.8535        0.972
##   122     80       1   0.8993  0.0318       0.8390        0.964
##   145     79       2   0.8766  0.0349       0.8108        0.948
##   153     77       1   0.8652  0.0362       0.7970        0.939
##   166     76       1   0.8538  0.0375       0.7834        0.931
##   167     75       1   0.8424  0.0387       0.7699        0.922
##   182     71       1   0.8305  0.0399       0.7559        0.913
##   186     70       1   0.8187  0.0411       0.7420        0.903
##   194     68       1   0.8066  0.0422       0.7280        0.894
##   199     67       1   0.7946  0.0432       0.7142        0.884
##   201     66       2   0.7705  0.0452       0.6869        0.864
##   208     62       1   0.7581  0.0461       0.6729        0.854
##   226     59       1   0.7452  0.0471       0.6584        0.843
##   239     57       1   0.7322  0.0480       0.6438        0.833
##   245     54       1   0.7186  0.0490       0.6287        0.821
##   268     51       1   0.7045  0.0501       0.6129        0.810
##   285     47       1   0.6895  0.0512       0.5962        0.798
##   293     45       1   0.6742  0.0523       0.5791        0.785
##   305     43       1   0.6585  0.0534       0.5618        0.772
##   310     42       1   0.6428  0.0544       0.5447        0.759
##   340     39       1   0.6264  0.0554       0.5267        0.745
##   345     38       1   0.6099  0.0563       0.5089        0.731
##   348     37       1   0.5934  0.0572       0.4913        0.717
##   350     36       1   0.5769  0.0579       0.4739        0.702
##   351     35       1   0.5604  0.0586       0.4566        0.688
##   361     33       1   0.5434  0.0592       0.4390        0.673
##   363     32       1   0.5265  0.0597       0.4215        0.658
##   371     30       1   0.5089  0.0603       0.4035        0.642
##   426     26       1   0.4893  0.0610       0.3832        0.625
##   433     25       1   0.4698  0.0617       0.3632        0.608
##   444     24       1   0.4502  0.0621       0.3435        0.590
##   450     23       1   0.4306  0.0624       0.3241        0.572
##   473     22       1   0.4110  0.0626       0.3050        0.554
##   520     19       1   0.3894  0.0629       0.2837        0.534
##   524     18       1   0.3678  0.0630       0.2628        0.515
##   550     15       1   0.3433  0.0634       0.2390        0.493
##   641     11       1   0.3121  0.0649       0.2076        0.469
##   654     10       1   0.2808  0.0655       0.1778        0.443
##   687      9       1   0.2496  0.0652       0.1496        0.417
##   705      8       1   0.2184  0.0641       0.1229        0.388
##   728      7       1   0.1872  0.0621       0.0978        0.359
##   731      6       1   0.1560  0.0590       0.0743        0.328
##   735      5       1   0.1248  0.0549       0.0527        0.295
##   765      3       1   0.0832  0.0499       0.0257        0.270</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="survival-analysis.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="do">##  Access to the sort summary table</span></span>
<span id="cb20-2"><a href="survival-analysis.html#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(fit)<span class="sc">$</span>table</span></code></pre></div>
<pre><code>##       records n.max n.start events    rmean se(rmean) median 0.95LCL 0.95UCL
## sex=1     138   138     138    112 326.0841  22.91156    270     212     310
## sex=2      90    90      90     53 460.6473  34.68985    426     348     550</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="survival-analysis.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="do">##  function survfit() returns a list of variables,components can be accessed as follow:</span></span>
<span id="cb22-2"><a href="survival-analysis.html#cb22-2" aria-hidden="true" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">time =</span> fit<span class="sc">$</span>time,</span>
<span id="cb22-3"><a href="survival-analysis.html#cb22-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">n.risk =</span> fit<span class="sc">$</span>n.risk,</span>
<span id="cb22-4"><a href="survival-analysis.html#cb22-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">n.event =</span> fit<span class="sc">$</span>n.event,</span>
<span id="cb22-5"><a href="survival-analysis.html#cb22-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">n.censor =</span> fit<span class="sc">$</span>n.censor,</span>
<span id="cb22-6"><a href="survival-analysis.html#cb22-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">surv =</span> fit<span class="sc">$</span>surv,</span>
<span id="cb22-7"><a href="survival-analysis.html#cb22-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">upper =</span> fit<span class="sc">$</span>upper,</span>
<span id="cb22-8"><a href="survival-analysis.html#cb22-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">lower =</span> fit<span class="sc">$</span>lower</span>
<span id="cb22-9"><a href="survival-analysis.html#cb22-9" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb22-10"><a href="survival-analysis.html#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(d)</span></code></pre></div>
<pre><code>##   time n.risk n.event n.censor      surv     upper     lower
## 1   11    138       3        0 0.9782609 1.0000000 0.9542301
## 2   12    135       1        0 0.9710145 0.9994124 0.9434235
## 3   13    134       2        0 0.9565217 0.9911586 0.9230952
## 4   15    132       1        0 0.9492754 0.9866017 0.9133612
## 5   26    131       1        0 0.9420290 0.9818365 0.9038355
## 6   30    130       1        0 0.9347826 0.9768989 0.8944820</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="survival-analysis.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Kaplan-Meier life table: summary of survival curves summary(fit)</span></span>
<span id="cb24-2"><a href="survival-analysis.html#cb24-2" aria-hidden="true" tabindex="-1"></a>res.sum <span class="ot">&lt;-</span> <span class="fu">surv_summary</span>(fit)</span>
<span id="cb24-3"><a href="survival-analysis.html#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res.sum)</span></code></pre></div>
<pre><code>##   time n.risk n.event n.censor      surv    std.err     upper     lower strata
## 1   11    138       3        0 0.9782609 0.01268978 1.0000000 0.9542301  sex=1
## 2   12    135       1        0 0.9710145 0.01470747 0.9994124 0.9434235  sex=1
## 3   13    134       2        0 0.9565217 0.01814885 0.9911586 0.9230952  sex=1
## 4   15    132       1        0 0.9492754 0.01967768 0.9866017 0.9133612  sex=1
## 5   26    131       1        0 0.9420290 0.02111708 0.9818365 0.9038355  sex=1
## 6   30    130       1        0 0.9347826 0.02248469 0.9768989 0.8944820  sex=1
##   sex
## 1   1
## 2   1
## 3   1
## 4   1
## 5   1
## 6   1</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="survival-analysis.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="do">## surv_summary对象还有一个名为“表格”的属性，其中包含有关生存曲线的信息，包括具有置信区间的生存中位数以及每条曲线中受试者的总数和事件数</span></span>
<span id="cb26-2"><a href="survival-analysis.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(res.sum, <span class="st">&quot;table&quot;</span>)</span></code></pre></div>
<pre><code>##       records n.max n.start events    rmean se(rmean) median 0.95LCL 0.95UCL
## sex=1     138   138     138    112 326.0841  22.91156    270     212     310
## sex=2      90    90      90     53 460.6473  34.68985    426     348     550</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="survival-analysis.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Confidence Interval type</span></span>
<span id="cb28-2"><a href="survival-analysis.html#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="do">## One of &quot;none&quot;, &quot;plain&quot;, &quot;log&quot; (the default), &quot;log-log&quot;, &quot;logit&quot; or &quot;arcsin&quot;.</span></span>
<span id="cb28-3"><a href="survival-analysis.html#cb28-3" aria-hidden="true" tabindex="-1"></a> <span class="co"># The none option 将导致不生成置信区间</span></span>
<span id="cb28-4"><a href="survival-analysis.html#cb28-4" aria-hidden="true" tabindex="-1"></a> <span class="co"># The plain option 导致标准间隔曲线 +-k *se(curve), 其中k由conf.int确定. </span></span>
<span id="cb28-5"><a href="survival-analysis.html#cb28-5" aria-hidden="true" tabindex="-1"></a> <span class="co"># The log option calculates intervals based on the cumulative hazard or log(survival). </span></span>
<span id="cb28-6"><a href="survival-analysis.html#cb28-6" aria-hidden="true" tabindex="-1"></a> <span class="co"># The log-log option bases the intervals on the log hazard or log(-log(survival)), 将间隔设置为以log危险或log（-log（survival））为基础</span></span>
<span id="cb28-7"><a href="survival-analysis.html#cb28-7" aria-hidden="true" tabindex="-1"></a> <span class="co"># The logit option on log(survival/(1- survival)) and arcsin on arcsin(survival).</span></span>
<span id="cb28-8"><a href="survival-analysis.html#cb28-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-9"><a href="survival-analysis.html#cb28-9" aria-hidden="true" tabindex="-1"></a><span class="do">## 1) Normalverteilungsannahme</span></span>
<span id="cb28-10"><a href="survival-analysis.html#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="do">## S(t) +- SE(S(t)) * u_(1-alpha/2)</span></span>
<span id="cb28-11"><a href="survival-analysis.html#cb28-11" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">survfit</span>(<span class="fu">Surv</span>(time, status) <span class="sc">~</span> sex, <span class="at">data =</span> lung, <span class="at">conf.type=</span><span class="st">&quot;plain&quot;</span>)</span>
<span id="cb28-12"><a href="survival-analysis.html#cb28-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-13"><a href="survival-analysis.html#cb28-13" aria-hidden="true" tabindex="-1"></a><span class="do">## 2) log-Methode,nicht in Vorlesung</span></span>
<span id="cb28-14"><a href="survival-analysis.html#cb28-14" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">survfit</span>(<span class="fu">Surv</span>(time, status) <span class="sc">~</span> sex, <span class="at">data =</span> lung, <span class="at">conf.type=</span><span class="st">&quot;log&quot;</span>)</span>
<span id="cb28-15"><a href="survival-analysis.html#cb28-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-16"><a href="survival-analysis.html#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="do">## 3) log-log-Methode </span></span>
<span id="cb28-17"><a href="survival-analysis.html#cb28-17" aria-hidden="true" tabindex="-1"></a><span class="do">## S(t)^(exp(+-SE(log(-logS(t)))*u_(1-alpha/2))), dabei muss SE(log(-logS(t))) mit Formel (2.16) berechnet werden </span></span>
<span id="cb28-18"><a href="survival-analysis.html#cb28-18" aria-hidden="true" tabindex="-1"></a> <span class="co"># ( nicht: S(t)^(exp(+-SE *(log(-logS(t)))*u_(1-alpha/2))) )</span></span>
<span id="cb28-19"><a href="survival-analysis.html#cb28-19" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">survfit</span>(<span class="fu">Surv</span>(time, status) <span class="sc">~</span> sex, <span class="at">data =</span> lung, <span class="at">conf.type=</span><span class="st">&quot;log-log&quot;</span>)</span>
<span id="cb28-20"><a href="survival-analysis.html#cb28-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-21"><a href="survival-analysis.html#cb28-21" aria-hidden="true" tabindex="-1"></a><span class="co"># 计算指定生存概率下相应的随访时间</span></span>
<span id="cb28-22"><a href="survival-analysis.html#cb28-22" aria-hidden="true" tabindex="-1"></a><span class="fu">quantile</span>(a ,<span class="at">probs =</span><span class="dv">1</span><span class="sc">-</span> <span class="fu">c</span>(<span class="fl">0.75</span>,<span class="fl">0.5</span>,<span class="fl">0.25</span>))</span></code></pre></div>
<pre><code>## $quantile
##        25  50  75
## sex=1 144 270 457
## sex=2 226 426 687
## 
## $lower
##        25  50  75
## sex=1 107 212 371
## sex=2 186 345 524
## 
## $upper
##        25  50  75
## sex=1 177 306 567
## sex=2 310 524 735</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="survival-analysis.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Survival curves comparing</span></span>
<span id="cb30-2"><a href="survival-analysis.html#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="do">## When rho=0, log-rank test or Mantel-Haenszel test is performed.</span></span>
<span id="cb30-3"><a href="survival-analysis.html#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="do">## When rho=1, the Peto correction test of Gehan-Wilcoxon is performed, which gives a greater weight to the early outcome events. But this is not the Wilcoxon test </span></span>
<span id="cb30-4"><a href="survival-analysis.html#cb30-4" aria-hidden="true" tabindex="-1"></a>surv_diff <span class="ot">&lt;-</span> <span class="fu">survdiff</span>(<span class="fu">Surv</span>(time, status) <span class="sc">~</span> sex, <span class="at">data =</span> lung)</span>
<span id="cb30-5"><a href="survival-analysis.html#cb30-5" aria-hidden="true" tabindex="-1"></a>surv_diff</span></code></pre></div>
<pre><code>## Call:
## survdiff(formula = Surv(time, status) ~ sex, data = lung)
## 
##         N Observed Expected (O-E)^2/E (O-E)^2/V
## sex=1 138      112     91.6      4.55      10.3
## sex=2  90       53     73.4      5.68      10.3
## 
##  Chisq= 10.3  on 1 degrees of freedom, p= 0.001</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="survival-analysis.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting information from a survdiff object</span></span>
<span id="cb32-2"><a href="survival-analysis.html#cb32-2" aria-hidden="true" tabindex="-1"></a>surv_diff <span class="ot">&lt;-</span> <span class="fu">survdiff</span>(<span class="fu">Surv</span>(time, status) <span class="sc">~</span> sex, <span class="at">data =</span> lung)</span>
<span id="cb32-3"><a href="survival-analysis.html#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> <span class="fu">pchisq</span>(surv_diff<span class="sc">$</span>chisq, <span class="fu">length</span>(surv_diff<span class="sc">$</span>n) <span class="sc">-</span> <span class="dv">1</span>)</span></code></pre></div>
<pre><code>## [1] 0.001311165</code></pre>
</div>
<div id="km-survival-plots-in-r" class="section level3" number="17.2.7">
<h3><span class="header-section-number">17.2.7</span> KM Survival Plots in R</h3>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="survival-analysis.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Change color, linetype by strata, risk.table color by strata</span></span>
<span id="cb34-2"><a href="survival-analysis.html#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(fit,</span>
<span id="cb34-3"><a href="survival-analysis.html#cb34-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">pval =</span> <span class="cn">TRUE</span>, <span class="at">conf.int =</span> <span class="cn">TRUE</span>,</span>
<span id="cb34-4"><a href="survival-analysis.html#cb34-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">risk.table =</span> <span class="cn">TRUE</span>,               <span class="co"># Add risk table</span></span>
<span id="cb34-5"><a href="survival-analysis.html#cb34-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">risk.table.col =</span> <span class="st">&quot;strata&quot;</span>,       <span class="co"># Change risk table color by groups</span></span>
<span id="cb34-6"><a href="survival-analysis.html#cb34-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">linetype =</span> <span class="st">&quot;strata&quot;</span>,             <span class="co"># Change line type by groups</span></span>
<span id="cb34-7"><a href="survival-analysis.html#cb34-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">surv.median.line =</span> <span class="st">&quot;hv&quot;</span>,         <span class="co"># Specify median survival</span></span>
<span id="cb34-8"><a href="survival-analysis.html#cb34-8" aria-hidden="true" tabindex="-1"></a>          <span class="at">ggtheme =</span> <span class="fu">theme_bw</span>(),            <span class="co"># Change ggplot2 theme</span></span>
<span id="cb34-9"><a href="survival-analysis.html#cb34-9" aria-hidden="true" tabindex="-1"></a>          <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">&quot;#E7B800&quot;</span>, <span class="st">&quot;#2E9FDF&quot;</span>))</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/Visualize%20survival%20curves-1.png" width="672" /></p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="survival-analysis.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="do">## The plot can be further customized</span></span>
<span id="cb35-2"><a href="survival-analysis.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(</span>
<span id="cb35-3"><a href="survival-analysis.html#cb35-3" aria-hidden="true" tabindex="-1"></a>   fit,                     <span class="co"># survfit object with calculated statistics.</span></span>
<span id="cb35-4"><a href="survival-analysis.html#cb35-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">pval =</span> <span class="cn">TRUE</span>,             <span class="co"># show p-value of log-rank test.</span></span>
<span id="cb35-5"><a href="survival-analysis.html#cb35-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">conf.int =</span> <span class="cn">TRUE</span>,         <span class="co"># show confidence intervals for </span></span>
<span id="cb35-6"><a href="survival-analysis.html#cb35-6" aria-hidden="true" tabindex="-1"></a>                            <span class="co"># point estimaes of survival curves.</span></span>
<span id="cb35-7"><a href="survival-analysis.html#cb35-7" aria-hidden="true" tabindex="-1"></a>   <span class="at">conf.int.style =</span> <span class="st">&quot;step&quot;</span>,  <span class="co"># customize style of confidence intervals</span></span>
<span id="cb35-8"><a href="survival-analysis.html#cb35-8" aria-hidden="true" tabindex="-1"></a>   <span class="at">xlab =</span> <span class="st">&quot;Time in days&quot;</span>,   <span class="co"># customize X axis label.</span></span>
<span id="cb35-9"><a href="survival-analysis.html#cb35-9" aria-hidden="true" tabindex="-1"></a>   <span class="at">break.time.by =</span> <span class="dv">200</span>,     <span class="co"># break X axis in time intervals by 200.</span></span>
<span id="cb35-10"><a href="survival-analysis.html#cb35-10" aria-hidden="true" tabindex="-1"></a>   <span class="at">ggtheme =</span> <span class="fu">theme_light</span>(), <span class="co"># customize plot and risk table with a theme.</span></span>
<span id="cb35-11"><a href="survival-analysis.html#cb35-11" aria-hidden="true" tabindex="-1"></a>   <span class="at">risk.table =</span> <span class="st">&quot;abs_pct&quot;</span>,  <span class="co"># absolute number and percentage at risk.</span></span>
<span id="cb35-12"><a href="survival-analysis.html#cb35-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">risk.table.y.text.col =</span> T,<span class="co"># colour risk table text annotations.</span></span>
<span id="cb35-13"><a href="survival-analysis.html#cb35-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">risk.table.y.text =</span> <span class="cn">FALSE</span>,<span class="co"># show bars instead of names in text annotations</span></span>
<span id="cb35-14"><a href="survival-analysis.html#cb35-14" aria-hidden="true" tabindex="-1"></a>                            <span class="co"># in legend of risk table.</span></span>
<span id="cb35-15"><a href="survival-analysis.html#cb35-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncensor.plot =</span> <span class="cn">TRUE</span>,      <span class="co"># plot the number of censored subjects at time t</span></span>
<span id="cb35-16"><a href="survival-analysis.html#cb35-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">surv.median.line =</span> <span class="st">&quot;hv&quot;</span>,  <span class="co"># add the median survival pointer.</span></span>
<span id="cb35-17"><a href="survival-analysis.html#cb35-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend.labs =</span> </span>
<span id="cb35-18"><a href="survival-analysis.html#cb35-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">&quot;Male&quot;</span>, <span class="st">&quot;Female&quot;</span>),    <span class="co"># change legend labels.</span></span>
<span id="cb35-19"><a href="survival-analysis.html#cb35-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> </span>
<span id="cb35-20"><a href="survival-analysis.html#cb35-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">&quot;#E7B800&quot;</span>, <span class="st">&quot;#2E9FDF&quot;</span>) <span class="co"># custom color palettes.</span></span>
<span id="cb35-21"><a href="survival-analysis.html#cb35-21" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/Visualize%20survival%20curves-2.png" width="672" /></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="survival-analysis.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Plot cumulative events</span></span>
<span id="cb36-2"><a href="survival-analysis.html#cb36-2" aria-hidden="true" tabindex="-1"></a> <span class="do">## fun = &quot;event&quot; cumulative events </span></span>
<span id="cb36-3"><a href="survival-analysis.html#cb36-3" aria-hidden="true" tabindex="-1"></a> <span class="do">## fun = &quot;log&quot;: log transformation </span></span>
<span id="cb36-4"><a href="survival-analysis.html#cb36-4" aria-hidden="true" tabindex="-1"></a> <span class="do">## fun = &quot;cumhaz&quot;: cumulative hazard</span></span>
<span id="cb36-5"><a href="survival-analysis.html#cb36-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-6"><a href="survival-analysis.html#cb36-6" aria-hidden="true" tabindex="-1"></a><span class="do">## plot cumulative events</span></span>
<span id="cb36-7"><a href="survival-analysis.html#cb36-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(fit,</span>
<span id="cb36-8"><a href="survival-analysis.html#cb36-8" aria-hidden="true" tabindex="-1"></a>          <span class="at">conf.int =</span> <span class="cn">TRUE</span>,</span>
<span id="cb36-9"><a href="survival-analysis.html#cb36-9" aria-hidden="true" tabindex="-1"></a>          <span class="at">risk.table.col =</span> <span class="st">&quot;strata&quot;</span>,          <span class="co"># Change risk table color by groups</span></span>
<span id="cb36-10"><a href="survival-analysis.html#cb36-10" aria-hidden="true" tabindex="-1"></a>          <span class="at">ggtheme =</span> <span class="fu">theme_bw</span>(),               <span class="co"># Change ggplot2 theme</span></span>
<span id="cb36-11"><a href="survival-analysis.html#cb36-11" aria-hidden="true" tabindex="-1"></a>          <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">&quot;#E7B800&quot;</span>, <span class="st">&quot;#2E9FDF&quot;</span>),</span>
<span id="cb36-12"><a href="survival-analysis.html#cb36-12" aria-hidden="true" tabindex="-1"></a>          <span class="at">fun =</span> <span class="st">&quot;event&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/Visualize%20survival%20curves-3.png" width="672" /></p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="survival-analysis.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot log survival curve</span></span>
<span id="cb37-2"><a href="survival-analysis.html#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(fit,</span>
<span id="cb37-3"><a href="survival-analysis.html#cb37-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">conf.int =</span> <span class="cn">TRUE</span>,</span>
<span id="cb37-4"><a href="survival-analysis.html#cb37-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">risk.table.col =</span> <span class="st">&quot;strata&quot;</span>, <span class="co"># Change risk table color by groups</span></span>
<span id="cb37-5"><a href="survival-analysis.html#cb37-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">ggtheme =</span> <span class="fu">theme_bw</span>(), <span class="co"># Change ggplot2 theme</span></span>
<span id="cb37-6"><a href="survival-analysis.html#cb37-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">&quot;#E7B800&quot;</span>, <span class="st">&quot;#2E9FDF&quot;</span>),</span>
<span id="cb37-7"><a href="survival-analysis.html#cb37-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">fun =</span> <span class="st">&quot;log&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/Visualize%20survival%20curves-4.png" width="672" /></p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="survival-analysis.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot cummulative hazard</span></span>
<span id="cb38-2"><a href="survival-analysis.html#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="do">## The cummulative hazard is commonly used to estimate the hazard probability</span></span>
<span id="cb38-3"><a href="survival-analysis.html#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(fit,</span>
<span id="cb38-4"><a href="survival-analysis.html#cb38-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">conf.int =</span> <span class="cn">TRUE</span>,</span>
<span id="cb38-5"><a href="survival-analysis.html#cb38-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">risk.table.col =</span> <span class="st">&quot;strata&quot;</span>, <span class="co"># Change risk table color by groups</span></span>
<span id="cb38-6"><a href="survival-analysis.html#cb38-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">ggtheme =</span> <span class="fu">theme_bw</span>(), <span class="co"># Change ggplot2 theme</span></span>
<span id="cb38-7"><a href="survival-analysis.html#cb38-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">&quot;#E7B800&quot;</span>, <span class="st">&quot;#2E9FDF&quot;</span>),</span>
<span id="cb38-8"><a href="survival-analysis.html#cb38-8" aria-hidden="true" tabindex="-1"></a>          <span class="at">fun =</span> <span class="st">&quot;cumhaz&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/Visualize%20survival%20curves-5.png" width="672" /></p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="survival-analysis.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Complex survival curves in grouping</span></span>
<span id="cb39-2"><a href="survival-analysis.html#cb39-2" aria-hidden="true" tabindex="-1"></a>fit2 <span class="ot">&lt;-</span> <span class="fu">survfit</span>( <span class="fu">Surv</span>(time, status) <span class="sc">~</span> sex <span class="sc">+</span> rx <span class="sc">+</span> adhere,</span>
<span id="cb39-3"><a href="survival-analysis.html#cb39-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> colon )</span>
<span id="cb39-4"><a href="survival-analysis.html#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot survival curves by sex and facet by rx and adhere</span></span>
<span id="cb39-5"><a href="survival-analysis.html#cb39-5" aria-hidden="true" tabindex="-1"></a>ggsurv <span class="ot">&lt;-</span> <span class="fu">ggsurvplot</span>(fit2, <span class="at">fun =</span> <span class="st">&quot;event&quot;</span>, <span class="at">conf.int =</span> <span class="cn">TRUE</span>,</span>
<span id="cb39-6"><a href="survival-analysis.html#cb39-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ggtheme =</span> <span class="fu">theme_bw</span>())</span>
<span id="cb39-7"><a href="survival-analysis.html#cb39-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-8"><a href="survival-analysis.html#cb39-8" aria-hidden="true" tabindex="-1"></a>ggsurv<span class="sc">$</span>plot <span class="sc">+</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb39-9"><a href="survival-analysis.html#cb39-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span> (<span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>)<span class="sc">+</span></span>
<span id="cb39-10"><a href="survival-analysis.html#cb39-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(rx <span class="sc">~</span> adhere)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/Visualize%20survival%20curves-6.png" width="672" /></p>
</div>
</div>
<div id="compare-the-survival-function" class="section level2" number="17.3">
<h2><span class="header-section-number">17.3</span> Compare the survival function</h2>
<div id="tests-of-equality-of-the-survival-function" class="section level3" number="17.3.1">
<h3><span class="header-section-number">17.3.1</span> Tests of equality of the survival function</h3>
<p>The log-rank test, is a hypothesis test to compare the survival distributions of two samples. It is a nonparametric test and appropriate to use when the data are right-skewed and censored (technically, the censoring must be non-informative).</p>
<p>The logrank test is based on the same assumptions as the Kaplan-Meier survival curve—namely, that censoring is unrelated to prognosis, the survival probabilities are the same for subjects recruited early and late in the study, and the events happened at the times specified. Deviations from these assumptions matter most if they are satisfied differently in the groups being compared, for example if censoring is more likely in one group than another</p>
<p>The calculation of the statistic for the nonparametric “Log-Rank” and “Wilcoxon” tests is given by :</p>
<p><span class="math display">\[Q = \frac{\bigg[\sum\limits_{i=1}^m w_j(d_{ij}-\hat e_{ij})\bigg]^2}{\sum\limits_{i=1}^m w_j^2\hat v_{ij}},\]</span></p>
<ul>
<li><span class="math inline">\(d_{i j}\)</span> is the observed number of failures in stratum <span class="math inline">\(i\)</span> at time <span class="math inline">\(t_{i j}\)</span></li>
<li><span class="math inline">\(\hat{e}_{i j}\)</span> is the expected number of failures in stratum <span class="math inline">\(i\)</span> at time <span class="math inline">\(t_{i j}\)</span></li>
<li><span class="math inline">\(\hat{v}_{i j}\)</span> is the estimator of the variance of <span class="math inline">\(d_{i j}\)</span></li>
<li><span class="math inline">\(w_{j}\)</span> is the weight of the difference at time <span class="math inline">\(t_{j}\)</span></li>
<li>The log-rank or Mantel-Haenzel test uses <span class="math inline">\(w_{j}=1\)</span></li>
<li>The Wilcoxon test uses <span class="math inline">\(w_{j}=n_{j}\)</span>, so that differences are weighted by the number at risk at time <span class="math inline">\(t_{j}\)</span></li>
</ul>
<p><strong>Specified, for group 1, the log-rank statistic can be written as</strong></p>
<p><span class="math display">\[
\sum_{j=1}^{r}\left(d_{1 j}-e_{1 j}\right)
\]</span>
where the summation is over all unique event times (in both groups), and there are a total of <span class="math inline">\(r\)</span> such
times. <span class="math inline">\(d_{1 j}\)</span> is the number of deaths that occur in group 1 at time <span class="math inline">\(j\)</span>, and <span class="math inline">\(e_{1 j}\)</span> is the expected number of events in group 1 at time <span class="math inline">\(j\)</span>. The expected number is given by <span class="math inline">\(n_{1 j} d_{j} / n_{j}\)</span>, where <span class="math inline">\(n_{j}\)</span> is the total number of cases that are at risk just prior to time <span class="math inline">\(j, n_{1 j}\)</span> is the number at risk just prior to time <span class="math inline">\(j\)</span> in group 1, and <span class="math inline">\(d_{j}\)</span> is the total number of deaths at time <span class="math inline">\(j\)</span> in both groups.</p>
<p><strong>The Wilcoxon statistic, given by</strong></p>
<p><span class="math display">\[\sum_{j=1}^{r} n_{j}\left(d_{1 j}-e_{1 j}\right)\]</span></p>
<p><strong>strata statement in proc lifetest in SAS</strong></p>
<pre><code>proc lifetest data=whas500 atrisk plots=survival(atrisk cb) outs=outwhas500;
strata gender;
time lenfol*fstat(0);
run;</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:Strata"></span>
<img src="02_Plots/Proc_lifetest_strata.png" alt="Figure: Strata statement in proc lifetest" width="100%" />
<p class="caption">
Figure 17.1: Figure: Strata statement in proc lifetest
</p>
</div>
<table>
<thead>
<tr class="header">
<th>Test of Equality over Strata</th>
<th>Title</th>
<th>Title</th>
<th>Title</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Test</td>
<td>Chi-Square</td>
<td>DF</td>
<td>Pr &gt; Chi-Square</td>
</tr>
<tr class="even">
<td>Log-Rank</td>
<td>7.7911</td>
<td>1</td>
<td>0.0053</td>
</tr>
<tr class="odd">
<td>Wilcoxon</td>
<td>5.5370</td>
<td>1</td>
<td>0.0186</td>
</tr>
<tr class="even">
<td>-2Log(LR)</td>
<td>10.5120</td>
<td>1</td>
<td>0.0012</td>
</tr>
</tbody>
</table>
<p><strong>log-rank test v.s. Wilcoxon test</strong></p>
<ul>
<li>If the log-rank test is meaningful but the Wilcoxon test is meaningless, it indicates that the long-term difference may be large, but not necessarily in the early stage, and the difference may not be large.</li>
<li>If the log-rank test is meaningless and the Wilcoxon test is meaningful, it indicates that there is a large difference in early survival and little difference in long-term survival. Because the Wilcoxon test is more important for early weights than for late weights (nj does not increase with time), it is not as sensitive as log-rank test to differences between groups that occur at later time points. In other words, although the two statistics test the same null hypothesis, they have different sensitivities to various deviations from the hypothesis.</li>
</ul>
<blockquote>
<p>如果对数秩检验有意义，而Wilcoxon检验毫无意义，则表明长期差异可能很大，但不一定在早期，差异可能不会很大。
如果对数秩检验无意义，而Wilcoxon检验有意义，则表明早期生存率差异很大，长期生存率差异很小。 因为Wilcoxon检验对于较早的重量比对较重的重量更重要（nj不会随时间增加）</p>
</blockquote>
<p>In particular, the log-rank test is more powerful for detecting differences of the form
<span class="math display">\[S_{1}(t)=\left[S_{2}(t)\right]^{\gamma}, \gamma &gt; 1\]</span></p>
<p>This equation defines a proportional hazards model, the log-rank test is closely related to tests for differences between two groups that are performed within the framework of Cox’s proportional hazards model</p>
<p>In contrast, the Wilcoxon test is more powerful than the log-rank test in situations where event times have log-normal distributions (discussed in the next chapter) with a common variance but with different means in the two groups.</p>
</div>
<div id="other-nonparametric-tests-for-strata-statement" class="section level3" number="17.3.2">
<h3><span class="header-section-number">17.3.2</span> Other nonparametric tests for STRATA statement</h3>
<p>In addition to the log-rank and Wilcoxon tests, which are produced by default, the STRATA statement also has options for four other nonparametric tests:</p>
<ul>
<li>Tarone-Ware,</li>
<li>Peto-Peto,</li>
<li>modified Peto-Peto,</li>
<li>Fleming-Harrington.</li>
</ul>
<p>Like the Wilcoxon and log-rank tests, all of these can be represented as a weighted sum of observed and expected numbers of events:</p>
<pre><code>STRATA treat / TESTS=ALL;</code></pre>
<p>For the Tarone-Ware test, <span class="math inline">\(w j\)</span> is the square root of <span class="math inline">\(n i\)</span>, so this test behaves much like the Wilcoxon test
in being more sensitive to differences at earlier rather than later times.
<span class="math display">\[
\sum_{j=1}^{r} w_{j}\left(d_{1 j}-e_{1 j}\right)
\]</span>
That’s also true of the two Peto tests, for which <span class="math inline">\(w_{j}\)</span> is a function of the survivor function itself.
<span class="math display">\[
\sum_{j=1}^{r} \hat{S}\left(t_{j}\right)\left(d_{1 j}-e_{1 j}\right)
\]</span>
Fleming-Harrington is actually a family of tests in which the weights depend on two parameters, <span class="math inline">\(p\)</span> and
<span class="math inline">\(q\)</span> which can be chosen by the user:
<span class="math display">\[w_{j}=\hat{S}\left(t_{j}\right)^{p}\left[1-\hat{S}\left(t_{j}\right)\right]^{q}\]</span></p>
<ul>
<li>When both <span class="math inline">\(\mathrm{p}\)</span> and <span class="math inline">\(\mathrm{q}\)</span> are 0 , you get the log-rank test.</li>
<li>When <span class="math inline">\(\mathrm{p}\)</span> is 1 and <span class="math inline">\(\mathrm{q}\)</span> is 0 , you get something very close to the Peto-Peto test.</li>
<li>When <span class="math inline">\(\mathrm{q}\)</span> is 1 and <span class="math inline">\(\mathrm{p}\)</span> is 0 , wi increases with time, unlike any of the other tests.</li>
</ul>
</div>
<div id="multiple-comparisons" class="section level3" number="17.3.3">
<h3><span class="header-section-number">17.3.3</span> Multiple comparisons</h3>
<p>When more than one strata, the ADJUST option tells PROC LIFETEST to produce p-values for all six pairwise comparisons of the four strata and then to report p-values that have been adjusted for multiple comparisons using Tukey’s method (other methods are also available):</p>
<pre><code>PROC LIFETEST DAta=my.recid;
 TIME week*arrest(0);
 STRATA wexp paro / ADJUST=TUKEY;
RUN;</code></pre>
<p>For numeric variables, you can use the STRATA statement to define groups by intervals rather than by unique values.</p>
<pre><code>PROC LIFETEST DAta=my.recid;
 TIME week*arrest(0);
 STRATA age(21 24 28) / ADJUST=BON;
RUN;

     age &lt; 21
21 ≤ age &lt; 24
24 ≤ age &lt; 28
28 ≤ age </code></pre>
</div>
<div id="comparing-survival-functions-using-log-rank-hr" class="section level3" number="17.3.4">
<h3><span class="header-section-number">17.3.4</span> Comparing survival functions using Log-Rank HR</h3>
<p>One way to compare two survival curves is to calculate the hazard ratio (HR)</p>
<pre><code>proc lifetest data=whas500 atrisk plots=hazard(bw=200) outs=outwhas500;
    strata bmi(15,18.5,25,30,40);
    time lenfol*fstat(0);
run;</code></pre>
<p>作为Kaplan-Meier计算的一部分，计算每个组中观察到的事件 (通常为死亡) 的数量, 以及假设生存时间 没有差异的零假设的预期事件的数量。危险比为:
- Treatment A and Treatment B. the observed and expected deaths are summed, to give
- <span class="math inline">\(O_{A}=\Sigma O_{A t}, O_{B}=\Sigma O_{B t}, E_{A}=\Sigma E_{A t}\)</span>, and <span class="math inline">\(E_{B}=\Sigma E_{B t}\)</span>
- Finally the Logrank statistic is calculated as
<span class="math display">\[
\chi_{\text {Logrank }}^{2}=\frac{\left(O_{A}-E_{A}\right)^{2}}{E_{A}}+\frac{\left(O_{B}-E_{B}\right)^{2}}{E_{B}}
\]</span>
- It can be written as
<span class="math display">\[
\chi_{\text {Logrank }}^{2}=\left(O_{A}-E_{A}\right)^{2}\left(\frac{1}{E_{A}}+\frac{1}{E_{B}}\right)
\]</span>
- since <span class="math inline">\(\left(O_{A}-E_{A}\right)^{2}=\left(O_{B}-E_{B}\right)^{2}\)</span>.
- We can obtain the ratio <span class="math inline">\(O_{A} / E_{A}\)</span> and <span class="math inline">\(\left.O_{B} / E_{B}\right)\)</span>, we can calculate the hazard ratio HR, defined as
the ratio of these two relative death rates; that is,
<span class="math display">\[H R=\frac{O_{A} / E_{A}}{O_{B} / E_{B}}\]</span></p>
<p><strong>Confidence Interval of HR</strong></p>
<p>In calculating <span class="math inline">\(C I s\)</span>, it is convenient if the statistic under consideration can be assumed to follow an
approximately Normal distribution. However, the estimate of the <span class="math inline">\(H R\)</span> is not normally distributed. In
particular it has a possible range of values from 0 to <span class="math inline">\(\infty\)</span>, with the null hypothesis value of unity not
located at the centre of this interval. To make the scale symmetric and to enable us to calculate <span class="math inline">\(C I s\)</span>,
we transform the estimate to make it approximately normally distributed. We do this by using log <span class="math inline">\(H R\)</span>,
rather than <span class="math inline">\(H R\)</span> itself, as the basis for our calculation. It is possible to show that a general
<span class="math inline">\(100(1-\alpha) \% C I\)</span> for the <span class="math inline">\(\log H R\)</span> is
Convert the estimated value to an approximately normal distribution. For this, we use logarithmic HR
<span class="math display">\[
\log H R-\left[z_{1-\alpha / 2} \times S E(\log H R)\right] \text { to } \log H R+\left[z_{1-\alpha / 2} \times S E(\log H R)\right]
\]</span>
HR itself is then
<span class="math display">\[
\exp \left[\operatorname { l o g } H R - [ z _ { 1 - \alpha / 2 } \times S E ( \operatorname { l o g } H R ) ] \text { to } \operatorname { e x p } \left[\log H R+\left[z_{1-\alpha / 2} \times S E(\log H R)\right]\right.\right.
\]</span>
In both these expressions
<span class="math display">\[S E(\log H R)=\sqrt{\left(\frac{1}{E_{A}}+\frac{1}{E_{B}}\right)}\]</span></p>
</div>
<div id="mantel-haenszel-hr" class="section level3" number="17.3.5">
<h3><span class="header-section-number">17.3.5</span> Mantel-Haenszel HR</h3>
<p>在研究中的事件总数很小的情况下, <span class="math inline">\(S E(\log H R)\)</span> 的估计并不总是可靠的。但对 <span class="math inline">\(S E\)</span> 的首选估计值需要 在每个死亡时间计算方差，称为超几何方差 hypergeometric variance 。始终总结这些差异会导致Logrank
测试的Mantel-Haenszel版本。时间 <span class="math inline">\(t\)</span> 的超几何方差由下式给出:
<span class="math display">\[
V_{t}=\frac{m_{t} n_{t} r_{t} s_{t}}{N_{t}^{2}\left(N_{t}-1\right)}
\]</span>
If all deaths occur at different times, that is, there is no fixed observation value, then
<span class="math display">\[r{t}=1, s{t}=N_{t}-1\]</span>
<span class="math display">\[V_{t}=m_{t} n_{t} / N_{t}^{2} .\]</span>
As for both <span class="math inline">\(E_{A t}\)</span> and <span class="math inline">\(E_{B t}\)</span>, we finally sum the individual <span class="math inline">\(V_{t}\)</span> calculated at each distinct event time to
obtain <span class="math inline">\(V=\Sigma V_{t}\)</span>. The Mantel-Haenszel test statistic is then defined as
<span class="math display">\[
\chi_{M H}^{2}=\frac{\left(O_{A}-E_{A}\right)^{2}}{V}
\]</span>
This has an approximate <span class="math inline">\(\chi^{2}\)</span> distribution with <span class="math inline">\(d f=1\)</span> in the same way as the Logrank test. It should be noted that <span class="math inline">\(\chi_{M H}^{2}\)</span> only differs from <span class="math inline">\(\chi_{\text {Logrank }}^{2}\)</span> by the expression for <span class="math inline">\(V\)</span>.
The use of the Mantel-Haenszel statistic to test for the difference between two groups also gives an alternative estimate of the HR:
<span class="math display">\[
H R_{M H}=\exp \left(\frac{O_{A}-E_{A}}{V}\right)
\]</span>
The corresponding <span class="math inline">\(S E\)</span> of the Mantel-Haenszel <span class="math inline">\(\log H R_{M H}\)</span> is
<span class="math display">\[S E\left(\log H R_{M H}\right)=1 / V^{1 / 2}\]</span>
This <span class="math inline">\(S E\left(\log H R_{M H}\right)\)</span> can be used to give <span class="math inline">\(C I\)</span> s derived from the Mantel-Haenszel estimate of the <span class="math inline">\(H R\)</span> as
<span class="math display">\[
\exp \left[\log H R_{M H}-\left(z_{1-\alpha / 2} / V^{1 / 2}\right)\right] \text { to } \exp \left[\log H R_{M H}+\left(z_{1-\alpha / 2} / V^{1 / 2}\right)\right]
\]</span></p>
<p>See more under <strong>Survival Analysis-Survival Analysis Using SAS - A Practical Guide (2nd Edition)</strong></p>
</div>
<div id="analysis-of-covariance-adjustment" class="section level3" number="17.3.6">
<h3><span class="header-section-number">17.3.6</span> Analysis-of-covariance (Adjustment)</h3>
<p><strong>References</strong>: Heller G. and Venkatraman E.S. (2004) A nonparametric test to compare survival distributions with
covariate adjustment. JRSS-B 66, 719-733.</p>
<p>The analysis of covariance is a technique that is used to improve the power of a k-sample test by adjusting for concomitant variables. In the classic normal linear model, this technique is used for increasing the precision of the k-sample test statistic derived from a randomized study or to adjust for sources of bias in observational studies. The power of the adjusted test statistic is a function of the strength of the association between the covariates and the response. However, the increase in power comes at the cost of model specification. The assumption of a specific linear regression function and a normal error distribution is imposed on the model structure. If either of these assumptions is incorrect, the resulting inference that is drawn from the adjusted test is questionable.</p>
<blockquote>
<p>协方差分析是一种通过调整伴随变量来提高 k 样本检验功效的技术。在经典的正态线性模型中，此技术用于提高从随机研究得出的 k 样本检验统计量的精度，或调整观察性研究中的偏倚来源。调整后的检验统计量的功效是协变量与响应之间关联强度的函数。然而，功率的增加是以模型规格为代价的。对模型结构强加了特定线性回归函数和正态误差分布的假设。如果这些假设中的任何一个不正确，则从调整后的测试中得出的推论是有问题的。</p>
</blockquote>
<p>For the proportional hazards model, the primary assumptions are that the relative risk function <span class="math inline">\(\lambda(t \mid z, \mathbf{x}) / \lambda_{0}(t)\)</span> is</p>
<ol style="list-style-type: lower-alpha">
<li>time invariant and</li>
<li>represented by a log-linear function of the covariates.</li>
</ol>
<p>For the analysis-of-covariance survival problem, an incorrect model specification can lead to erroneous inference on the group effect.</p>
<p>In this paper, a k-sample test statistic is developed for survival time data with the right censoring present which adjusts for three or fewer concomitant covariates but does not require the assumption of proportional hazards or a parametric specification of the relative risk function. The statistic is asymptotically valid when the observed covariates and the group classifier are either independent or dependent, assuming that the underlying failure time and censoring variables are independent conditionally on the group classification and covariates.</p>
<blockquote>
<p>对于协方差生存问题的分析，不正确的模型规范会导致对组效应的错误推断。</p>
</blockquote>
<blockquote>
<p>在本文中，为生存时间数据开发了一个 k 样本检验统计量，其中存在正确的删失，它针对三个或更少的伴随协变量进行调整，但不需要假设比例风险或相对风险函数的参数规范。当观察到的协变量和组分类器独立或相关时，统计量渐近有效，假设基础失效时间和审查变量在组分类和协变量的条件下独立。</p>
</blockquote>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="survival-analysis.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clinfun)</span>
<span id="cb45-2"><a href="survival-analysis.html#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(survival)</span>
<span id="cb45-3"><a href="survival-analysis.html#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(pbc)</span>
<span id="cb45-4"><a href="survival-analysis.html#cb45-4" aria-hidden="true" tabindex="-1"></a>pbc1 <span class="ot">&lt;-</span> pbc</span>
<span id="cb45-5"><a href="survival-analysis.html#cb45-5" aria-hidden="true" tabindex="-1"></a>pbc1<span class="sc">$</span>trt[pbc1<span class="sc">$</span>trt <span class="sc">==</span> <span class="sc">-</span><span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb45-6"><a href="survival-analysis.html#cb45-6" aria-hidden="true" tabindex="-1"></a>pbc1<span class="sc">$</span>copper[pbc1<span class="sc">$</span>copper <span class="sc">==</span> <span class="sc">-</span><span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb45-7"><a href="survival-analysis.html#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="fu">calogrank</span>(pbc1<span class="sc">$</span>time, pbc1<span class="sc">$</span>status, pbc1<span class="sc">$</span>trt, pbc1[,<span class="fu">c</span>(<span class="st">&quot;copper&quot;</span>)])</span></code></pre></div>
<pre><code>## Call:
## calogrank(ftime = pbc1$time, fstatus = pbc1$status, grp = pbc1$trt, 
##     cvt = pbc1[, c(&quot;copper&quot;)])
## 
## n=143, 167 observations deleted due to missingness.
## 
##        N Observed Expected Exptd-adj (O-E)^2/V (O-aE)^2/aV
## grp=1 74       64     64.8      9.91     0.023        28.0
## grp=2 69       60     59.2      9.09     0.023        24.8
## 
##  Unconditional chisq= 0.02  on 1 degrees of freedom, p= 0.88 
##       Adjusted chisq= 0.15  on 1 degrees of freedom, p= 0.697</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="survival-analysis.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">calogrank</span>(pbc1<span class="sc">$</span>time, pbc1<span class="sc">$</span>status, pbc1<span class="sc">$</span>trt, pbc1[,<span class="fu">c</span>(<span class="st">&quot;protime&quot;</span>, <span class="st">&quot;copper&quot;</span>)])</span></code></pre></div>
<pre><code>## Call:
## calogrank(ftime = pbc1$time, fstatus = pbc1$status, grp = pbc1$trt, 
##     cvt = pbc1[, c(&quot;protime&quot;, &quot;copper&quot;)])
## 
## n=143, 167 observations deleted due to missingness.
## 
##        N Observed Expected Exptd-adj (O-E)^2/V (O-aE)^2/aV
## grp=1 74       64     64.8      9.93     0.023        46.3
## grp=2 69       60     59.2      9.07     0.023        41.1
## 
##  Unconditional chisq= 0.02  on 1 degrees of freedom, p= 0.88 
##       Adjusted chisq= 0.56  on 1 degrees of freedom, p= 0.453</code></pre>
</div>
</div>
<div id="accelerated-failure-time-aft-model" class="section level2" number="17.4">
<h2><span class="header-section-number">17.4</span> Accelerated failure time (AFT) model</h2>
<div id="introduction-12" class="section level3" number="17.4.1">
<h3><span class="header-section-number">17.4.1</span> Introduction</h3>
<p>The accelerated failure time model (AFT model) is a parametric model that can replace the commonly used proportional hazard model. The proportional hazard model assumes that the role of the covariate is to multiply the risk by a certain constant, while the AFT model assumes that the role of the covariate is to <strong>accelerate or slow down the life course of the disease by a certain constant.</strong> The AFT model describes the relationship between the survivor functions of any two individuals in its most general form.</p>
<p>Assuming that <span class="math inline">\(T\)</span> is the failure time and <span class="math inline">\(x\)</span> is the covariate, the assumption of the accelerated failure time model is that the survival time of a person is equal to the baseline survival time of the population, the acceleration factor of this person, and its mathematical form is as follows:
<span class="math display">\[
T=t * e^{\theta \cdot x}, t=e^{\mu+\sigma * W}
\]</span>
Common forms also include:
<span class="math display">\[
S(t \mid x)=S_{0}\left(t * e^{\theta \cdot x}\right)
\]</span>
<span class="math display">\[
Y=\log (T)=\mu+\gamma \cdot x+\sigma * W
\]</span>
Among them, the baseline survival time <span class="math inline">\(t\)</span> of the population obeys a certain probability distribution, and the <span class="math inline">\(W\)</span> is also a random variable that obeys a certain probability distribution.</p>
<p><strong>PH &amp; AFT</strong></p>
<ul>
<li>AFT Model is based on time T modeling, and PH Model is based on Hazard modeling</li>
<li>Weibull regression is the only regression that satisfies both model assumptions</li>
</ul>
<p>For PH Model: <span class="math display">\[\quad \lambda_{i}(t)=\lambda_{0}(t) \exp \left(x_{i}^{T} \beta\right)\]</span>
For AFT Model: <span class="math display">\[\lambda_{i}(t)=\lambda_{0}\left(\exp \left(-x_{i}^{T} \beta\right) t\right) \exp \left(-x_{i}^{T} \beta\right)\]</span></p>
<p>这是由于 <span class="math inline">\(T_{i}=\exp \left(x_{i}^{T} \beta\right) T_{0}\)</span>, 因此
<span class="math display">\[
S_{i}(t)=P\left(T_{i} \geq t\right)=P\left(T_{0} \geq \exp \left(-x_{i}^{T} \beta\right) t\right)=S_{0}\left(\exp \left(-x_{i}^{T} \beta\right) t\right)
\]</span>
<span class="math display">\[
f_{i}(t)=f_{0}\left(\exp \left(-x_{i}^{T} \beta\right) t\right) \exp \left(-x_{i}^{T} \beta\right)
\]</span>
<span class="math display">\[
\lambda_{i}(t)=\exp \left(-x_{i}^{T} \beta\right) \lambda_{0}\left(\exp \left(-x_{i}^{T} \beta\right) t\right)
\]</span></p>
<p>Intuitively, the larger the <span class="math inline">\(x_{i}^{T} \beta\)</span>, the longer the survival time. Since S is a monotonic non-decreasing function, the survival curve is higher. From a logarithmic perspective</p>
<p>PH Model:
<span class="math display">\[
\log \lambda_{i}(t)=\log \lambda_{0}(\exp (\log t))+x_{i}^{T} \beta
\]</span>
AFT Model:
<span class="math display">\[
\log \lambda_{i}(t)=\log \lambda_{0}\left(\exp \left(\log t-x_{i}^{T} \beta\right)\right)-x_{i}^{T} \beta
\]</span></p>
</div>
<div id="proc-lifereg" class="section level3" number="17.4.2">
<h3><span class="header-section-number">17.4.2</span> PROC LIFEREG</h3>
<p>The LIFEREG program uses the maximum likelihood method to generate the estimated value of the parameter regression model using censored survival data [method of maximum likelihood]. To some extent, PROREG LIFEREG has been obscured by the PHREG program, which uses a method known as partial likelihoo for semiparametric regression analysis [method known as partial likelihoo].</p>
<p>PROC LIFEREG is by no means obsolete. It can do some better things than PROC PHREG, and it can do other things that PROC PHREG simply cannot do:</p>
<ul>
<li>PROC LIFEREG accommodates left censoring and interval censoring. PROC PHREG allows only right censoring.</li>
<li>With PROC LIFEREG, you can test certain hypotheses about the shape of the hazard function. PROC PHREG gives you only nonparametric estimates of the survivor function, which can be difficult to interpret.</li>
<li>If the shape of the survival distribution is known, PROC LIFEREG produces more efficient estimates (with smaller standard errors) than PROC PHREG.</li>
<li>PROC LIFEREG can easily generate predicted event times for any specified set of covariate values. This is more difficult with PHREG, and often impossible.</li>
<li>PROC LIFEREG’s greatest limitation is that it does not handle time-dependent covariates, something at which PROC PHREG excels.</li>
</ul>
<!-- LIFEREG程序使用最大似然方法使用删失的生存数据生成参数回归模型的估计值 [method of maximum likelihood]。在某种程度上，PROREG LIFEREG已被PHREG程序所遮盖，该程序使用称为偏似然的方法进行半参数回归分析 [method known as partial likelihoo]。 -->
<!-- PROC LIFEREG绝不是过时的。它可以做一些比PROC PHREG更好的事情，它可以做PROC PHREG根本无法做的其他事情： -->
<!-- PROC LIFEREG可以容纳左检查和间隔检查。 -->
<!-- 使用PROC LIFEREG，您可以检验关于危害函数形状的某些假设。 PROC PHREG仅给您幸存者功能的非参数估计，这可能很难解释。 -->
<!-- 如果已知生存分布的形状，则PROC LIFEREG会比PROC PHREG产生更有效的估计（标准误差较小）。  -->
<!-- PROC LIFEREG可以轻松地为任何指定的一组协变量值生成预测事件时间。对于PHREG，这更困难，而且通常是不可能的。  -->
<!-- PROC LIFEREG的最大局限性在于它不能处理与时间相关的协变量，而PROC PHREG擅长于此。 -->
<p>What PROC LIFEREG actually estimates is a special case of this model that is quite similar in form to an ordinary linear regression model. Let <span class="math inline">\(T_{i}\)</span> be a random variable denoting the event time for the ith individual in the sample, and let <span class="math inline">\(x_{i 1}, \ldots, x_{i k}\)</span> be the values of <span class="math inline">\(k\)</span> covariates for that same individual. The model is then
<span class="math display">\[
\log T_{i}=\beta_{0}+\beta_{1} x_{i 1}+\ldots+\beta_{k} x_{i k}+\sigma \varepsilon_{i}
\]</span>
where <span class="math inline">\(\varepsilon_{i}\)</span> is a random disturbance term, and <span class="math inline">\(\beta_{0}, \ldots, \beta_{k}\)</span>, and <span class="math inline">\(\sigma\)</span> are parameters to be estimated. Exponentiating both sides gives an alternative way of expressing the model:
<span class="math display">\[
T_{i}=\exp \left(\beta_{0}+\beta_{1} x_{i 1}+\ldots+\beta_{k} x_{i k}+\sigma \varepsilon_{i}\right)
\]</span>
In regression analysis, <span class="math inline">\(-\log (T)\)</span> represents fixed effects, and <span class="math inline">\(\epsilon\)</span> represents noise. Different <span class="math inline">\(\epsilon\)</span> distribution means different distribution of <span class="math inline">\(T_{0}\)</span>.</p>
<p>According to a simple transformation, we have <span class="math inline">\(T_{i}=\exp \left(x_{i}^{T} \beta\right) T_{0}, T_{0}=\exp (W)\)</span>. Therefore, the AFT model parameter <span class="math inline">\(\beta\)</span> represents: when the jth dimension changes in <span class="math inline">\(\Delta_{j}\)</span>, the survival time will change <span class="math inline">\(\exp \left(\Delta_{j} \beta_{j}\right )\)</span></p>
</div>
<div id="residuum-distribution" class="section level3" number="17.4.3">
<h3><span class="header-section-number">17.4.3</span> Residuum distribution</h3>
<p>Assuming <span class="math inline">\(Y=\log T\)</span>, the AFT Model can be expressed as
<span class="math display">\[
Y_{i}=x_{i}^{T} \beta+W_{i}
\]</span>
<span class="math inline">\(W_{i} \sim f\)</span> represents the <strong>independent residual</strong></p>
<ul>
<li>If <span class="math inline">\(W \sim N\left(0, \sigma^{2}\right)\)</span>, the model at this time is called <strong>Lognormal-AFT Model</strong>. Because the assumption of normality for <span class="math inline">\(Y\)</span> is equivalent to the assumption of lognormality. At this time, if there is no censorship, we can directly use the LSE (least squares) method to solve the problem (in fact, there is rarely such a possibility). Survival data usually has at least some vetted observations, and it is difficult to process these observations with <strong>OLS (Ordinary Least Squares)</strong>. Instead, we can use <strong>maximum likelihood estimation</strong>.</li>
<li>If <span class="math inline">\(W \sim W \operatorname{libull}(\lambda, \gamma)\)</span>, the model at this time is called <strong>Weibull regression model</strong>. It is a special AFT Model.</li>
<li>Following surmmaries the model</li>
</ul>
<p><span class="math display">\[
\begin{array}{c|l}
\text { Distribution of } \varepsilon &amp; \text { Distribution of } T \\
\hline \text { extreme value (2 par.) } &amp; \text { Weibull } \\
\text { extreme value (1 par.) } &amp; \text { exponential } \\
\text { log-gamma } &amp; \text { gamma } \\
\text { logistic } &amp; \text { log-logistic } \\
\text { normal } &amp; \text { log-normal }
\end{array}
\]</span></p>
<p><strong>Specify Model in SAS</strong></p>
<pre><code>*** Lognormal-AFT Model;
PROC LIFEREG DATA=recid;
 MODEL week*arrest(0)=fin age race wexp mar paro prio / DISTRIBUTION=LNORMAL;
RUN;</code></pre>
</div>
<div id="exponential-model" class="section level3" number="17.4.4">
<h3><span class="header-section-number">17.4.4</span> Exponential Model</h3>
<p>The simplest model estimated by PROC LIFEREG is the exponential model, which is called by DISTRIBUTION = EXPONENTIAL in the MODEL statement. The model specifies that <span class="math inline">\(\varepsilon\)</span> has a standard extreme-value distribution and constrains <span class="math inline">\(\sigma=1\)</span>. If <span class="math inline">\(\varepsilon\)</span> has an extreme value distribution, then the logarithm T also has an extreme value distribution, provided that the covariate is the condition.
Constant hazard function, expresses the exponential regression model as
<span class="math display">\[
\log h(t)=\beta_{0}^{\bullet}+\beta_{1}^{\bullet} x_{1}+\ldots+\beta_{k}^{\bullet} x_{k}
\]</span></p>
<p><strong>Interpretation</strong></p>
<p><span class="math display">\[\lambda_{i}(t)=\lambda(t) \exp \left(x_{i}^{T} \beta\right), \lambda(t) \geq 0\]</span>
Where <span class="math inline">\(\lambda(t)=\lambda\)</span>
<span class="math display">\[\frac{\lambda_{1}(t)}{\lambda_{2}(t)}=\exp \left(\Delta_{j} \beta_{j}\right), \Delta_{j}=x_{1 j}-x_{2 j}\]</span>
Therefore, the meaning of the parameter <span class="math inline">\(\beta_{j}\)</span> is: when the data <span class="math inline">\(x_{j}\)</span> changes with the size of <span class="math inline">\(\Delta_{j}\)</span>, the Hazard Ratio of the model will change <span class="math inline">\(\exp \left(\Delta_{ j} \beta_{j}\right)\)</span></p>
</div>
<div id="weibull-model" class="section level3" number="17.4.5">
<h3><span class="header-section-number">17.4.5</span> Weibull Model</h3>
<p>The Weibull model is a slight modification of the exponential model with great consequences. By specifying DISTRIBUTION = WEIBULL in the MODEL statement, we retain the assumption that <span class="math inline">\(\varepsilon\)</span> has a standard extreme value distribution, but <strong>relax the assumption of <span class="math inline">\(\sigma=1\)</span>.</strong></p>
<ul>
<li>When <span class="math inline">\(\sigma\)</span>&gt; 1, the danger decreases with time.</li>
<li>When <span class="math inline">\(0.5&lt;\sigma&lt;1\)</span>, the risk is increasing at a reduced rate.</li>
<li>When <span class="math inline">\(0&lt;\sigma&lt;0.5\)</span>, the danger is increasing at an increasing rate.</li>
<li>-When <span class="math inline">\(\sigma=0.5\)</span>, the hazard function is a straight line and its starting point is 0.</li>
</ul>
<p><strong>Weibull Distribution</strong></p>
<p><span class="math display">\[
\begin{array}{l}
\lambda(t)=\lambda \gamma(t \lambda)^{\gamma-1} \\
\Lambda(t)=(t \lambda)^{\gamma} \\
S(t)=\exp \left[-(t \lambda)^{\gamma}\right] \\
f(t)=\lambda \gamma(t \lambda)^{\gamma-1} \exp \left((t \lambda)^{\gamma}\right)
\end{array}
\]</span></p>
<p><strong>Connection with exponential distribution</strong></p>
<p>If <span class="math inline">\(X \sim \operatorname{Exp}(\tau)\)</span>, then <span class="math inline">\(T=X^{\sigma} \sim\)</span> Weibull <span class="math inline">\(\left(\gamma=1 / \sigma, \lambda=\tau^{\ sigma}\right)\)</span>
also,
- For exponential distribution, <span class="math inline">\(\quad \log S(t)=(-\log t)+C_(1)\)</span>
- For Weibull distribution, <span class="math inline">\(\quad \log S(t)=\gamma(-\log t)+C_{2}\)</span></p>
<p>T has a Weibull distribution conditioned on covariates.it has a relatively simple survival function, which is easy to manipulate mathematically: (where <span class="math inline">\(\mathbf{x}_{i}\)</span> is a vector of the covariate values and $ $ is a vector of coefficients.)
<span class="math display">\[
S_{i}(t)=\exp \left\{-\left[t_{i} e^{-\beta \mathbf{x}_{i}}\right]^{\frac{1}{\sigma}}\right\}
\]</span>
Secondly, in addition to the AFT model, the Weibull model is also a proportional hazard model. This means that its coefficient (after proper conversion) can be interpreted as a relative hazard ratio. In fact, the Weibull model (and its special case, the exponential model) is the only model that belongs to both categories. Like the exponential model, there is a completely equivalent relationship between the logarithmic risk forms of the model
<span class="math display">\[
\log h(t)=\alpha \log t+\beta_{0}^{\bullet}+\beta_{1}^{\bullet} x_{1}+\ldots+\beta_{k}^{\bullet} x_{k}
\]</span>
and the log-survival time model
<span class="math display">\[
\log T_{i}=\beta_{0}+\beta_{1} x_{1}+\ldots+\beta_{k} x_{k}+\sigma \varepsilon
\]</span>
However, the relationship between the parameters is slightly more complicated. Specifically, for the Weibull model
<span class="math display">\[
\beta_{j}^{\bullet}=\frac{-\beta_{j}}{\sigma} \text { for } j=1, \ldots, k
\]</span>
<span class="math display">\[\alpha=(1 / \sigma)-1.\]</span></p>
<pre><code>## Output
## σˆ (labeled “Scale” 0.7124  in Output) is between 0 and 1
## calculate αˆ = (1/.7124)–1=0.4037, which is the coefficient for log t 
   in the log-hazard model. 
## “Weibull Shape.” This is not an independent parameter but merely 
   the reciprocal of the Shape parameter (that is, 1/σˆ ). </code></pre>
</div>
<div id="log-normal-model" class="section level3" number="17.4.6">
<h3><span class="header-section-number">17.4.6</span> Log-Normal Model</h3>
<p>Unlike the Weibull model, the lognormal model has a non-monotonic hazard function. When t=0, the danger is 0. It rises to a peak, and then drops to 0 as t approaches infinity. The log normal is not a proportional hazard model, and its risk function cannot be expressed in a closed form. However, it can be expressed as a regression model, where the dependent variable is the log hazard.
<span class="math display">\[
\log h(t)=\log h_{0}\left(t e^{-\beta x}\right)-\beta \mathbf{x}
\]</span>
where <span class="math inline">\(h_{0}\)</span> (.) can be interpreted as the hazard function for an individual with <span class="math inline">\(\mathbf{x}=\mathbf{0}\)</span>. (personal hazard function)</p>
</div>
<div id="log-logistic-model" class="section level3" number="17.4.7">
<h3><span class="header-section-number">17.4.7</span> Log-Logistic Model</h3>
<p>An inverted U-shaped hazard is the log-logistic model, which assumes that ε has a logistic distribution with
<span class="math display">\[
f(\varepsilon)=\frac{e^{\varepsilon}}{\left(1+e^{\varepsilon}\right)^{2}}
\]</span></p>
<p>DISTRIBUTION=LLOGISTIC, The logical distribution is symmetrical, with an average value of 0, and is very similar in shape to the normal distribution. If <span class="math inline">\(\varepsilon\)</span> has a logarithmic distribution, then the logarithm <span class="math inline">\(T\)</span> (although the mean is non-zero) has a logarithmic distribution. The loglogistic hazard function is
<span class="math display">\[
h(t)=\frac{\lambda \gamma(\lambda t)^{\gamma-1}}{1+(\lambda t)^{\gamma}}
\]</span>
where <span class="math inline">\(\gamma=1 / \sigma\)</span>, and
<span class="math display">\[
\lambda=\exp \left\{-\left[\beta_{0}+\beta_{1} x_{1}+\ldots+\beta_{k} x_{k}\right]\right\}
\]</span></p>
<ul>
<li>When <span class="math inline">\(\sigma&lt;1\)</span>, the logarithmic logic hazard is similar to the lognormal hazard: starting from 0, rising to the peak, and then falling to 0.</li>
<li>When <span class="math inline">\(\sigma&gt;1\)</span>, the hazard behaves like a diminishing Weibull hazard: it starts at infinity and then decreases towards 0.</li>
<li>When <span class="math inline">\(\sigma=1\)</span>, the hazard value at t <span class="math inline">\(=0\)</span> is <span class="math inline">\(\lambda\)</span>, and then decreases toward 0 as t tends to infinity.</li>
</ul>
<p>Although the logarithmic logic model has the complexity of the hazard function, it has a fairly simple survival function
<span class="math display">\[
S(t)=\frac{1}{1+(\lambda t)^{\gamma}}
\]</span>
As before, <span class="math inline">\(\gamma=1 / \sigma\)</span> and <span class="math inline">\(\lambda=\exp \left\{-\left[\beta_{0}+\beta_{1} x_{1}+\ldots+\beta_{k} x_{k}\right]\right\} .\)</span> A little algebra shows that this can be written as
<span class="math display">\[
\log \left[\frac{S(t)}{1-S(t)}\right]=\beta_{0}^{\bullet}+\beta_{1}^{\bullet} x_{1}+\ldots+\beta_{k}^{\bullet} x_{k}-\gamma \log t
\]</span>
Where <span class="math inline">\(\beta_{i}^{\bullet}=\beta_{i} / \sigma\)</span> for <span class="math inline">\(i=1, \ldots, k\)</span>. 这只是事件 <span class="math inline">\(t\)</span> 之前发生的概率的逻辑回归模型。</p>
</div>
<div id="fit-statistics-for-model-comparsion" class="section level3" number="17.4.8">
<h3><span class="header-section-number">17.4.8</span> Fit statistics for model comparsion</h3>
<p><strong>The statistical information largely depends on the sample size.</strong></p>
<ul>
<li>Akaike’s information criterion (AIC), is a modification of the <span class="math inline">\(-2\)</span> log-likelihood that penalizes models for having more covariates. where <span class="math inline">\(\mathrm{k}\)</span> is the number of covariates.
<span class="math display">\[
A I C=-2 \log L+2 k
\]</span></li>
<li>(AICC) is a “corrected” version of the AIC that may have better behavior in small samples:
<span class="math display">\[
A I C C=A I C+\frac{2 k(k+1)}{n-k-1}
\]</span></li>
<li>Bayesian information criterion (also known as Schwarz’s criterion) gives a more severe penalization for additional covariates (对其他协变量进行更严厉的惩罚)
<span class="math display">\[
B I C=-2 \log L+k \log n
\]</span></li>
</ul>
</div>
<div id="graphical-method-for-distinguishing-different-distributions-in-sas" class="section level3" number="17.4.9">
<h3><span class="header-section-number">17.4.9</span> Graphical method for distinguishing different distributions in SAS</h3>
<ul>
<li>the PLOTS=LS option produces a plot of -log and t. If the true distribution is an exponential distribution, the graph should produce a straight line with an origin of 0</li>
<li>The LLS option produces a plot of −loglog versus log t, if the true distribution is Weibull, it should be a straight line;</li>
</ul>
</div>
<div id="prediction-and-hazard-function" class="section level3" number="17.4.10">
<h3><span class="header-section-number">17.4.10</span> Prediction and hazard function</h3>
<p>After fitting the model with PROC LIFEREG, sometimes it is desirable to generate predicted survival times for the observations in the data set. If a single point estimate is to be made for each person, the predicted median survival time may be the best. You can easily get this information using the OUTPUT statement.</p>
<p>P in the OUTPUT statement = option request percentile. By default, PROC LIFEREG calculates the 50th percentile (ie median). You can use the QUANTILE keyword to request other percentiles, as described in the PROC LIFEREG documentation. The STD keyword requests the standard error of the median.</p>
<pre><code>PROC LIFEREG DATA=recid;
 MODEL week*arrest(0)=fin age race wexp mar paro prio
 / D=WEIBULL;
 OUTPUT OUT=a P=median STD=s;
RUN;</code></pre>
<!-- 还可以获取不在原始数据集中的一组协变量值的预测值。在估算模型之前，只需将具有所需协变量值且事件时间设置为缺失的其他观测值附加到数据集即可。
这些添加的观察值不用于估计模型，但是将为它们生成预测值。研究人员通常不希望预测生存时间，而是要预测生存到指定时间的概率（例如5年生存概率）。
虽然这些不是由PROC LIFEREG直接计算的，但计算起来却相当简单。这是通过将可由OUTPUT语句生成的线性预测值替换为幸存函数的公式来实现的 %PREDICT. -->
<!-- 使用OUTEST =选项将参数估计值写入数据集。接下来，在OUTPUT语句中使用XBETA =选项请求将线性预测变量包含在第二个数据集中。最后，调用宏，指示两个数据集的名称，分配给线性预测变量的名称以及计算生存概率的时间。 -->
<p><strong>Predict the probability of survival to a specified time</strong></p>
<pre><code>PROC LIFEREG DATA=recid OUTEST=a;
 MODEL week*arrest(0) = fin age race wexp mar paro prio
 / D=WEIBULL;
 OUTPUT OUT=b XBETA=lp;
RUN;
%PREDICT(OUTEST=a,OUT=b,XBETA=lp,TIME=30)


%macro predict (outest=, out=_last_,xbeta=,time=);
 data _pred_;
 _p_=1;
 set &amp;outest point=_p_;
 set &amp;out;
 lp=&amp;xbeta;
 t=&amp;time;
 gamma=1/_scale_;
 alpha=exp(-lp*gamma);
 prob=0;
 _dist_=upcase(_dist_);
 if _dist_=&#39;WEIBULL&#39; or _dist_=&#39;EXPONENTIAL&#39; or _dist_=&#39;EXPONENT&#39; then prob=exp(-
 alpha*t**gamma);
 if _dist_=&#39;LOGNORMAL&#39; or _dist_=&#39;LNORMAL&#39; then prob=1-probnorm((log(t)-lp)/_scale_);
 if _dist_=&#39;LLOGISTIC&#39; or _dist_=&#39;LLOGISTC&#39; then prob=1/(1+alpha*t**gamma);
 if _dist_=&#39;GAMMA&#39; then do;
  d=_shape1_;
  k=1/(d*d);
  u=(t*exp(-lp))**gamma;
  prob=1-probgam(k*u**d,k);
  if d lt 0 then prob=1-prob;
  end;
 drop lp gamma alpha _dist_ _scale_ intercept
  _shape1_ _model_ _name_ _type_ _status_ _prob_ _lnlike_ d k u;
 run;
 proc print data=_pred_;
 run;
%mend predict;

*** %LIFEHAZ generated a graph of hazard versus time;
PROC LIFEREG DATA=recid OUTEST=a;
 MODEL week*arrest(0) = fin age race wexp mar paro prio
 / D=WEIBULL;
 OUTPUT OUT=b XBETA=lp;
RUN;
%LIFEHAZ(OUTEST=a,OUT=b,XBETA=lp)

%macro lifehaz(outest=,out=,obsno=0,xbeta=lp);
data;
 set &amp;outest;
 call symput(&#39;time&#39;,_NAME_);
run;
proc means data=&amp;out noprint;
 var &amp;time &amp;xbeta;
 output out=_c_ min(&amp;time)=min max(&amp;time)=max mean(&amp;xbeta)=mean;
run;
data;
 set &amp;outest;
 call symput(&#39;model&#39;,_dist_);
 s=_scale_;
 d=_shape1_;
 _y_=&amp;obsno;
 set _c_ (keep=min max mean);
 if _y_=0 then m=mean;
 else do;
 set &amp;out (keep=&amp;xbeta) point=_y_;
 m=&amp;xbeta;
 end;
 inc=(max-min)/300;
 g=1/s;
 alph=exp(-m*g);
 _dist_=upcase(_dist_);
if _dist_=&#39;LOGNORMAL&#39; or _dist_=&#39;LNORMAL&#39; then do;
 do t=min to max by inc;
 z=(log(t)-m)/s;
 f=exp(-z*z/2)/(t*s*sqrt(2*3.14159));
 Surv=1-probnorm(z);
 h=f/Surv;
 output;
 end;
end;
else if _dist_=&#39;GAMMA&#39; then do;
 k=1/(d*d);
 do t=min to max by inc;
 u=(t*exp(-m))**(1/s);
 f=abs(d)*(k*u**d)**k*exp(-k*u**d)/(s*gamma(k)*t);
 Surv=1-probgam(k*u**d,k);
 if d lt 0 then Surv=1-Surv;
 h=f/Surv;
 output;
 end;
end;
else if _dist_=&#39;WEIBULL&#39; or _dist_=&#39;EXPONENTIAL&#39; or _dist_=&#39;EXPONENT&#39; then do;
 do t=min to max by inc;
 h=g*alph*t**(g-1);
 output;
 end;
end;
else if _dist_=&#39;LLOGISTIC&#39; or _dist_=&#39;LLOGISTC&#39; then do;
 do t=min to max by inc;
 h=g*alph*t**(g-1)/(1+alph*t**g);
 output;
 end;
end;
else put &#39;ERROR:DISTRIBUTION NOT FITTED BY LIFEREG&#39;;
run;
proc gplot;
 plot h*t / haxis=axis2 vaxis=axis1 vzero;
 symbol1 i=join v=none c=black;
 axis1 label=(f=titalic angle=90 &#39;Hazard&#39;);
 axis2 label=(f=titalic justify=c &#39;time&#39; f=titalic justify=c &quot;&amp;model&quot;);
run; quit;
%mend lifehaz;</code></pre>
</div>
<div id="left-censoring-and-interval-censoring" class="section level3" number="17.4.11">
<h3><span class="header-section-number">17.4.11</span> Left Censoring and Interval Censoring</h3>
<p>PROC LIFEREG excludes any observations with a time of 0 or less, because it must take the logarithm of the event time as the first step. One way is to assign some arbitrarily chosen number between 0 and 1. A more elegant solution is to treat this situation as being censored at time 1.</p>
<!-- PROC LIFEREG排除时间为0或更短的任何观测值，因为它必须将事件时间的对数作为第一步。一种方法是在0到1之间分配一些任意选择的数字。一种更优雅的解决方案是将这种情况视为在时间1处被审查。 -->
<ul>
<li>For uncensored observations, LOWER and UPPER have the same value.</li>
<li>For observations with YEARS=0, we set LOWER=. and UPPER=1.</li>
<li>For right-censored observations (including those with events other than the one of interest), UPPER=. and LOWER=YEARS.</li>
<li>If both UPPER and LOWER are missing (which happens for individuals with YEARS=0 who have events other than the one of interest), the observation is excluded. T</li>
</ul>
<pre><code>DATA leaders2;
 SET leaders;
 lower=years;
 upper=years;
 IF years=0 THEN DO;
 lower=.;
 upper=1;
 END;
 IF lost IN (0,1,2) THEN upper=.;
RUN;

*** fit an exponential model;
PROC LIFEREG DATA=leaders2;
 CLASS region;
 MODEL (lower,upper)= manner start military age conflict
 loginc literacy region / D=EXPONENTIAL;
RUN;</code></pre>
</div>
</div>
<div id="cox-proportional-hazards-model" class="section level2" number="17.5">
<h2><span class="header-section-number">17.5</span> Cox Proportional Hazards Model</h2>
<div id="introduction-13" class="section level3" number="17.5.1">
<h3><span class="header-section-number">17.5.1</span> Introduction</h3>
<p>The basic model that does not include time-dependent covariates or nonproportional hazards. The model is usually written as
<span class="math display">\[h_{i}(t)=\lambda_{0}(t) \exp \left(\beta_{1} x_{i 1}+\ldots+\beta_{k} x_{i k}\right)\]</span>
Taking the logarithm of both sides, we can rewrite the model as
<span class="math display">\[\log h_{i}(t)=\alpha(t)+\beta_{1} x_{i 1}+\ldots+\beta_{k} x_{i k}\]</span>
Take the ratio of the hazards for two individuals <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>
<span class="math display">\[\frac{h_{i}(t)}{h_{j}(t)}=\exp \left\{\beta_{1}\left(x_{i 1}-x_{j}\right)+\ldots+\beta_{k}\left(x_{i k}-x_{j k}\right)\right\}\]</span></p>
<p><strong>Partial Likelihood</strong></p>
<p>尚若不考虑删失, <span class="math inline">\(\quad j\)</span> 个观测 <span class="math inline">\(T_{1} \sim \lambda_{1}(t), T_{2} \sim \lambda_{2}(t), \ldots T_{j} \sim \lambda_{j}(t)\)</span> 在Cox Regression的假设下满
足
<span class="math display">\[
P\left(T_{1}&lt;T_{2}&lt;\ldots&lt;T_{J}\right)=\prod_{i=1}^{J} \frac{\lambda_{j}(t)}{\sum_{k=j}^{J} \lambda_{k}(t)}=\prod_{i=1}^{J} \frac{\exp \left(x_{j}^{T} \beta\right)}{\sum_{k=j}^{J} \exp \left(x_{k}^{T} \beta\right)}
\]</span>
Furthermore, when the censor be considered, the time point of death event is denoted as <span class="math inline">\(t_{j}\)</span>, then the <span class="math inline">\(\mathrm{P}(death)\)</span>
<span class="math display">\[
\mathrm{P}(death)=\frac{\exp \left(x_{j}^{T} \beta\right)}{\sum_{k \in R\left(t_{j}\right)} \exp \left(x_{k} \beta\right)}
\]</span>
given the subject’s covariates <span class="math inline">\(\beta\)</span>; the joint probability of observing each subject’s failure time can be calculated given the subject’s covariates values <span class="math inline">\(x_{j}\)</span> likelihood <span class="math inline">\(L(\beta)\)</span> could be expressed as
<span class="math display">\[
L(\beta)=\prod_{j} \frac{\exp \left(x_{j}^{T} \beta\right)}{\sum_{k \in R\left(t_{j}\right)} \exp \left(x_{k} \beta\right)}
\]</span>
<span class="math inline">\(k \in R\left(t_{j}\right)\)</span> represents at time <span class="math inline">\(t_{j}-\)</span> the subject <span class="math inline">\(k\)</span> is still alive. Here we temporarily assume that there is only one death observation at time <span class="math inline">\(t_{j}\)</span>.</p>
<p>Strictly speaking, this is not a likelihood function. <strong>Because the likelihood function should represent the continuous multiplication of the distribution function under the given data</strong>, but the above formula is obviously not. Even speaking, the continuous multiplication should be about the continuous multiplication of the individual, and the above formula is about the point in time. Cox (1975) called this form of likelihood Partial Likelihood, and proved that all its properties are the same as likelihood.</p>
</div>
<div id="parameter-estimate" class="section level3" number="17.5.2">
<h3><span class="header-section-number">17.5.2</span> Parameter estimate</h3>
<div id="nr-iteration-for-parameter-estimate" class="section level4" number="17.5.2.1">
<h4><span class="header-section-number">17.5.2.1</span> NR iteration for parameter estimate</h4>
<p>The iteration form is
<span class="math display">\[
\hat{\beta}^{(r+1)}=\hat{\beta}^{(r)}+\left(X^{T} WX\right)^{-1} X^{T} (dP d)
\]</span></p>
<ul>
<li><span class="math inline">\(w_{i}=\exp \left(x_{i}^{T} \beta\right)\)</span></li>
<li><span class="math inline">\(Y_{i}\left(t_{j}\right)\)</span> 1 if the individual is still alive at the moment <span class="math inline">\(t_{j}\)</span>, <span class="math inline">\(Y_{i}\left(t_{j}\right)\)</span> Is the index of an individual’s survival at a certain time <span class="math inline">\(t_{j}\)</span></li>
<li><span class="math inline">\(\pi_{ij}=Y_{i}\left(t_{j}\right) \frac{w_{i}}{\sum_{k \in R\left(t_{j}\right)} w_ {k}}=Y_{i}\left(t_{j}\right) \frac{w_{i}}{\sum_{k=1}^{n} Y_{k}\left(t_{j} \right) w_{k}}\)</span></li>
<li><span class="math inline">\(P=\left\{\pi_{i j}\right\}_{i, j}\)</span>, represents the relative death risk of the individual <span class="math inline">\(i\)</span> at time <span class="math inline">\(t_{j}\)</span>. So we only consider individuals who are still alive</li>
<li><span class="math inline">\(W_{k k}=-\sum_{i} \delta_{i} \pi_{k i}\left(1-\pi_{k i}\right), W_{k k}\)</span> represents the weighting of the individual. Whenever one of the remaining individuals <span class="math inline">\(i\)</span> dies, if <span class="math inline">\(k\)</span> is still alive, its weight increases by <span class="math inline">\(\pi(1-\pi)\)</span></li>
<li><span class="math inline">\(W_{k j}=\sum_{i} \delta_{i} \pi_{k i} \pi_{j i}, k j\)</span> represents the interaction of the individual <span class="math inline">\(k\)</span>. Whenever one of the remaining individuals dies, if <span class="math inline">\(k, j\)</span> are still alive <span class="math inline">\((\)</span> Y will
Set), the interaction weight increases <span class="math inline">\(\pi_(k) \pi_(j)\)</span>
Mehr zu diesem AusgangstextFür weitere Übersetzungsinformationen ist ein Ausgangstext erforderlich
Feedback geben
Seitenleisten</li>
</ul>
</div>
<div id="model-fitting-in-sas" class="section level4" number="17.5.2.2">
<h4><span class="header-section-number">17.5.2.2</span> Model fitting in SAS</h4>
<pre><code>proc phreg data = whas500;
    class gender;
    model lenfol*fstat(0) = gender age;
run;

## Output
## “Ties Handling: BRESLOW.” 
   This line refers to the default method for handling ties—two or more 
   observations that have exactly the same event time. Although Breslow’s method 
   is nearly universal

## Analysis of Maximum Likelihood Estimates
   
## Model Fit Statistics
## -2 LOG L
## AIC
## SBC: Schwartz’s Bayesian criterion (SBC) statistics 
   The main purpose is to punish models with more covariates, allowing people to compare non-nested models
   
## Testing Global Null Hypothesis
## Displays test of hypothesis that all coefficients in the model are 0
   模型整体是否可以预测危险率变化的整体检验。这些检验是渐近的等价比, 一般首选似然比检验
## Likelihood Ratio
## Socre
## Wald</code></pre>
</div>
<div id="model-fitting-in-r" class="section level4" number="17.5.2.3">
<h4><span class="header-section-number">17.5.2.3</span> Model fitting in R</h4>
<p><strong>Multivariate Cox regression analysis</strong></p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="survival-analysis.html#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;survival&quot;</span>)</span>
<span id="cb55-2"><a href="survival-analysis.html#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;survminer&quot;</span>)</span>
<span id="cb55-3"><a href="survival-analysis.html#cb55-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-4"><a href="survival-analysis.html#cb55-4" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;lung&quot;</span>)</span>
<span id="cb55-5"><a href="survival-analysis.html#cb55-5" aria-hidden="true" tabindex="-1"></a>res.cox <span class="ot">&lt;-</span> <span class="fu">coxph</span>(<span class="fu">Surv</span>(time, status) <span class="sc">~</span> age <span class="sc">+</span> sex <span class="sc">+</span> ph.ecog, <span class="at">data =</span>  lung)</span>
<span id="cb55-6"><a href="survival-analysis.html#cb55-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res.cox)</span></code></pre></div>
<pre><code>## Call:
## coxph(formula = Surv(time, status) ~ age + sex + ph.ecog, data = lung)
## 
##   n= 227, number of events= 164 
##    (1 observation deleted due to missingness)
## 
##              coef exp(coef)  se(coef)      z Pr(&gt;|z|)    
## age      0.011067  1.011128  0.009267  1.194 0.232416    
## sex     -0.552612  0.575445  0.167739 -3.294 0.000986 ***
## ph.ecog  0.463728  1.589991  0.113577  4.083 4.45e-05 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
##         exp(coef) exp(-coef) lower .95 upper .95
## age        1.0111     0.9890    0.9929    1.0297
## sex        0.5754     1.7378    0.4142    0.7994
## ph.ecog    1.5900     0.6289    1.2727    1.9864
## 
## Concordance= 0.637  (se = 0.025 )
## Likelihood ratio test= 30.5  on 3 df,   p=1e-06
## Wald test            = 29.93  on 3 df,   p=1e-06
## Score (logrank) test = 30.5  on 3 df,   p=1e-06</code></pre>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="survival-analysis.html#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Visualizing the estimated distribution of survival times</span></span>
<span id="cb57-2"><a href="survival-analysis.html#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot the baseline survival function</span></span>
<span id="cb57-3"><a href="survival-analysis.html#cb57-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(<span class="fu">survfit</span>(res.cox), <span class="at">data=</span>lung, </span>
<span id="cb57-4"><a href="survival-analysis.html#cb57-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> <span class="st">&quot;#2E9FDF&quot;</span>,</span>
<span id="cb57-5"><a href="survival-analysis.html#cb57-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">ggtheme =</span> <span class="fu">theme_minimal</span>())</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/Cox%20model%20fitting%20in%20R-1.png" width="672" /></p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="survival-analysis.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Assess the impact of gender on the estimated survival rate</span></span>
<span id="cb58-2"><a href="survival-analysis.html#cb58-2" aria-hidden="true" tabindex="-1"></a>sex_df <span class="ot">&lt;-</span> <span class="fu">with</span>(lung,</span>
<span id="cb58-3"><a href="survival-analysis.html#cb58-3" aria-hidden="true" tabindex="-1"></a>               <span class="fu">data.frame</span>(<span class="at">sex =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), </span>
<span id="cb58-4"><a href="survival-analysis.html#cb58-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">age =</span> <span class="fu">rep</span>(<span class="fu">mean</span>(age, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="dv">2</span>),</span>
<span id="cb58-5"><a href="survival-analysis.html#cb58-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ph.ecog =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>)</span>
<span id="cb58-6"><a href="survival-analysis.html#cb58-6" aria-hidden="true" tabindex="-1"></a>                          )</span>
<span id="cb58-7"><a href="survival-analysis.html#cb58-7" aria-hidden="true" tabindex="-1"></a>               )</span>
<span id="cb58-8"><a href="survival-analysis.html#cb58-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Survival curves</span></span>
<span id="cb58-9"><a href="survival-analysis.html#cb58-9" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">survfit</span>(res.cox, <span class="at">newdata =</span> sex_df)</span>
<span id="cb58-10"><a href="survival-analysis.html#cb58-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(fit, <span class="at">data=</span>sex_df, <span class="at">conf.int =</span> <span class="cn">TRUE</span>, </span>
<span id="cb58-11"><a href="survival-analysis.html#cb58-11" aria-hidden="true" tabindex="-1"></a>           <span class="at">legend.labs=</span><span class="fu">c</span>(<span class="st">&quot;Sex=1&quot;</span>, <span class="st">&quot;Sex=2&quot;</span>),</span>
<span id="cb58-12"><a href="survival-analysis.html#cb58-12" aria-hidden="true" tabindex="-1"></a>           <span class="at">ggtheme =</span> <span class="fu">theme_minimal</span>())</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/Cox%20model%20fitting%20in%20R-2.png" width="672" /></p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="survival-analysis.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Stratified Cox regression</span></span>
<span id="cb59-2"><a href="survival-analysis.html#cb59-2" aria-hidden="true" tabindex="-1"></a>res.cox.strata <span class="ot">&lt;-</span> <span class="fu">coxph</span>(<span class="fu">Surv</span>(time, status) <span class="sc">~</span> age <span class="sc">+</span> <span class="fu">strata</span>(sex), <span class="at">data =</span>  lung)</span>
<span id="cb59-3"><a href="survival-analysis.html#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res.cox.strata)</span></code></pre></div>
<pre><code>## Call:
## coxph(formula = Surv(time, status) ~ age + strata(sex), data = lung)
## 
##   n= 228, number of events= 165 
## 
##         coef exp(coef) se(coef)     z Pr(&gt;|z|)  
## age 0.016215  1.016347 0.009187 1.765   0.0776 .
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
##     exp(coef) exp(-coef) lower .95 upper .95
## age     1.016     0.9839    0.9982     1.035
## 
## Concordance= 0.546  (se = 0.026 )
## Likelihood ratio test= 3.18  on 1 df,   p=0.07
## Wald test            = 3.12  on 1 df,   p=0.08
## Score (logrank) test = 3.12  on 1 df,   p=0.08</code></pre>
<p><strong>Univariate coxph function to multiple covariate</strong></p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="survival-analysis.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="do">##  apply the univariate coxph function to multiple covariates at once</span></span>
<span id="cb61-2"><a href="survival-analysis.html#cb61-2" aria-hidden="true" tabindex="-1"></a>covariates <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;age&quot;</span>, <span class="st">&quot;sex&quot;</span>,  <span class="st">&quot;ph.karno&quot;</span>, <span class="st">&quot;ph.ecog&quot;</span>, <span class="st">&quot;wt.loss&quot;</span>)</span>
<span id="cb61-3"><a href="survival-analysis.html#cb61-3" aria-hidden="true" tabindex="-1"></a>univ_formulas <span class="ot">&lt;-</span> <span class="fu">sapply</span>(covariates,</span>
<span id="cb61-4"><a href="survival-analysis.html#cb61-4" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">function</span>(x) <span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">&#39;Surv(time, status)~&#39;</span>, x)))</span>
<span id="cb61-5"><a href="survival-analysis.html#cb61-5" aria-hidden="true" tabindex="-1"></a>univ_models <span class="ot">&lt;-</span> <span class="fu">lapply</span>( univ_formulas, <span class="cf">function</span>(x){<span class="fu">coxph</span>(x, <span class="at">data =</span> lung)})</span>
<span id="cb61-6"><a href="survival-analysis.html#cb61-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-7"><a href="survival-analysis.html#cb61-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Surv(time, status) ~ age</span></span>
<span id="cb61-8"><a href="survival-analysis.html#cb61-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Surv(time, status) ~ sex</span></span>
<span id="cb61-9"><a href="survival-analysis.html#cb61-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Surv(time, status) ~ ph.karno</span></span>
<span id="cb61-10"><a href="survival-analysis.html#cb61-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Surv(time, status) ~ ph.ecog</span></span>
<span id="cb61-11"><a href="survival-analysis.html#cb61-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Surv(time, status) ~ wt.loss</span></span>
<span id="cb61-12"><a href="survival-analysis.html#cb61-12" aria-hidden="true" tabindex="-1"></a><span class="do">## Surv(time, status) ~ sex</span></span>
<span id="cb61-13"><a href="survival-analysis.html#cb61-13" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb61-14"><a href="survival-analysis.html#cb61-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract data </span></span>
<span id="cb61-15"><a href="survival-analysis.html#cb61-15" aria-hidden="true" tabindex="-1"></a>univ_results <span class="ot">&lt;-</span> <span class="fu">lapply</span>(univ_models,</span>
<span id="cb61-16"><a href="survival-analysis.html#cb61-16" aria-hidden="true" tabindex="-1"></a>                       <span class="cf">function</span>(x){ </span>
<span id="cb61-17"><a href="survival-analysis.html#cb61-17" aria-hidden="true" tabindex="-1"></a>                          x <span class="ot">&lt;-</span> <span class="fu">summary</span>(x)</span>
<span id="cb61-18"><a href="survival-analysis.html#cb61-18" aria-hidden="true" tabindex="-1"></a>                          p.value<span class="ot">&lt;-</span><span class="fu">signif</span>(x<span class="sc">$</span>wald[<span class="st">&quot;pvalue&quot;</span>], <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb61-19"><a href="survival-analysis.html#cb61-19" aria-hidden="true" tabindex="-1"></a>                          wald.test<span class="ot">&lt;-</span><span class="fu">signif</span>(x<span class="sc">$</span>wald[<span class="st">&quot;test&quot;</span>], <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb61-20"><a href="survival-analysis.html#cb61-20" aria-hidden="true" tabindex="-1"></a>                          beta<span class="ot">&lt;-</span><span class="fu">signif</span>(x<span class="sc">$</span>coef[<span class="dv">1</span>], <span class="at">digits=</span><span class="dv">2</span>);<span class="co">#coeficient beta</span></span>
<span id="cb61-21"><a href="survival-analysis.html#cb61-21" aria-hidden="true" tabindex="-1"></a>                          HR <span class="ot">&lt;-</span><span class="fu">signif</span>(x<span class="sc">$</span>coef[<span class="dv">2</span>], <span class="at">digits=</span><span class="dv">2</span>);<span class="co">#exp(beta)</span></span>
<span id="cb61-22"><a href="survival-analysis.html#cb61-22" aria-hidden="true" tabindex="-1"></a>                          HR.confint.lower <span class="ot">&lt;-</span> <span class="fu">signif</span>(x<span class="sc">$</span>conf.int[,<span class="st">&quot;lower .95&quot;</span>], <span class="dv">2</span>)</span>
<span id="cb61-23"><a href="survival-analysis.html#cb61-23" aria-hidden="true" tabindex="-1"></a>                          HR.confint.upper <span class="ot">&lt;-</span> <span class="fu">signif</span>(x<span class="sc">$</span>conf.int[,<span class="st">&quot;upper .95&quot;</span>],<span class="dv">2</span>)</span>
<span id="cb61-24"><a href="survival-analysis.html#cb61-24" aria-hidden="true" tabindex="-1"></a>                          HR <span class="ot">&lt;-</span> <span class="fu">paste0</span>(HR, <span class="st">&quot; (&quot;</span>, </span>
<span id="cb61-25"><a href="survival-analysis.html#cb61-25" aria-hidden="true" tabindex="-1"></a>                                       HR.confint.lower, <span class="st">&quot;-&quot;</span>, HR.confint.upper, <span class="st">&quot;)&quot;</span>)</span>
<span id="cb61-26"><a href="survival-analysis.html#cb61-26" aria-hidden="true" tabindex="-1"></a>                          res<span class="ot">&lt;-</span><span class="fu">c</span>(beta, HR, wald.test, p.value)</span>
<span id="cb61-27"><a href="survival-analysis.html#cb61-27" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">names</span>(res)<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;beta&quot;</span>, <span class="st">&quot;HR (95% CI for HR)&quot;</span>, <span class="st">&quot;wald.test&quot;</span>, </span>
<span id="cb61-28"><a href="survival-analysis.html#cb61-28" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">&quot;p.value&quot;</span>)</span>
<span id="cb61-29"><a href="survival-analysis.html#cb61-29" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">return</span>(res)</span>
<span id="cb61-30"><a href="survival-analysis.html#cb61-30" aria-hidden="true" tabindex="-1"></a>                          <span class="co"># return(exp(cbind(coef(x),confint(x)))) ## For KI</span></span>
<span id="cb61-31"><a href="survival-analysis.html#cb61-31" aria-hidden="true" tabindex="-1"></a>                         })</span>
<span id="cb61-32"><a href="survival-analysis.html#cb61-32" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">as.data.frame</span>(univ_results, <span class="at">check.names =</span> <span class="cn">FALSE</span>))</span>
<span id="cb61-33"><a href="survival-analysis.html#cb61-33" aria-hidden="true" tabindex="-1"></a><span class="fu">as.data.frame</span>(res)</span></code></pre></div>
<pre><code>##            beta HR (95% CI for HR) wald.test p.value
## age       0.019            1 (1-1)       4.1   0.042
## sex       -0.53   0.59 (0.42-0.82)        10  0.0015
## ph.karno -0.016      0.98 (0.97-1)       7.9   0.005
## ph.ecog    0.48        1.6 (1.3-2)        18 2.7e-05
## wt.loss  0.0013         1 (0.99-1)      0.05    0.83</code></pre>
</div>
</div>
<div id="parameter-test" class="section level3" number="17.5.3">
<h3><span class="header-section-number">17.5.3</span> Parameter test</h3>
<p><strong>Wald Test</strong></p>
<p><span class="math display">\[\quad \hat{\beta} \sim N\left(\beta,\left(X^{T} W X\right)^{-1}\right)\]</span>
<strong>Likelihood Ratio Test</strong></p>
<p>Likelihood Ratio Test (LRT) is easy to use. LRT is often used to determine whether two models (different parameter dimensions) are significantly different. in particular
<span class="math display">\[-2\left(l\left(\hat{\beta}_{1}\right)-l\left(\hat{\beta}_{2}\right)\right) \sim \chi_{p}^{2}, \quad\]</span></p>
<p><strong>Score Test</strong></p>
<p>It is very difficult to obtain the confidence interval with Score Method. But we can use it for hypothesis test
<span class="math display">\[u(\beta)=\sum_{i=1}^{n} \delta_{i} x_{i}-\sum_{i=1}^{n} \sum_{k=1}^{n} Y_{k}\left(t_{i}\right) \frac{w_{k}}{\sum_{j=1}^{n} Y_{j}\left(t_{i}\right) w_{j}} x_{i} \delta_{i}=X^{T}(d-P d)\]</span>
<span class="math display">\[I(\beta)=\sum_{i=1}^{n} \sum_{k=1}^{n} \delta_{i} x_{i}\left[\frac{Y_{k}\left(t_{i}\right) w_{k}}{\sum_{j=1}^{n} Y_{j}\left(t_{i}\right) w_{j}} \frac{x_{k}^{T}\left(\sum_{j=1}^{n} Y_{j}\left(t_{i}\right) w_{j}\right)+\sum_{j=1}^{n} Y_{j}\left(t_{i}\right) w_{j} x_{j}^{T}}{\sum_{j=1}^{n} Y_{j}\left(t_{i}\right) w_{j}}\right]\]</span>
The statistic is <span class="math inline">\(u\left(\beta_{0}\right) / \sqrt{I\left(\beta_{0}\right)}\)</span>, there is no need to calculate <span class="math inline">\(\hat{\beta}\)</span> so as to add it to the model There is no need to refit the parameters for new variables.</p>
</div>
<div id="graphs-of-the-survival-and-baseline-hazard-function-in-sas" class="section level3" number="17.5.4">
<h3><span class="header-section-number">17.5.4</span> Graphs of the survival and baseline hazard function in SAS</h3>
<p><strong>plots=survival</strong></p>
<p>When only plots=survival is specified on the proc phreg statement, SAS will produce one graph, a “reference curve” of the survival function at the reference level of all categorical predictors and at the mean of all continuous predictors.</p>
<pre><code>proc phreg data=whas500 plots=survival;
    class gender;
    model lenfol*fstat(0) = gender age;;
run;</code></pre>
<p><strong>Use the baseline statement to generate survival plots by group</strong></p>
<pre><code>***Whether it is a survival curve or a hazad curve, it is necessary to use 
   the baseline description and create a small data set of covariate values 
   to estimate the curve of interest;
proc format;
    value gender 0 = &quot;male&quot; 1 = &quot;female&quot;;
run;
data covs;
    format gender gender.;
    input gender age;
    datalines;
    0 69.845947
    1 69.845947
    ;
run;

*** covariates = option in baseline statement to specify the dataset 
*** rowid = If a variable is specified after the option, SAS will create a separate 
    line for each level of the variable in the same graph;

proc phreg data = whas500 plots(overlay)=(survival);
    class gender;
    model lenfol*fstat(0) = gender age;
    baseline covariates=covs out=base / rowid=gender;
run;</code></pre>
<p><strong>hazardratio statement</strong></p>
<p>Using the hazardratio statement and graphs to interpret effects, particularly interactions. Below, we show how to use the hazardratio statement to request that SAS estimate 3 hazard ratios at specific levels of our covariates</p>
<ul>
<li>After the keyword hazardratio, we can optionally apply a label, then we specify the variable whose levels are to be compared in the hazard, and finally after the option keyword at we tell SAS at which level of our other covariates to evaluate this hazard ratio. If the variable whose hazard rates are to computed is not involved in an interaction, specification of additional covariates is unncessary since the hazard ratio is constant across levels of all other covariates (a main effect).</li>
<li>We calculate the hazard ratio describing a one-unit increase in age, or <strong>HR(age+1) HR(age)</strong>, for both genders. Notice the =ALL following gender, which is used only with class variables to request the hazard ratio at all levels of the class variable.</li>
<li>We also calculate the hazard ratio between females and males, or <strong>HR(gender=1)HR(gender=0)</strong> at ages 0, 20, 40, 60, and 80.</li>
<li>Finally, we calculate the hazard ratio describing a 5-unit increase in bmi, or <strong>HR(bmi+5) HR(bmi)</strong>, at clinically revelant BMI scores. Notice the additional option units=5. BMI classes are typically separated by about 5 points, so we would like to see how the hazard ratio between (approximately) adjacent BMI classes changes as bmi increases.</li>
</ul>
<pre><code>proc phreg data = whas500;
    class gender;
    model lenfol*fstat(0) = gender|age bmi|bmi hr ;
    hazardratio &#39;Effect of 1-unit change in age by gender&#39; age / at(gender=ALL);
    hazardratio &#39;Effect of gender across ages&#39; gender / at(age=(0 20 40 60 80));
    hazardratio &#39;Effect of 5-unit change in bmi across bmi&#39; bmi / at(bmi = (15 18.5 25 30 40)) units=5;
run;</code></pre>
<table>
<thead>
<tr class="header">
<th>Effect of gender across ages: Hazard Ratios for GENDER</th>
<th>Title</th>
<th>Title</th>
<th>Title</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Description</td>
<td>Point Estimate</td>
<td>95% Wald CI Lower Limits</td>
<td>CI Upper Limits</td>
</tr>
<tr class="even">
<td>GENDER Female vs Male At AGE=0</td>
<td>8.247</td>
<td>1.177</td>
<td>57.783</td>
</tr>
<tr class="odd">
<td>GENDER Female vs Male At AGE=20</td>
<td>4.594</td>
<td>1.064</td>
<td>19.841</td>
</tr>
<tr class="even">
<td>GENDER Female vs Male At AGE=40</td>
<td>2.559</td>
<td>0.955</td>
<td>6.857</td>
</tr>
<tr class="odd">
<td>GENDER Female vs Male At AGE=60</td>
<td>1.426</td>
<td>0.837</td>
<td>2.429</td>
</tr>
<tr class="even">
<td>GENDER Female vs Male At AGE=80</td>
<td>0.794</td>
<td>0.601</td>
<td>1.049</td>
</tr>
</tbody>
</table>
</div>
<div id="check-proportional-hazards" class="section level3" number="17.5.5">
<h3><span class="header-section-number">17.5.5</span> Check proportional hazards</h3>
<p>A central assumption of Cox regression is that covariate effects on the hazard rate, namely <strong>hazard ratios, are constant over time</strong>. For example, if males have twice the hazard rate of females 1 day after followup, the Cox model assumes that males have twice the hazard rate at 1000 days after follow up as well. Violations of the proportional hazard assumption may cause bias in the estimated coefficients as well as incorrect inference regarding significance of effects.</p>
<ol style="list-style-type: decimal">
<li>Graphing Kaplan-Meier survival function estimates to assess proportional hazards for categorical covariates
<ul>
<li>In the case of categorical covariates, graphs of the Kaplan-Meier estimates of the survival function provide quick and easy checks of proportional hazards. If proportional hazards holds, <strong>the graphs of the survival function should look “parallel”</strong>, in the sense that they should have basically the same shape, should not cross, and should start close and then diverge slowly through follow up time.</li>
<li>Another graphical methods for checking proportional hazards is to plot log(-log(S(t))) vs. t or log(t) and look for parallelism. This can be done only for categorical covariates</li>
</ul>
<pre><code>proc lifetest data=whas500 atrisk plots=survival(atrisk cb) outs=outwhas500;
strata gender;
time lenfol*fstat(0);
run;</code></pre></li>
<li>Plotting scaled <strong>Schoenfeld residuals</strong> vs functions of time to assess proportional hazards of a continuous covariate
<ul>
<li><span class="math display">\[E(s^\star_{kp}) + \hat{\beta}_p \approx \beta_j(t_k)\]</span></li>
<li><span class="math inline">\(s^\star_{kp}\)</span> is the scaled Schoenfeld residual of the covariate <span class="math inline">\(p\)</span> at time <span class="math inline">\(k\)</span>, <span class="math inline">\(\beta_p\)</span> is the time-invariant coefficient, and <span class="math inline">\(\beta_j(tk)\)</span> is the time-varying coefficient. In other words, the average value of the Schoenfeld residual of the coefficient <span class="math inline">\(k\)</span> at time <span class="math inline">\(k\)</span> estimates the coefficient change at time <span class="math inline">\(k\)</span>. Therefore, if the average value of the whole time is 0, it means that the coefficient <span class="math inline">\(p\)</span> will not change with time, and the proportional hazard assumption applies to the covariate <span class="math inline">\(p\)</span>.</li>
</ul>
<pre><code>proc phreg data=whas500;
class gender;
model lenfol*fstat(0) = gender|age bmi|bmi hr;
output out=schoen ressch=schgender schage schgenderage
   schbmi schbmibmi schhr;
run;

data schoen;
   set schoen;
   loglenfol = log(lenfol);
run;

proc loess data = schoen;
   model schage=lenfol / smooth=(0.2 0.4 0.6 0.8);
run;
proc loess data = schoen;
   model schage=loglenfol / smooth=(0.2 0.4 0.6 0.8);
run;</code></pre></li>
</ol>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="survival-analysis.html#cb68-1" aria-hidden="true" tabindex="-1"></a>res.cox <span class="ot">&lt;-</span> <span class="fu">coxph</span>(<span class="fu">Surv</span>(time, status) <span class="sc">~</span> age <span class="sc">+</span> sex <span class="sc">+</span> wt.loss, <span class="at">data =</span>  lung)</span>
<span id="cb68-2"><a href="survival-analysis.html#cb68-2" aria-hidden="true" tabindex="-1"></a>test.ph <span class="ot">&lt;-</span> <span class="fu">cox.zph</span>(res.cox)</span>
<span id="cb68-3"><a href="survival-analysis.html#cb68-3" aria-hidden="true" tabindex="-1"></a>test.ph</span></code></pre></div>
<pre><code>##          chisq df    p
## age     0.5077  1 0.48
## sex     2.5489  1 0.11
## wt.loss 0.0144  1 0.90
## GLOBAL  3.0051  3 0.39</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="survival-analysis.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="do">## The test of each covariate is not statistically significant, and the global test is not statistically significant. Therefore, we can take proportional risk.</span></span>
<span id="cb70-2"><a href="survival-analysis.html#cb70-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-3"><a href="survival-analysis.html#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="do">## In principle, Schoenfeld residuals have nothing to do with time. A chart showing a non-random graph showing time changes over time indicates a violation of the PH assumption</span></span>
<span id="cb70-4"><a href="survival-analysis.html#cb70-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-5"><a href="survival-analysis.html#cb70-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcoxzph</span>(test.ph)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/Schoenfeld%20residuals-1.png" width="672" /></p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="survival-analysis.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="do">## The solid line is the smooth spline fit of the curve, and the dashed line represents the +/- 2 standard error band around the fit. System deviation from the horizontal line indicates non-proportional hazard, because proportional hazard assumes that the estimated values β1, β2, and β3 do not change much over time.</span></span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Using assess with the ph option to check proportional hazards
<ul>
<li>Option ph tells SAS that in addition to checking the functional form, we also want to evaluate the proportional hazard. As before, we specify the resampling option to request the highest test of the null hypothesis that the proportional hazard holds. These tests calculate the proportion of simulated score processes that produce a maximum score greater than the observed maximum score process. A small percentage (p-value) indicates that the percentage is violated</li>
</ul>
<pre><code>proc phreg data=whas500;
    class gender;
    model lenfol*fstat(0) = gender|age bmi|bmi hr;
    assess var=(age bmi bmi*bmi hr) ph / resample ;
run;</code></pre>
<table>
<thead>
<tr class="header">
<th>Supremum Test for Proportionals Hazards Assumptio</th>
<th>Title</th>
<th>Title</th>
<th>Title</th>
<th>Title</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Variable</td>
<td>Maximum Absolute Value</td>
<td>Replications</td>
<td>Seed</td>
<td>Pr &gt; MaxAbsVal</td>
</tr>
<tr class="even">
<td>GENDERFemale</td>
<td>0.6394</td>
<td>1000</td>
<td>778428000</td>
<td>0.7680</td>
</tr>
<tr class="odd">
<td>AGE</td>
<td>0.4965</td>
<td>1000</td>
<td>778428000</td>
<td>0.9600</td>
</tr>
<tr class="even">
<td>BMI</td>
<td>5.9813</td>
<td>1000</td>
<td>778428000</td>
<td>0.2890</td>
</tr>
<tr class="odd">
<td>BMIBMI</td>
<td>5.9350</td>
<td>1000</td>
<td>778428000</td>
<td>0.3160</td>
</tr>
<tr class="even">
<td>HR</td>
<td>0.8861</td>
<td>1000</td>
<td>778428000</td>
<td>0.3080</td>
</tr>
</tbody>
</table></li>
</ol>
</div>
<div id="dealing-with-nonproportionality" class="section level3" number="17.5.6">
<h3><span class="header-section-number">17.5.6</span> Dealing with nonproportionality</h3>
<p>If nonproportional hazards are detected, the researcher has many options with how to address the violation (Therneau &amp; Grambsch, 2000):</p>
<p>Ignore the nonproportionality if it appears the changes in the coefficient over time are very small or if it appears the outliers are driving the changes in the coefficient. In large datasets, very small departures from proportional hazards can be detected. If, say, a regression coefficient changes only by 1% over time, it is unlikely that any overarching conclusions of the study would be affected. Additionally, a few heavily influential points may be causing nonproportional hazards to be detected, so it is important to use graphical methods to ensure this is not the case.</p>
<p><strong>Stratify</strong> the model by the nonproportional covariate. Stratification allows each stratum to have its own baseline hazard, which solves the problem of nonproportionality. However, one cannot test whether the stratifying variable itself affects the hazard rate significantly. Additionally, although stratifying by a categorical covariate works naturally, it is often difficult to know how to best discretize a continuous covariate. This can be easily accomplished in proc phreh with the strata statement.</p>
<p>Include <strong>covariate interactions with time</strong> as predictors in the Cox model. This can be accomplished through programming statements in proc phreg, as these interactions are time-varying covariates themselves. Indeed, including such an interaction has been used as a test of proportional hazards — a significant interaction indicates violation of the assumption. Below, we provide code that shows how to include a covariate interaction with time in the model.</p>
</div>
<div id="model-diagnostics-1" class="section level3" number="17.5.7">
<h3><span class="header-section-number">17.5.7</span> Model Diagnostics</h3>
<div id="cox-snell-residuals" class="section level4" number="17.5.7.1">
<h4><span class="header-section-number">17.5.7.1</span> Cox-Snell Residuals</h4>
<p>The residual represents the part that the explanatory variable cannot explain, in the case of censoring
<span class="math display">\[\hat{e}_{i}=\hat{\Lambda}_{0}\left(t_{i}\right) \exp \left(x_{i}^{T} \beta\right)\]</span></p>
<p><span class="math inline">\(\hat(\Lambda)\)</span> can be estimated by <span class="math inline">\(KM\)</span> (or <span class="math inline">\(NA\)</span>). Compare Cox-Snell Residuals with the standard exponential distribution, because <span class="math inline">\(Y=\Lambda(t) \sim \operatorname(Exp)(1)]\)</span> , Get the residual plot. However, this has a little flaw, because it is not easy for us to observe the situation close to the origin. Therefore, we use standardized difference for transformation.
<span class="math display">\[
(\hat{\Lambda}(e)-\Lambda(e)) / S E=(\hat{\Lambda}(e)-e) / S E
\]</span>
However, the Cox-Snell residual is flawed: it has no way to tell us where the assumptions of the model are wrong.</p>
</div>
<div id="martingale-residuals" class="section level4" number="17.5.7.2">
<h4><span class="header-section-number">17.5.7.2</span> Martingale residuals</h4>
<p><span class="math display">\[\hat{m}_{i}=\delta_{i}-\hat{\Lambda}_{i}\left(t_{i}\right)=\delta_{i}-\hat{e}_{i}\]</span></p>
<p>It represents the difference between the observed value of the sample death index <span class="math inline">\(\left(\delta_{i}\right)\)</span> and the average value <span class="math inline">\(\left(\hat{e}_{i}\right)\)</span>. Among them, the average is about the average over time.
Positive and negative are meaningful. Positive, it means that the patient died earlier than the expected time; negative, it means that the patient lived longer (or censored).
But it is asymmetrical. The maximum value of <span class="math inline">\(\quad \hat{m}_{i}\)</span> is 1, but the minimum value is infinite.</p>
</div>
<div id="deviance-residuals" class="section level4" number="17.5.7.3">
<h4><span class="header-section-number">17.5.7.3</span> Deviance Residuals</h4>
<p><span class="math display">\[\hat{d}_{i}=\operatorname{sign}\left(\hat{m}_{i}\right) \sqrt{2\left(\tilde{l}_{i}-l_{i}\right)},\]</span></p>
<p><span class="math inline">\(\hat{m}_{i}=\delta_{i}-\hat{e}_{i}, \quad l_{i}\)</span> is the corresponding likelihood of i sample (given model), <span class="math inline">\(\tilde{l }_{i}\)</span> represents the maximum likelihood (generally the full model).
It retains the positive and negative nature of the residual error, and adds the idea of likelihood (in fact, many tests are based on deviance), which is symmetric.</p>
</div>
<div id="schoenfeld-residuals" class="section level4" number="17.5.7.4">
<h4><span class="header-section-number">17.5.7.4</span> Schoenfeld residuals</h4>
<p>The Schoenfeld residual of the observation j and the covariate p is defined as the difference between the covariate p of the observation j and the weighted average of the covariate values of all subjects who are still at risk when the event occurs on the observation j.
<span class="math display">\[E(s^\star_{kp}) + \hat{\beta}_p \approx \beta_j(t_k)\]</span></p>
<p>Schoenfeld Residuals are used to validate the above assumptions made by the Cox model. The Cox model makes three assumptions:</p>
<ol style="list-style-type: decimal">
<li>Common baseline hazard rate <span class="math inline">\(\lambda(t)\)</span> : At any time <span class="math inline">\(t\)</span>, all individuals are assumed to experience the same baseline hazard <span class="math inline">\(\lambda(t)\)</span>. For example, if a study consists of males and females belonging to different races and age groups, then at any time <span class="math inline">\(t\)</span> during the study, white males who entered the study when they were in the 18-34 years age range are assumed to experience the same baseline hazard <span class="math inline">\(\lambda(t)\)</span> as Asian females who entered the study when they <span class="math inline">\(40-64\)</span> year old. This is clearly a strong assumption and one must validate it before accepting results of the Cox model for your data set.</li>
<li>Time invariant co-variates <span class="math inline">\(X\)</span> : The effect of the regression variables <span class="math inline">\(\boldsymbol{X}\)</span> (a.k.a. co-variates) on the instantaneous hazard experienced by an individual is assumed to remain constant over time. For example, if a volunteer enters a study of cancer risk to smokers, and if the participant’s genetic makeup is a co-variate, then the effect of their genes on the hazard of contracting cancer is assumed to remain constant throughout the study. Notation-wise, <span class="math inline">\(\boldsymbol{X}\left(t_{-} i\right)=\boldsymbol{X}\)</span> for all <span class="math inline">\(t_{-} i\)</span>. This is again a rather strong assumption that should be validated. For example, gene expression can vary with time in response to a number of factors.</li>
<li>Time invariant regression coefficients <span class="math inline">\(\beta:\)</span> The regression coefficients <span class="math inline">\(\boldsymbol{\beta}\)</span> do not vary with time. In other words, <span class="math inline">\(\boldsymbol{\beta}\left(\mathrm{t}_{-} i\right)=\boldsymbol{\beta}\)</span> for all <span class="math inline">\(t_{-} i\)</span>. One can see that assumptions (2) and (3) are closely related. For example, the effect of gene expression increasing with time can be expressed as the coefficient of the gene regression variable increasing with time.</li>
</ol>
</div>
<div id="testing-influential-observations" class="section level4" number="17.5.7.5">
<h4><span class="header-section-number">17.5.7.5</span> Testing influential observations</h4>
<p>To test influential observations or outliers, we can visualize either:</p>
<ul>
<li>the deviance residuals or</li>
<li>the dfbeta values</li>
</ul>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="survival-analysis.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="do">## ggcoxdiagnostics(fit, type = , linear.predictions = TRUE)</span></span>
<span id="cb73-2"><a href="survival-analysis.html#cb73-2" aria-hidden="true" tabindex="-1"></a><span class="do">## type: the type of residuals to present on Y axis. Allowed values include one of </span></span>
<span id="cb73-3"><a href="survival-analysis.html#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="do">## c(“martingale”, “deviance”, “score”, “schoenfeld”, “dfbeta”, “dfbetas”, “scaledsch”, “partial”).</span></span>
<span id="cb73-4"><a href="survival-analysis.html#cb73-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcoxdiagnostics</span>(res.cox, <span class="at">type =</span> <span class="st">&quot;dfbeta&quot;</span>,</span>
<span id="cb73-5"><a href="survival-analysis.html#cb73-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">linear.predictions =</span> <span class="cn">FALSE</span>, <span class="at">ggtheme =</span> <span class="fu">theme_bw</span>())</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/Testing%20influential%20observations-1.png" width="672" /></p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="survival-analysis.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcoxdiagnostics</span>(res.cox, <span class="at">type =</span> <span class="st">&quot;deviance&quot;</span>,</span>
<span id="cb74-2"><a href="survival-analysis.html#cb74-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">linear.predictions =</span> <span class="cn">FALSE</span>, <span class="at">ggtheme =</span> <span class="fu">theme_bw</span>())</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/Testing%20influential%20observations-2.png" width="672" /></p>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="proportion-response-regression.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="advanced-survival-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/17-Survival-Analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
